{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0f6ac7df",
   "metadata": {},
   "source": [
    "## 1. Data Preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bdd0a6db",
   "metadata": {},
   "source": [
    "### R kernel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "31e0e85a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load packages\n",
    "suppressPackageStartupMessages(library(Seurat))\n",
    "suppressPackageStartupMessages(library(dplyr))\n",
    "suppressPackageStartupMessages(library(patchwork))\n",
    "suppressPackageStartupMessages(library(Matrix))\n",
    "suppressPackageStartupMessages(library(parallel))\n",
    "suppressPackageStartupMessages(library(iTALK))\n",
    "suppressPackageStartupMessages(library(CellChat))\n",
    "suppressPackageStartupMessages(library(SingleCellSignalR))\n",
    "suppressPackageStartupMessages(library(CCInx))\n",
    "suppressPackageStartupMessages(library(scMLnet))\n",
    "\n",
    "# load tool functions\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_CellChat.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_iTalk.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_SingleCellSignalR.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_scMLnet.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_CCInx.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/CellPhoneDB process.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/NATMI process.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_CellPhoneDB.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/run_NATMI.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab507691",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load single-cell sample data\n",
    "sampleA <- readRDS(\"https://zenodo.org/record/6497091/files/GSE122960.rds?download=1\")\n",
    "sampleB <- readRDS(\"https://zenodo.org/record/6497091/files/GSE128033.rds?download=1\")\n",
    "sampleC <- readRDS(\"https://zenodo.org/record/6497091/files/GSE135893.rds?download=1\")\n",
    "sampleD <- readRDS(\"https://zenodo.org/record/6497091/files/GSE136831.rds?download=1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6020856c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  ..@ i       : int [1:15028219] 33 35 45 54 55 59 61 72 86 97 ...\n",
      "  ..@ p       : int [1:10414] 0 3337 5097 7069 10281 13289 15511 18031 19740 21390 ...\n",
      "  ..@ Dim     : int [1:2] 33694 10413\n",
      "  ..@ Dimnames:List of 2\n",
      "  .. ..$ : chr [1:33694] \"RP11-34P13.3\" \"FAM138A\" \"OR4F5\" \"RP11-34P13.7\" ...\n",
      "  .. ..$ : chr [1:10413] \"AAACCTGAGCACCGCT-1_2\" \"AAACCTGCACAGATTC-1_2\" \"AAACCTGGTAAGGGCT-1_2\" \"AAACCTGGTCAAACTC-1_2\" ...\n",
      "  ..@ x       : num [1:15028219] 0.686 0.401 1.09 0.401 0.686 ...\n",
      "  ..@ factors : list()\n",
      "Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  ..@ i       : int [1:39789428] 47 59 86 97 153 176 189 197 216 219 ...\n",
      "  ..@ p       : int [1:20628] 0 1379 2847 7202 8373 9651 13150 14990 16263 18922 ...\n",
      "  ..@ Dim     : int [1:2] 45068 20627\n",
      "  ..@ Dimnames:List of 2\n",
      "  .. ..$ : chr [1:45068] \"RP11-34P13.3\" \"FAM138A\" \"OR4F5\" \"RP11-34P13.7\" ...\n",
      "  .. ..$ : chr [1:20627] \"AAACCTGAGAGGACGG-1_2\" \"AAACCTGAGCGTCAAG-1_2\" \"AAACCTGCAGGAATGC-1_2\" \"AAACCTGCATTCCTCG-1_2\" ...\n",
      "  ..@ x       : num [1:39789428] 1.2 1.73 1.2 1.2 2.97 ...\n",
      "  ..@ factors : list()\n",
      "Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  ..@ i       : int [1:60386718] 27 37 49 54 59 61 63 67 69 72 ...\n",
      "  ..@ p       : int [1:28779] 0 4133 7586 9719 11958 14647 16357 19497 22648 25794 ...\n",
      "  ..@ Dim     : int [1:2] 33694 28778\n",
      "  ..@ Dimnames:List of 2\n",
      "  .. ..$ : chr [1:33694] \"RP11-34P13.3\" \"FAM138A\" \"OR4F5\" \"RP11-34P13.7\" ...\n",
      "  .. ..$ : chr [1:28778] \"F00431_AACACGTGTCTAACGT\" \"F00431_CTGAAGTGTATAATGG\" \"F01172_CGAATGTAGGTAGCTG\" \"F01172_CTCTACGAGGTAGCTG\" ...\n",
      "  ..@ x       : num [1:60386718] 0.436 0.436 0.436 0.436 0.738 ...\n",
      "  ..@ factors : list()\n",
      "Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  ..@ i       : int [1:178143321] 8 22 23 27 39 40 42 53 59 63 ...\n",
      "  ..@ p       : int [1:78431] 0 1812 3673 6039 7531 9442 11531 14945 17378 19635 ...\n",
      "  ..@ Dim     : int [1:2] 45947 78430\n",
      "  ..@ Dimnames:List of 2\n",
      "  .. ..$ : chr [1:45947] \"TSPAN6\" \"TNMD\" \"DPM1\" \"SCYL3\" ...\n",
      "  .. ..$ : chr [1:78430] \"010I_AAACCTGAGAAGCCCA\" \"010I_AAACCTGAGCTAAGAT\" \"010I_AAACCTGAGTACGTAA\" \"010I_AAACCTGAGTCGAGTG\" ...\n",
      "  ..@ x       : num [1:178143321] 1.15 1.15 2.01 1.15 2.27 ...\n",
      "  ..@ factors : list()\n"
     ]
    }
   ],
   "source": [
    "# normalized data of the sample\n",
    "str(sampleA@assays$RNA@data)\n",
    "str(sampleB@assays$RNA@data)\n",
    "str(sampleC@assays$RNA@data)\n",
    "str(sampleD@assays$RNA@data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7c2be687",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "        AT1         AT2 Endothelial  Fibroblast  Macrophage        Mast \n",
       "        210        2293         299         147        5892         153 \n",
       "   Monocyte       Tcell \n",
       "        774         645 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        AT1         AT2 Endothelial  Fibroblast  Macrophage        Mast \n",
       "       1233        1548        2553        3420        7622        1405 \n",
       "   Monocyte       Tcell \n",
       "       1153        1693 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        AT1         AT2 Endothelial  Fibroblast  Macrophage        Mast \n",
       "         96        3225        3496         152       16507         360 \n",
       "   Monocyte       Tcell \n",
       "       2113        2829 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        AT1         AT2 Endothelial  Fibroblast  Macrophage        Mast \n",
       "        176         496        3379        1443       60483         572 \n",
       "   Monocyte       Tcell \n",
       "       6793        5088 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# cell types and amounts of the sample\n",
    "table(sampleA$cell.type)\n",
    "table(sampleB$cell.type)\n",
    "table(sampleC$cell.type)\n",
    "table(sampleD$cell.type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "48d91397",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t208 obs. of  4 variables:\n",
      " $ source  : chr  \"AT1\" \"AT1\" \"AT1\" \"AT1\" ...\n",
      " $ target  : chr  \"Monocyte\" \"Monocyte\" \"Monocyte\" \"Macrophage\" ...\n",
      " $ ligand  : chr  \"TGFB1\" \"TGFB1\" \"TGFB1\" \"TGFB1\" ...\n",
      " $ receptor: chr  \"TGFBR1\" \"TGFBR2\" \"TGFBR3\" \"TGFBR1\" ...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1</td><td>Monocyte  </td><td>TGFB1</td><td>TGFBR1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1</td><td>Monocyte  </td><td>TGFB1</td><td>TGFBR2</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT1</td><td>Monocyte  </td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT1</td><td>Macrophage</td><td>TGFB1</td><td>TGFBR1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AT1</td><td>Macrophage</td><td>TGFB1</td><td>TGFBR2</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AT1</td><td>Macrophage</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & source & target & ligand & receptor\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1 & Monocyte   & TGFB1 & TGFBR1\\\\\n",
       "\t2 & AT1 & Monocyte   & TGFB1 & TGFBR2\\\\\n",
       "\t3 & AT1 & Monocyte   & TGFB1 & TGFBR3\\\\\n",
       "\t4 & AT1 & Macrophage & TGFB1 & TGFBR1\\\\\n",
       "\t5 & AT1 & Macrophage & TGFB1 & TGFBR2\\\\\n",
       "\t6 & AT1 & Macrophage & TGFB1 & TGFBR3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1 | Monocyte   | TGFB1 | TGFBR1 |\n",
       "| 2 | AT1 | Monocyte   | TGFB1 | TGFBR2 |\n",
       "| 3 | AT1 | Monocyte   | TGFB1 | TGFBR3 |\n",
       "| 4 | AT1 | Macrophage | TGFB1 | TGFBR1 |\n",
       "| 5 | AT1 | Macrophage | TGFB1 | TGFBR2 |\n",
       "| 6 | AT1 | Macrophage | TGFB1 | TGFBR3 |\n",
       "\n"
      ],
      "text/plain": [
       "  source target     ligand receptor\n",
       "1 AT1    Monocyte   TGFB1  TGFBR1  \n",
       "2 AT1    Monocyte   TGFB1  TGFBR2  \n",
       "3 AT1    Monocyte   TGFB1  TGFBR3  \n",
       "4 AT1    Macrophage TGFB1  TGFBR1  \n",
       "5 AT1    Macrophage TGFB1  TGFBR2  \n",
       "6 AT1    Macrophage TGFB1  TGFBR3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load gold standard sample\n",
    "gold <- read.table(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/data/IPF gold standard.txt\", header = TRUE, sep = \"\\t\")\n",
    "str(gold)\n",
    "head(gold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "23a66ede",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in rec$Receptor == genes:\n",
      "“longer object length is not a multiple of shorter object length”\n",
      "Warning message in tf$TF == rec$TF:\n",
      "“longer object length is not a multiple of shorter object length”\n"
     ]
    }
   ],
   "source": [
    "# remove unwanted genes for scMLnet\n",
    "genes <- unique(c(gold$ligand, gold$receptor))\n",
    "rec <- read.table(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/scMLnet_database/RecTF.txt\", header = T, sep = \"\\t\")\n",
    "tf <- read.table(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/scMLnet_database/TFTargetGene.txt\", header = T, sep = \"\\t\")\n",
    "rec <- rec[rec$Receptor == genes,]\n",
    "tf <- tf[tf$TF == rec$TF,]\n",
    "genes2 <- unique(c(genes, rec$TF, tf$Targets))\n",
    "sampleA2 <- subset(sampleA, features = genes2)\n",
    "sampleB2 <- subset(sampleB, features = genes2)\n",
    "sampleC2 <- subset(sampleC, features = genes2)\n",
    "sampleD2 <- subset(sampleD, features = genes2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "65b541f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove unwanted genes for CellChat, iTalk, SingleCellSignalR, CCInx, CellPhoneDB, and NATMI\n",
    "sampleA <- subset(sampleA, features = genes) \n",
    "sampleB <- subset(sampleB, features = genes)\n",
    "sampleC <- subset(sampleC, features = genes) \n",
    "sampleD <- subset(sampleD, features = genes) "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16eeeccb",
   "metadata": {},
   "source": [
    "## 2. Cell-cell Interaction Prediction "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "90fe23c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Create a CellChat object from a data matrix\n",
      "\n",
      "Set cell identities for the new CellChat object\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The cell groups used for CellChat analysis are  Macrophage Fibroblast AT2 Monocyte Tcell AT1 Endothelial Mast \n",
      "[1] \"Total predicted L-R pairs: 1\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Create a CellChat object from a data matrix\n",
      "\n",
      "Set cell identities for the new CellChat object\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The cell groups used for CellChat analysis are  Endothelial AT2 Macrophage Monocyte Fibroblast Tcell Mast AT1 \n",
      "[1] \"Total predicted L-R pairs: 7\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Create a CellChat object from a data matrix\n",
      "\n",
      "Set cell identities for the new CellChat object\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The cell groups used for CellChat analysis are  AT2 AT1 Endothelial Mast Monocyte Tcell Macrophage Fibroblast \n",
      "[1] \"Total predicted L-R pairs: 8\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Create a CellChat object from a data matrix\n",
      "\n",
      "Set cell identities for the new CellChat object\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The cell groups used for CellChat analysis are  Macrophage Monocyte Tcell Mast Fibroblast AT2 Endothelial AT1 \n",
      "[1] \"Total predicted L-R pairs: 15\"\n"
     ]
    }
   ],
   "source": [
    "# CellChat\n",
    "CellChatA <- run_CellChat(sampleA)\n",
    "CellChatB <- run_CellChat(sampleB)\n",
    "CellChatC <- run_CellChat(sampleC)\n",
    "CellChatD <- run_CellChat(sampleD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f268db76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>prob</th><th scope=col>pval</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1</td><td>Fibroblast</td><td>PDGFA</td><td>PDGFRB</td><td>0.002962098</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & prob & pval\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & AT1 & Fibroblast & PDGFA & PDGFRB & 0.002962098 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | prob &lt;dbl&gt; | pval &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | AT1 | Fibroblast | PDGFA | PDGFRB | 0.002962098 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  source target     ligand receptor prob        pval\n",
       "1 AT1    Fibroblast PDGFA  PDGFRB   0.002962098 0   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>prob</th><th scope=col>pval</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Macrophage</td><td>Macrophage</td><td>TGFB1</td><td>TGFBR1</td><td>0.0014066582</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Macrophage</td><td>Macrophage</td><td>TGFB1</td><td>TGFBR2</td><td>0.0014066582</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>21</th><td>Monocyte  </td><td>Macrophage</td><td>TGFB1</td><td>TGFBR1</td><td>0.0008797044</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>Monocyte  </td><td>Macrophage</td><td>TGFB1</td><td>TGFBR2</td><td>0.0008797044</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Macrophage</td><td>Macrophage</td><td>CCL3 </td><td>CCR1  </td><td>0.0013310644</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Monocyte  </td><td>Macrophage</td><td>CCL3 </td><td>CCR1  </td><td>0.0013381925</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & prob & pval\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & Macrophage & Macrophage & TGFB1 & TGFBR1 & 0.0014066582 & 0\\\\\n",
       "\t2 & Macrophage & Macrophage & TGFB1 & TGFBR2 & 0.0014066582 & 0\\\\\n",
       "\t21 & Monocyte   & Macrophage & TGFB1 & TGFBR1 & 0.0008797044 & 0\\\\\n",
       "\t22 & Monocyte   & Macrophage & TGFB1 & TGFBR2 & 0.0008797044 & 0\\\\\n",
       "\t3 & Macrophage & Macrophage & CCL3  & CCR1   & 0.0013310644 & 0\\\\\n",
       "\t4 & Monocyte   & Macrophage & CCL3  & CCR1   & 0.0013381925 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | prob &lt;dbl&gt; | pval &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | Macrophage | Macrophage | TGFB1 | TGFBR1 | 0.0014066582 | 0 |\n",
       "| 2 | Macrophage | Macrophage | TGFB1 | TGFBR2 | 0.0014066582 | 0 |\n",
       "| 21 | Monocyte   | Macrophage | TGFB1 | TGFBR1 | 0.0008797044 | 0 |\n",
       "| 22 | Monocyte   | Macrophage | TGFB1 | TGFBR2 | 0.0008797044 | 0 |\n",
       "| 3 | Macrophage | Macrophage | CCL3  | CCR1   | 0.0013310644 | 0 |\n",
       "| 4 | Monocyte   | Macrophage | CCL3  | CCR1   | 0.0013381925 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "   source     target     ligand receptor prob         pval\n",
       "1  Macrophage Macrophage TGFB1  TGFBR1   0.0014066582 0   \n",
       "2  Macrophage Macrophage TGFB1  TGFBR2   0.0014066582 0   \n",
       "21 Monocyte   Macrophage TGFB1  TGFBR1   0.0008797044 0   \n",
       "22 Monocyte   Macrophage TGFB1  TGFBR2   0.0008797044 0   \n",
       "3  Macrophage Macrophage CCL3   CCR1     0.0013310644 0   \n",
       "4  Monocyte   Macrophage CCL3   CCR1     0.0013381925 0   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>prob</th><th scope=col>pval</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1       </td><td>Fibroblast</td><td>PDGFA</td><td>PDGFRA</td><td>0.012570570</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1       </td><td>Fibroblast</td><td>PDGFA</td><td>PDGFRB</td><td>0.010027325</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Monocyte  </td><td>Macrophage</td><td>CCL3 </td><td>CCR1  </td><td>0.002577753</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Macrophage</td><td>Macrophage</td><td>CCL3 </td><td>CCR1  </td><td>0.002839586</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Mast      </td><td>Monocyte  </td><td>CSF1 </td><td>CSF1R </td><td>0.005637249</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Fibroblast</td><td>Monocyte  </td><td>CSF1 </td><td>CSF1R </td><td>0.005129437</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & prob & pval\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & AT1        & Fibroblast & PDGFA & PDGFRA & 0.012570570 & 0\\\\\n",
       "\t2 & AT1        & Fibroblast & PDGFA & PDGFRB & 0.010027325 & 0\\\\\n",
       "\t3 & Monocyte   & Macrophage & CCL3  & CCR1   & 0.002577753 & 0\\\\\n",
       "\t4 & Macrophage & Macrophage & CCL3  & CCR1   & 0.002839586 & 0\\\\\n",
       "\t5 & Mast       & Monocyte   & CSF1  & CSF1R  & 0.005637249 & 0\\\\\n",
       "\t6 & Fibroblast & Monocyte   & CSF1  & CSF1R  & 0.005129437 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | prob &lt;dbl&gt; | pval &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | AT1        | Fibroblast | PDGFA | PDGFRA | 0.012570570 | 0 |\n",
       "| 2 | AT1        | Fibroblast | PDGFA | PDGFRB | 0.010027325 | 0 |\n",
       "| 3 | Monocyte   | Macrophage | CCL3  | CCR1   | 0.002577753 | 0 |\n",
       "| 4 | Macrophage | Macrophage | CCL3  | CCR1   | 0.002839586 | 0 |\n",
       "| 5 | Mast       | Monocyte   | CSF1  | CSF1R  | 0.005637249 | 0 |\n",
       "| 6 | Fibroblast | Monocyte   | CSF1  | CSF1R  | 0.005129437 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  source     target     ligand receptor prob        pval\n",
       "1 AT1        Fibroblast PDGFA  PDGFRA   0.012570570 0   \n",
       "2 AT1        Fibroblast PDGFA  PDGFRB   0.010027325 0   \n",
       "3 Monocyte   Macrophage CCL3   CCR1     0.002577753 0   \n",
       "4 Macrophage Macrophage CCL3   CCR1     0.002839586 0   \n",
       "5 Mast       Monocyte   CSF1   CSF1R    0.005637249 0   \n",
       "6 Fibroblast Monocyte   CSF1   CSF1R    0.005129437 0   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>prob</th><th scope=col>pval</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Macrophage</td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR1</td><td>0.010531960</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Macrophage</td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR2</td><td>0.010531960</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>21</th><td>Monocyte  </td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR1</td><td>0.004239185</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>Monocyte  </td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR2</td><td>0.004239185</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Tcell     </td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR1</td><td>0.015503463</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>31</th><td>Tcell     </td><td>Fibroblast</td><td>TGFB1</td><td>TGFBR2</td><td>0.015503463</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & prob & pval\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & Macrophage & Fibroblast & TGFB1 & TGFBR1 & 0.010531960 & 0\\\\\n",
       "\t2 & Macrophage & Fibroblast & TGFB1 & TGFBR2 & 0.010531960 & 0\\\\\n",
       "\t21 & Monocyte   & Fibroblast & TGFB1 & TGFBR1 & 0.004239185 & 0\\\\\n",
       "\t22 & Monocyte   & Fibroblast & TGFB1 & TGFBR2 & 0.004239185 & 0\\\\\n",
       "\t3 & Tcell      & Fibroblast & TGFB1 & TGFBR1 & 0.015503463 & 0\\\\\n",
       "\t31 & Tcell      & Fibroblast & TGFB1 & TGFBR2 & 0.015503463 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | prob &lt;dbl&gt; | pval &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | Macrophage | Fibroblast | TGFB1 | TGFBR1 | 0.010531960 | 0 |\n",
       "| 2 | Macrophage | Fibroblast | TGFB1 | TGFBR2 | 0.010531960 | 0 |\n",
       "| 21 | Monocyte   | Fibroblast | TGFB1 | TGFBR1 | 0.004239185 | 0 |\n",
       "| 22 | Monocyte   | Fibroblast | TGFB1 | TGFBR2 | 0.004239185 | 0 |\n",
       "| 3 | Tcell      | Fibroblast | TGFB1 | TGFBR1 | 0.015503463 | 0 |\n",
       "| 31 | Tcell      | Fibroblast | TGFB1 | TGFBR2 | 0.015503463 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "   source     target     ligand receptor prob        pval\n",
       "1  Macrophage Fibroblast TGFB1  TGFBR1   0.010531960 0   \n",
       "2  Macrophage Fibroblast TGFB1  TGFBR2   0.010531960 0   \n",
       "21 Monocyte   Fibroblast TGFB1  TGFBR1   0.004239185 0   \n",
       "22 Monocyte   Fibroblast TGFB1  TGFBR2   0.004239185 0   \n",
       "3  Tcell      Fibroblast TGFB1  TGFBR1   0.015503463 0   \n",
       "31 Tcell      Fibroblast TGFB1  TGFBR2   0.015503463 0   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(CellChatA)\n",
    "head(CellChatB)\n",
    "head(CellChatC)\n",
    "head(CellChatD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2d41b2af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 509\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n",
      "[1] \"Total predicted L-R pairs: 511\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n",
      "[1] \"Total predicted L-R pairs: 566\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[========>--------------------------------------------------------------]  12%\n",
      "\n",
      "[=================>-----------------------------------------------------]  25%\n",
      "\n",
      "[==========================>--------------------------------------------]  38%\n",
      "\n",
      "[===================================>-----------------------------------]  50%\n",
      "\n",
      "[===========================================>---------------------------]  62%\n",
      "\n",
      "[====================================================>------------------]  75%\n",
      "\n",
      "[=============================================================>---------]  88%\n",
      "\n",
      "[=======================================================================] 100%\n",
      "                                                                              \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 407\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    }
   ],
   "source": [
    "# iTalk\n",
    "iTalkA <- run_iTalk(sampleA, top = 208)\n",
    "iTalkB <- run_iTalk(sampleB, top = 208)\n",
    "iTalkC <- run_iTalk(sampleC, top = 208)\n",
    "iTalkD <- run_iTalk(sampleD, top = 208)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a3202010",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>source_mean_exprs</th><th scope=col>target_mean_exprs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>23</th><td>Endothelial</td><td>Fibroblast</td><td>CTGF</td><td>LRP1</td><td>1.2286853</td><td>0.8454993</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>Endothelial</td><td>Macrophage</td><td>CTGF</td><td>LRP1</td><td>1.2286853</td><td>0.5840580</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Fibroblast </td><td>Fibroblast</td><td>CTGF</td><td>LRP1</td><td>0.8118479</td><td>0.8454993</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>AT1        </td><td>Fibroblast</td><td>CTGF</td><td>LRP1</td><td>0.6911836</td><td>0.8454993</td></tr>\n",
       "\t<tr><th scope=row>1</th><td>Fibroblast </td><td>Macrophage</td><td>CTGF</td><td>LRP1</td><td>0.8118479</td><td>0.5840580</td></tr>\n",
       "\t<tr><th scope=row>25</th><td>Endothelial</td><td>Monocyte  </td><td>CTGF</td><td>LRP1</td><td>1.2286853</td><td>0.3453390</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & source\\_mean\\_exprs & target\\_mean\\_exprs\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t23 & Endothelial & Fibroblast & CTGF & LRP1 & 1.2286853 & 0.8454993\\\\\n",
       "\t22 & Endothelial & Macrophage & CTGF & LRP1 & 1.2286853 & 0.5840580\\\\\n",
       "\t2 & Fibroblast  & Fibroblast & CTGF & LRP1 & 0.8118479 & 0.8454993\\\\\n",
       "\t16 & AT1         & Fibroblast & CTGF & LRP1 & 0.6911836 & 0.8454993\\\\\n",
       "\t1 & Fibroblast  & Macrophage & CTGF & LRP1 & 0.8118479 & 0.5840580\\\\\n",
       "\t25 & Endothelial & Monocyte   & CTGF & LRP1 & 1.2286853 & 0.3453390\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | source_mean_exprs &lt;dbl&gt; | target_mean_exprs &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 23 | Endothelial | Fibroblast | CTGF | LRP1 | 1.2286853 | 0.8454993 |\n",
       "| 22 | Endothelial | Macrophage | CTGF | LRP1 | 1.2286853 | 0.5840580 |\n",
       "| 2 | Fibroblast  | Fibroblast | CTGF | LRP1 | 0.8118479 | 0.8454993 |\n",
       "| 16 | AT1         | Fibroblast | CTGF | LRP1 | 0.6911836 | 0.8454993 |\n",
       "| 1 | Fibroblast  | Macrophage | CTGF | LRP1 | 0.8118479 | 0.5840580 |\n",
       "| 25 | Endothelial | Monocyte   | CTGF | LRP1 | 1.2286853 | 0.3453390 |\n",
       "\n"
      ],
      "text/plain": [
       "   source      target     ligand receptor source_mean_exprs target_mean_exprs\n",
       "23 Endothelial Fibroblast CTGF   LRP1     1.2286853         0.8454993        \n",
       "22 Endothelial Macrophage CTGF   LRP1     1.2286853         0.5840580        \n",
       "2  Fibroblast  Fibroblast CTGF   LRP1     0.8118479         0.8454993        \n",
       "16 AT1         Fibroblast CTGF   LRP1     0.6911836         0.8454993        \n",
       "1  Fibroblast  Macrophage CTGF   LRP1     0.8118479         0.5840580        \n",
       "25 Endothelial Monocyte   CTGF   LRP1     1.2286853         0.3453390        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>source_mean_exprs</th><th scope=col>target_mean_exprs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1401</th><td>Monocyte  </td><td>Endothelial</td><td>IL1B</td><td>IL1R1</td><td>1.742668</td><td>0.6223556</td></tr>\n",
       "\t<tr><th scope=row>841</th><td>Tcell     </td><td>Macrophage </td><td>CCL4</td><td>CCR1 </td><td>3.764675</td><td>0.2709540</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>Fibroblast</td><td>Fibroblast </td><td>CTGF</td><td>LRP1 </td><td>1.186583</td><td>0.7871848</td></tr>\n",
       "\t<tr><th scope=row>1441</th><td>Monocyte  </td><td>Fibroblast </td><td>IL1B</td><td>IL1R1</td><td>1.742668</td><td>0.4734596</td></tr>\n",
       "\t<tr><th scope=row>851</th><td>Tcell     </td><td>Monocyte   </td><td>CCL4</td><td>CCR1 </td><td>3.764675</td><td>0.2031794</td></tr>\n",
       "\t<tr><th scope=row>1411</th><td>Monocyte  </td><td>AT2        </td><td>IL1B</td><td>IL1R1</td><td>1.742668</td><td>0.3727768</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & source\\_mean\\_exprs & target\\_mean\\_exprs\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1401 & Monocyte   & Endothelial & IL1B & IL1R1 & 1.742668 & 0.6223556\\\\\n",
       "\t841 & Tcell      & Macrophage  & CCL4 & CCR1  & 3.764675 & 0.2709540\\\\\n",
       "\t14 & Fibroblast & Fibroblast  & CTGF & LRP1  & 1.186583 & 0.7871848\\\\\n",
       "\t1441 & Monocyte   & Fibroblast  & IL1B & IL1R1 & 1.742668 & 0.4734596\\\\\n",
       "\t851 & Tcell      & Monocyte    & CCL4 & CCR1  & 3.764675 & 0.2031794\\\\\n",
       "\t1411 & Monocyte   & AT2         & IL1B & IL1R1 & 1.742668 & 0.3727768\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | source_mean_exprs &lt;dbl&gt; | target_mean_exprs &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1401 | Monocyte   | Endothelial | IL1B | IL1R1 | 1.742668 | 0.6223556 |\n",
       "| 841 | Tcell      | Macrophage  | CCL4 | CCR1  | 3.764675 | 0.2709540 |\n",
       "| 14 | Fibroblast | Fibroblast  | CTGF | LRP1  | 1.186583 | 0.7871848 |\n",
       "| 1441 | Monocyte   | Fibroblast  | IL1B | IL1R1 | 1.742668 | 0.4734596 |\n",
       "| 851 | Tcell      | Monocyte    | CCL4 | CCR1  | 3.764675 | 0.2031794 |\n",
       "| 1411 | Monocyte   | AT2         | IL1B | IL1R1 | 1.742668 | 0.3727768 |\n",
       "\n"
      ],
      "text/plain": [
       "     source     target      ligand receptor source_mean_exprs target_mean_exprs\n",
       "1401 Monocyte   Endothelial IL1B   IL1R1    1.742668          0.6223556        \n",
       "841  Tcell      Macrophage  CCL4   CCR1     3.764675          0.2709540        \n",
       "14   Fibroblast Fibroblast  CTGF   LRP1     1.186583          0.7871848        \n",
       "1441 Monocyte   Fibroblast  IL1B   IL1R1    1.742668          0.4734596        \n",
       "851  Tcell      Monocyte    CCL4   CCR1     3.764675          0.2031794        \n",
       "1411 Monocyte   AT2         IL1B   IL1R1    1.742668          0.3727768        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>source_mean_exprs</th><th scope=col>target_mean_exprs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>18</th><td>Endothelial</td><td>Fibroblast</td><td>CTGF </td><td>LRP1  </td><td>1.0213757</td><td>1.703772</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>AT1        </td><td>Fibroblast</td><td>CTGF </td><td>LRP1  </td><td>0.6978218</td><td>1.703772</td></tr>\n",
       "\t<tr><th scope=row>24</th><td>Fibroblast </td><td>Fibroblast</td><td>CTGF </td><td>LRP1  </td><td>0.6749418</td><td>1.703772</td></tr>\n",
       "\t<tr><th scope=row>132</th><td>Monocyte   </td><td>AT1       </td><td>TGFB1</td><td>ITGB6 </td><td>0.6380866</td><td>1.557565</td></tr>\n",
       "\t<tr><th scope=row>236</th><td>Monocyte   </td><td>AT1       </td><td>TGFB1</td><td>TGFBR2</td><td>0.6380866</td><td>1.421618</td></tr>\n",
       "\t<tr><th scope=row>136</th><td>Macrophage </td><td>AT1       </td><td>TGFB1</td><td>ITGB6 </td><td>0.5808058</td><td>1.557565</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & source\\_mean\\_exprs & target\\_mean\\_exprs\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t18 & Endothelial & Fibroblast & CTGF  & LRP1   & 1.0213757 & 1.703772\\\\\n",
       "\t12 & AT1         & Fibroblast & CTGF  & LRP1   & 0.6978218 & 1.703772\\\\\n",
       "\t24 & Fibroblast  & Fibroblast & CTGF  & LRP1   & 0.6749418 & 1.703772\\\\\n",
       "\t132 & Monocyte    & AT1        & TGFB1 & ITGB6  & 0.6380866 & 1.557565\\\\\n",
       "\t236 & Monocyte    & AT1        & TGFB1 & TGFBR2 & 0.6380866 & 1.421618\\\\\n",
       "\t136 & Macrophage  & AT1        & TGFB1 & ITGB6  & 0.5808058 & 1.557565\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | source_mean_exprs &lt;dbl&gt; | target_mean_exprs &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 18 | Endothelial | Fibroblast | CTGF  | LRP1   | 1.0213757 | 1.703772 |\n",
       "| 12 | AT1         | Fibroblast | CTGF  | LRP1   | 0.6978218 | 1.703772 |\n",
       "| 24 | Fibroblast  | Fibroblast | CTGF  | LRP1   | 0.6749418 | 1.703772 |\n",
       "| 132 | Monocyte    | AT1        | TGFB1 | ITGB6  | 0.6380866 | 1.557565 |\n",
       "| 236 | Monocyte    | AT1        | TGFB1 | TGFBR2 | 0.6380866 | 1.421618 |\n",
       "| 136 | Macrophage  | AT1        | TGFB1 | ITGB6  | 0.5808058 | 1.557565 |\n",
       "\n"
      ],
      "text/plain": [
       "    source      target     ligand receptor source_mean_exprs target_mean_exprs\n",
       "18  Endothelial Fibroblast CTGF   LRP1     1.0213757         1.703772         \n",
       "12  AT1         Fibroblast CTGF   LRP1     0.6978218         1.703772         \n",
       "24  Fibroblast  Fibroblast CTGF   LRP1     0.6749418         1.703772         \n",
       "132 Monocyte    AT1        TGFB1  ITGB6    0.6380866         1.557565         \n",
       "236 Monocyte    AT1        TGFB1  TGFBR2   0.6380866         1.421618         \n",
       "136 Macrophage  AT1        TGFB1  ITGB6    0.5808058         1.557565         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>source_mean_exprs</th><th scope=col>target_mean_exprs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>134</th><td>Tcell     </td><td>AT1        </td><td>TGFB1</td><td>ITGB6 </td><td>1.1480619</td><td>1.589999</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Fibroblast</td><td>Fibroblast </td><td>CTGF </td><td>LRP1  </td><td>1.3232609</td><td>1.176309</td></tr>\n",
       "\t<tr><th scope=row>236</th><td>Tcell     </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR2</td><td>1.1480619</td><td>1.199099</td></tr>\n",
       "\t<tr><th scope=row>237</th><td>Tcell     </td><td>AT1        </td><td>TGFB1</td><td>TGFBR2</td><td>1.1480619</td><td>1.164342</td></tr>\n",
       "\t<tr><th scope=row>289</th><td>Tcell     </td><td>Fibroblast </td><td>TGFB1</td><td>TGFBR3</td><td>1.1480619</td><td>1.158633</td></tr>\n",
       "\t<tr><th scope=row>128</th><td>Macrophage</td><td>AT1        </td><td>TGFB1</td><td>ITGB6 </td><td>0.8076374</td><td>1.589999</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & source & target & ligand & receptor & source\\_mean\\_exprs & target\\_mean\\_exprs\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t134 & Tcell      & AT1         & TGFB1 & ITGB6  & 1.1480619 & 1.589999\\\\\n",
       "\t5 & Fibroblast & Fibroblast  & CTGF  & LRP1   & 1.3232609 & 1.176309\\\\\n",
       "\t236 & Tcell      & Endothelial & TGFB1 & TGFBR2 & 1.1480619 & 1.199099\\\\\n",
       "\t237 & Tcell      & AT1         & TGFB1 & TGFBR2 & 1.1480619 & 1.164342\\\\\n",
       "\t289 & Tcell      & Fibroblast  & TGFB1 & TGFBR3 & 1.1480619 & 1.158633\\\\\n",
       "\t128 & Macrophage & AT1         & TGFB1 & ITGB6  & 0.8076374 & 1.589999\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | source_mean_exprs &lt;dbl&gt; | target_mean_exprs &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 134 | Tcell      | AT1         | TGFB1 | ITGB6  | 1.1480619 | 1.589999 |\n",
       "| 5 | Fibroblast | Fibroblast  | CTGF  | LRP1   | 1.3232609 | 1.176309 |\n",
       "| 236 | Tcell      | Endothelial | TGFB1 | TGFBR2 | 1.1480619 | 1.199099 |\n",
       "| 237 | Tcell      | AT1         | TGFB1 | TGFBR2 | 1.1480619 | 1.164342 |\n",
       "| 289 | Tcell      | Fibroblast  | TGFB1 | TGFBR3 | 1.1480619 | 1.158633 |\n",
       "| 128 | Macrophage | AT1         | TGFB1 | ITGB6  | 0.8076374 | 1.589999 |\n",
       "\n"
      ],
      "text/plain": [
       "    source     target      ligand receptor source_mean_exprs target_mean_exprs\n",
       "134 Tcell      AT1         TGFB1  ITGB6    1.1480619         1.589999         \n",
       "5   Fibroblast Fibroblast  CTGF   LRP1     1.3232609         1.176309         \n",
       "236 Tcell      Endothelial TGFB1  TGFBR2   1.1480619         1.199099         \n",
       "237 Tcell      AT1         TGFB1  TGFBR2   1.1480619         1.164342         \n",
       "289 Tcell      Fibroblast  TGFB1  TGFBR3   1.1480619         1.158633         \n",
       "128 Macrophage AT1         TGFB1  ITGB6    0.8076374         1.589999         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(iTalkA)\n",
    "head(iTalkB)\n",
    "head(iTalkC)\n",
    "head(iTalkD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "2e4fd4d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No such file as table_dge_Macrophage.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Fibroblast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT2.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Monocyte.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Tcell.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT1.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Endothelial.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Mast.txt in the cluster-analysis folder\n",
      "Paracrine signaling: \n",
      "Checking for signaling between cell types\n",
      "0 No significant interaction found from Macrophage to Fibroblast\n",
      "0 No significant interaction found from Macrophage to AT2\n",
      "0 No significant interaction found from Macrophage to Monocyte\n",
      "0 No significant interaction found from Macrophage to Tcell\n",
      "0 No significant interaction found from Macrophage to AT1\n",
      "0 No significant interaction found from Macrophage to Endothelial\n",
      "0 No significant interaction found from Macrophage to Mast\n",
      "0 No significant interaction found from Fibroblast to Macrophage\n",
      "0 No significant interaction found from Fibroblast to AT2\n",
      "1 interactions from Fibroblast to Monocyte\n",
      "0 No significant interaction found from Fibroblast to Tcell\n",
      "0 No significant interaction found from Fibroblast to AT1\n",
      "0 No significant interaction found from Fibroblast to Endothelial\n",
      "0 No significant interaction found from Fibroblast to Mast\n",
      "0 No significant interaction found from AT2 to Macrophage\n",
      "0 No significant interaction found from AT2 to Fibroblast\n",
      "0 No significant interaction found from AT2 to Monocyte\n",
      "0 No significant interaction found from AT2 to Tcell\n",
      "0 No significant interaction found from AT2 to AT1\n",
      "0 No significant interaction found from AT2 to Endothelial\n",
      "0 No significant interaction found from AT2 to Mast\n",
      "0 No significant interaction found from Monocyte to Macrophage\n",
      "0 No significant interaction found from Monocyte to Fibroblast\n",
      "0 No significant interaction found from Monocyte to AT2\n",
      "0 No significant interaction found from Monocyte to Tcell\n",
      "0 No significant interaction found from Monocyte to AT1\n",
      "0 No significant interaction found from Monocyte to Endothelial\n",
      "0 No significant interaction found from Monocyte to Mast\n",
      "0 No significant interaction found from Tcell to Macrophage\n",
      "0 No significant interaction found from Tcell to Fibroblast\n",
      "0 No significant interaction found from Tcell to AT2\n",
      "0 No significant interaction found from Tcell to Monocyte\n",
      "0 No significant interaction found from Tcell to AT1\n",
      "0 No significant interaction found from Tcell to Endothelial\n",
      "0 No significant interaction found from Tcell to Mast\n",
      "0 No significant interaction found from AT1 to Macrophage\n",
      "2 interactions from AT1 to Fibroblast\n",
      "0 No significant interaction found from AT1 to AT2\n",
      "0 No significant interaction found from AT1 to Monocyte\n",
      "0 No significant interaction found from AT1 to Tcell\n",
      "0 No significant interaction found from AT1 to Endothelial\n",
      "0 No significant interaction found from AT1 to Mast\n",
      "0 No significant interaction found from Endothelial to Macrophage\n",
      "0 No significant interaction found from Endothelial to Fibroblast\n",
      "0 No significant interaction found from Endothelial to AT2\n",
      "0 No significant interaction found from Endothelial to Monocyte\n",
      "0 No significant interaction found from Endothelial to Tcell\n",
      "0 No significant interaction found from Endothelial to AT1\n",
      "0 No significant interaction found from Endothelial to Mast\n",
      "0 No significant interaction found from Mast to Macrophage\n",
      "1 interactions from Mast to Fibroblast\n",
      "0 No significant interaction found from Mast to AT2\n",
      "1 interactions from Mast to Monocyte\n",
      "0 No significant interaction found from Mast to Tcell\n",
      "0 No significant interaction found from Mast to AT1\n",
      "1 interactions from Mast to Endothelial\n",
      "[1] \"Total predicted L-R pairs: 6\"\n",
      "No such file as table_dge_Endothelial.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT2.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Macrophage.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Monocyte.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Fibroblast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Tcell.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Mast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT1.txt in the cluster-analysis folder\n",
      "Paracrine signaling: \n",
      "Checking for signaling between cell types\n",
      "0 No significant interaction found from Endothelial to AT2\n",
      "0 No significant interaction found from Endothelial to Macrophage\n",
      "1 interactions from Endothelial to Monocyte\n",
      "0 No significant interaction found from Endothelial to Fibroblast\n",
      "0 No significant interaction found from Endothelial to Tcell\n",
      "0 No significant interaction found from Endothelial to Mast\n",
      "0 No significant interaction found from Endothelial to AT1\n",
      "0 No significant interaction found from AT2 to Endothelial\n",
      "0 No significant interaction found from AT2 to Macrophage\n",
      "0 No significant interaction found from AT2 to Monocyte\n",
      "0 No significant interaction found from AT2 to Fibroblast\n",
      "0 No significant interaction found from AT2 to Tcell\n",
      "0 No significant interaction found from AT2 to Mast\n",
      "0 No significant interaction found from AT2 to AT1\n",
      "0 No significant interaction found from Macrophage to Endothelial\n",
      "0 No significant interaction found from Macrophage to AT2\n",
      "0 No significant interaction found from Macrophage to Monocyte\n",
      "0 No significant interaction found from Macrophage to Fibroblast\n",
      "0 No significant interaction found from Macrophage to Tcell\n",
      "0 No significant interaction found from Macrophage to Mast\n",
      "0 No significant interaction found from Macrophage to AT1\n",
      "0 No significant interaction found from Monocyte to Endothelial\n",
      "0 No significant interaction found from Monocyte to AT2\n",
      "0 No significant interaction found from Monocyte to Macrophage\n",
      "0 No significant interaction found from Monocyte to Fibroblast\n",
      "0 No significant interaction found from Monocyte to Tcell\n",
      "0 No significant interaction found from Monocyte to Mast\n",
      "0 No significant interaction found from Monocyte to AT1\n",
      "0 No significant interaction found from Fibroblast to Endothelial\n",
      "0 No significant interaction found from Fibroblast to AT2\n",
      "0 No significant interaction found from Fibroblast to Macrophage\n",
      "0 No significant interaction found from Fibroblast to Monocyte\n",
      "0 No significant interaction found from Fibroblast to Tcell\n",
      "0 No significant interaction found from Fibroblast to Mast\n",
      "0 No significant interaction found from Fibroblast to AT1\n",
      "0 No significant interaction found from Tcell to Endothelial\n",
      "0 No significant interaction found from Tcell to AT2\n",
      "0 No significant interaction found from Tcell to Macrophage\n",
      "0 No significant interaction found from Tcell to Monocyte\n",
      "0 No significant interaction found from Tcell to Fibroblast\n",
      "0 No significant interaction found from Tcell to Mast\n",
      "0 No significant interaction found from Tcell to AT1\n",
      "0 No significant interaction found from Mast to Endothelial\n",
      "0 No significant interaction found from Mast to AT2\n",
      "1 interactions from Mast to Macrophage\n",
      "1 interactions from Mast to Monocyte\n",
      "0 No significant interaction found from Mast to Fibroblast\n",
      "0 No significant interaction found from Mast to Tcell\n",
      "0 No significant interaction found from Mast to AT1\n",
      "0 No significant interaction found from AT1 to Endothelial\n",
      "0 No significant interaction found from AT1 to AT2\n",
      "0 No significant interaction found from AT1 to Macrophage\n",
      "0 No significant interaction found from AT1 to Monocyte\n",
      "0 No significant interaction found from AT1 to Fibroblast\n",
      "0 No significant interaction found from AT1 to Tcell\n",
      "0 No significant interaction found from AT1 to Mast\n",
      "[1] \"Total predicted L-R pairs: 3\"\n",
      "No such file as table_dge_AT2.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT1.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Endothelial.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Mast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Monocyte.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Tcell.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Macrophage.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Fibroblast.txt in the cluster-analysis folder\n",
      "Paracrine signaling: \n",
      "Checking for signaling between cell types\n",
      "0 No significant interaction found from AT2 to AT1\n",
      "0 No significant interaction found from AT2 to Endothelial\n",
      "0 No significant interaction found from AT2 to Mast\n",
      "0 No significant interaction found from AT2 to Monocyte\n",
      "0 No significant interaction found from AT2 to Tcell\n",
      "0 No significant interaction found from AT2 to Macrophage\n",
      "0 No significant interaction found from AT2 to Fibroblast\n",
      "0 No significant interaction found from AT1 to AT2\n",
      "0 No significant interaction found from AT1 to Endothelial\n",
      "0 No significant interaction found from AT1 to Mast\n",
      "0 No significant interaction found from AT1 to Monocyte\n",
      "0 No significant interaction found from AT1 to Tcell\n",
      "0 No significant interaction found from AT1 to Macrophage\n",
      "3 interactions from AT1 to Fibroblast\n",
      "0 No significant interaction found from Endothelial to AT2\n",
      "0 No significant interaction found from Endothelial to AT1\n",
      "0 No significant interaction found from Endothelial to Mast\n",
      "0 No significant interaction found from Endothelial to Monocyte\n",
      "0 No significant interaction found from Endothelial to Tcell\n",
      "0 No significant interaction found from Endothelial to Macrophage\n",
      "0 No significant interaction found from Endothelial to Fibroblast\n",
      "1 interactions from Mast to AT2\n",
      "1 interactions from Mast to AT1\n",
      "0 No significant interaction found from Mast to Endothelial\n",
      "0 No significant interaction found from Mast to Monocyte\n",
      "0 No significant interaction found from Mast to Tcell\n",
      "0 No significant interaction found from Mast to Macrophage\n",
      "0 No significant interaction found from Mast to Fibroblast\n",
      "0 No significant interaction found from Monocyte to AT2\n",
      "0 No significant interaction found from Monocyte to AT1\n",
      "0 No significant interaction found from Monocyte to Endothelial\n",
      "0 No significant interaction found from Monocyte to Mast\n",
      "1 interactions from Monocyte to Tcell\n",
      "0 No significant interaction found from Monocyte to Macrophage\n",
      "0 No significant interaction found from Monocyte to Fibroblast\n",
      "0 No significant interaction found from Tcell to AT2\n",
      "0 No significant interaction found from Tcell to AT1\n",
      "0 No significant interaction found from Tcell to Endothelial\n",
      "0 No significant interaction found from Tcell to Mast\n",
      "0 No significant interaction found from Tcell to Monocyte\n",
      "0 No significant interaction found from Tcell to Macrophage\n",
      "0 No significant interaction found from Tcell to Fibroblast\n",
      "0 No significant interaction found from Macrophage to AT2\n",
      "0 No significant interaction found from Macrophage to AT1\n",
      "0 No significant interaction found from Macrophage to Endothelial\n",
      "0 No significant interaction found from Macrophage to Mast\n",
      "0 No significant interaction found from Macrophage to Monocyte\n",
      "0 No significant interaction found from Macrophage to Tcell\n",
      "0 No significant interaction found from Macrophage to Fibroblast\n",
      "0 No significant interaction found from Fibroblast to AT2\n",
      "0 No significant interaction found from Fibroblast to AT1\n",
      "0 No significant interaction found from Fibroblast to Endothelial\n",
      "0 No significant interaction found from Fibroblast to Mast\n",
      "1 interactions from Fibroblast to Monocyte\n",
      "3 interactions from Fibroblast to Tcell\n",
      "2 interactions from Fibroblast to Macrophage\n",
      "[1] \"Total predicted L-R pairs: 12\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No such file as table_dge_Macrophage.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Monocyte.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Tcell.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Mast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Fibroblast.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT2.txt in the cluster-analysis folder\n",
      "No such file as table_dge_Endothelial.txt in the cluster-analysis folder\n",
      "No such file as table_dge_AT1.txt in the cluster-analysis folder\n",
      "Paracrine signaling: \n",
      "Checking for signaling between cell types\n",
      "0 No significant interaction found from Macrophage to Monocyte\n",
      "0 No significant interaction found from Macrophage to Tcell\n",
      "0 No significant interaction found from Macrophage to Mast\n",
      "0 No significant interaction found from Macrophage to Fibroblast\n",
      "0 No significant interaction found from Macrophage to AT2\n",
      "0 No significant interaction found from Macrophage to Endothelial\n",
      "0 No significant interaction found from Macrophage to AT1\n",
      "0 No significant interaction found from Monocyte to Macrophage\n",
      "0 No significant interaction found from Monocyte to Tcell\n",
      "0 No significant interaction found from Monocyte to Mast\n",
      "0 No significant interaction found from Monocyte to Fibroblast\n",
      "0 No significant interaction found from Monocyte to AT2\n",
      "0 No significant interaction found from Monocyte to Endothelial\n",
      "0 No significant interaction found from Monocyte to AT1\n",
      "0 No significant interaction found from Tcell to Macrophage\n",
      "0 No significant interaction found from Tcell to Monocyte\n",
      "0 No significant interaction found from Tcell to Mast\n",
      "0 No significant interaction found from Tcell to Fibroblast\n",
      "0 No significant interaction found from Tcell to AT2\n",
      "0 No significant interaction found from Tcell to Endothelial\n",
      "0 No significant interaction found from Tcell to AT1\n",
      "0 No significant interaction found from Mast to Macrophage\n",
      "0 No significant interaction found from Mast to Monocyte\n",
      "0 No significant interaction found from Mast to Tcell\n",
      "0 No significant interaction found from Mast to Fibroblast\n",
      "0 No significant interaction found from Mast to AT2\n",
      "0 No significant interaction found from Mast to Endothelial\n",
      "0 No significant interaction found from Mast to AT1\n",
      "0 No significant interaction found from Fibroblast to Macrophage\n",
      "0 No significant interaction found from Fibroblast to Monocyte\n",
      "0 No significant interaction found from Fibroblast to Tcell\n",
      "0 No significant interaction found from Fibroblast to Mast\n",
      "0 No significant interaction found from Fibroblast to AT2\n",
      "0 No significant interaction found from Fibroblast to Endothelial\n",
      "0 No significant interaction found from Fibroblast to AT1\n",
      "0 No significant interaction found from AT2 to Macrophage\n",
      "0 No significant interaction found from AT2 to Monocyte\n",
      "0 No significant interaction found from AT2 to Tcell\n",
      "0 No significant interaction found from AT2 to Mast\n",
      "0 No significant interaction found from AT2 to Fibroblast\n",
      "0 No significant interaction found from AT2 to Endothelial\n",
      "0 No significant interaction found from AT2 to AT1\n",
      "0 No significant interaction found from Endothelial to Macrophage\n",
      "0 No significant interaction found from Endothelial to Monocyte\n",
      "0 No significant interaction found from Endothelial to Tcell\n",
      "0 No significant interaction found from Endothelial to Mast\n",
      "0 No significant interaction found from Endothelial to Fibroblast\n",
      "0 No significant interaction found from Endothelial to AT2\n",
      "0 No significant interaction found from Endothelial to AT1\n",
      "0 No significant interaction found from AT1 to Macrophage\n",
      "0 No significant interaction found from AT1 to Monocyte\n",
      "0 No significant interaction found from AT1 to Tcell\n",
      "0 No significant interaction found from AT1 to Mast\n",
      "0 No significant interaction found from AT1 to Fibroblast\n",
      "0 No significant interaction found from AT1 to AT2\n",
      "0 No significant interaction found from AT1 to Endothelial\n",
      "[1] \"Total predicted L-R pairs: 0\"\n"
     ]
    }
   ],
   "source": [
    "# SingleCellSignalR\n",
    "SCSRA <- run_SingleCellSignalR(sampleA)\n",
    "SCSRB <- run_SingleCellSignalR(sampleB)\n",
    "SCSRC <- run_SingleCellSignalR(sampleC)\n",
    "SCSRD <- run_SingleCellSignalR(sampleD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f47bfa79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>LRscore</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Fibroblast</td><td>Monocyte   </td><td>CCL2 </td><td>CCR2  </td><td>0.68742187149298 </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1       </td><td>Fibroblast </td><td>PDGFA</td><td>PDGFRB</td><td>0.761042984780823</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT1       </td><td>Fibroblast </td><td>PDGFC</td><td>PDGFRB</td><td>0.689645362246055</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Mast      </td><td>Fibroblast </td><td>TGFB1</td><td>TGFBR3</td><td>0.598945955404346</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Mast      </td><td>Monocyte   </td><td>CCL2 </td><td>CCR2  </td><td>0.548699786843868</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Mast      </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td><td>0.718179341579129</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & LRscore\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Fibroblast & Monocyte    & CCL2  & CCR2   & 0.68742187149298 \\\\\n",
       "\t2 & AT1        & Fibroblast  & PDGFA & PDGFRB & 0.761042984780823\\\\\n",
       "\t3 & AT1        & Fibroblast  & PDGFC & PDGFRB & 0.689645362246055\\\\\n",
       "\t4 & Mast       & Fibroblast  & TGFB1 & TGFBR3 & 0.598945955404346\\\\\n",
       "\t5 & Mast       & Monocyte    & CCL2  & CCR2   & 0.548699786843868\\\\\n",
       "\t6 & Mast       & Endothelial & TGFB1 & TGFBR3 & 0.718179341579129\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | LRscore &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Fibroblast | Monocyte    | CCL2  | CCR2   | 0.68742187149298  |\n",
       "| 2 | AT1        | Fibroblast  | PDGFA | PDGFRB | 0.761042984780823 |\n",
       "| 3 | AT1        | Fibroblast  | PDGFC | PDGFRB | 0.689645362246055 |\n",
       "| 4 | Mast       | Fibroblast  | TGFB1 | TGFBR3 | 0.598945955404346 |\n",
       "| 5 | Mast       | Monocyte    | CCL2  | CCR2   | 0.548699786843868 |\n",
       "| 6 | Mast       | Endothelial | TGFB1 | TGFBR3 | 0.718179341579129 |\n",
       "\n"
      ],
      "text/plain": [
       "  source     target      ligand receptor LRscore          \n",
       "1 Fibroblast Monocyte    CCL2   CCR2     0.68742187149298 \n",
       "2 AT1        Fibroblast  PDGFA  PDGFRB   0.761042984780823\n",
       "3 AT1        Fibroblast  PDGFC  PDGFRB   0.689645362246055\n",
       "4 Mast       Fibroblast  TGFB1  TGFBR3   0.598945955404346\n",
       "5 Mast       Monocyte    CCL2   CCR2     0.548699786843868\n",
       "6 Mast       Endothelial TGFB1  TGFBR3   0.718179341579129"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>LRscore</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Endothelial</td><td>Monocyte  </td><td>CCL2</td><td>CCR2 </td><td>0.595674932115517</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Mast       </td><td>Macrophage</td><td>CSF1</td><td>CSF1R</td><td>0.662648748227722</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Mast       </td><td>Monocyte  </td><td>CSF1</td><td>CSF1R</td><td>0.676553266700939</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & LRscore\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Endothelial & Monocyte   & CCL2 & CCR2  & 0.595674932115517\\\\\n",
       "\t2 & Mast        & Macrophage & CSF1 & CSF1R & 0.662648748227722\\\\\n",
       "\t3 & Mast        & Monocyte   & CSF1 & CSF1R & 0.676553266700939\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | LRscore &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Endothelial | Monocyte   | CCL2 | CCR2  | 0.595674932115517 |\n",
       "| 2 | Mast        | Macrophage | CSF1 | CSF1R | 0.662648748227722 |\n",
       "| 3 | Mast        | Monocyte   | CSF1 | CSF1R | 0.676553266700939 |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target     ligand receptor LRscore          \n",
       "1 Endothelial Monocyte   CCL2   CCR2     0.595674932115517\n",
       "2 Mast        Macrophage CSF1   CSF1R    0.662648748227722\n",
       "3 Mast        Monocyte   CSF1   CSF1R    0.676553266700939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>LRscore</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1     </td><td>Fibroblast</td><td>PDGFA</td><td>PDGFRA</td><td>0.760414319142841</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1     </td><td>Fibroblast</td><td>PDGFA</td><td>PDGFRB</td><td>0.732949670274543</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT1     </td><td>Fibroblast</td><td>PDGFC</td><td>PDGFRA</td><td>0.50380082682914 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Mast    </td><td>AT2       </td><td>TGFB1</td><td>ITGB6 </td><td>0.755698094707322</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Mast    </td><td>AT1       </td><td>TGFB1</td><td>ITGB6 </td><td>0.797994657314319</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Monocyte</td><td>Tcell     </td><td>CCL3 </td><td>CCR4  </td><td>0.582479392671887</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & LRscore\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1      & Fibroblast & PDGFA & PDGFRA & 0.760414319142841\\\\\n",
       "\t2 & AT1      & Fibroblast & PDGFA & PDGFRB & 0.732949670274543\\\\\n",
       "\t3 & AT1      & Fibroblast & PDGFC & PDGFRA & 0.50380082682914 \\\\\n",
       "\t4 & Mast     & AT2        & TGFB1 & ITGB6  & 0.755698094707322\\\\\n",
       "\t5 & Mast     & AT1        & TGFB1 & ITGB6  & 0.797994657314319\\\\\n",
       "\t6 & Monocyte & Tcell      & CCL3  & CCR4   & 0.582479392671887\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | LRscore &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AT1      | Fibroblast | PDGFA | PDGFRA | 0.760414319142841 |\n",
       "| 2 | AT1      | Fibroblast | PDGFA | PDGFRB | 0.732949670274543 |\n",
       "| 3 | AT1      | Fibroblast | PDGFC | PDGFRA | 0.50380082682914  |\n",
       "| 4 | Mast     | AT2        | TGFB1 | ITGB6  | 0.755698094707322 |\n",
       "| 5 | Mast     | AT1        | TGFB1 | ITGB6  | 0.797994657314319 |\n",
       "| 6 | Monocyte | Tcell      | CCL3  | CCR4   | 0.582479392671887 |\n",
       "\n"
      ],
      "text/plain": [
       "  source   target     ligand receptor LRscore          \n",
       "1 AT1      Fibroblast PDGFA  PDGFRA   0.760414319142841\n",
       "2 AT1      Fibroblast PDGFA  PDGFRB   0.732949670274543\n",
       "3 AT1      Fibroblast PDGFC  PDGFRA   0.50380082682914 \n",
       "4 Mast     AT2        TGFB1  ITGB6    0.755698094707322\n",
       "5 Mast     AT1        TGFB1  ITGB6    0.797994657314319\n",
       "6 Monocyte Tcell      CCL3   CCR4     0.582479392671887"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>LRscore</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & LRscore\\\\\n",
       "  & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 5\n",
       "\n",
       "| <!--/--> | source &lt;lgl&gt; | target &lt;lgl&gt; | ligand &lt;lgl&gt; | receptor &lt;lgl&gt; | LRscore &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  source target ligand receptor LRscore\n",
       "1 NA     NA     NA     NA       NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(SCSRA)\n",
    "head(SCSRB)\n",
    "head(SCSRC)\n",
    "head(SCSRD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "cfa13970",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Calculating cluster-wise gene statistics --\n",
      "\n",
      "Scaling node weights per cell type...\n",
      "\n",
      "Building node metadata...\n",
      "\n",
      "Building edge list...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 5419\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Calculating cluster-wise gene statistics --\n",
      "\n",
      "Scaling node weights per cell type...\n",
      "\n",
      "Building node metadata...\n",
      "\n",
      "Building edge list...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 6287\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Calculating cluster-wise gene statistics --\n",
      "\n",
      "Scaling node weights per cell type...\n",
      "\n",
      "Building node metadata...\n",
      "\n",
      "Building edge list...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 5444\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Calculating cluster-wise gene statistics --\n",
      "\n",
      "Scaling node weights per cell type...\n",
      "\n",
      "Building node metadata...\n",
      "\n",
      "Building edge list...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 5950\"\n",
      "[1] \"Choose top L-R pairs: 208\"\n"
     ]
    }
   ],
   "source": [
    "# CCInx\n",
    "CCInxA <- run_CCInx(sampleA, top = 208)\n",
    "CCInxB <- run_CCInx(sampleB, top = 208)\n",
    "CCInxC <- run_CCInx(sampleC, top = 208)\n",
    "CCInxD <- run_CCInx(sampleD, top = 208)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bbbc2588",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>edgeWeight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT2</td><td>Fibroblast</td><td>ITGB6</td><td>ITGB1</td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1</td><td>Fibroblast</td><td>ITGB6</td><td>ITGB1</td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT1</td><td>AT2       </td><td>ITGB1</td><td>ITGB6</td><td>0.987788248680122</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT1</td><td>AT1       </td><td>ITGB1</td><td>ITGB6</td><td>0.987788248680122</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AT1</td><td>AT1       </td><td>ITGB6</td><td>ITGB1</td><td>0.987788248680122</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AT2</td><td>Monocyte  </td><td>ITGB6</td><td>ITGB1</td><td>0.987206853721975</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & edgeWeight\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT2 & Fibroblast & ITGB6 & ITGB1 & 1                \\\\\n",
       "\t2 & AT1 & Fibroblast & ITGB6 & ITGB1 & 1                \\\\\n",
       "\t3 & AT1 & AT2        & ITGB1 & ITGB6 & 0.987788248680122\\\\\n",
       "\t4 & AT1 & AT1        & ITGB1 & ITGB6 & 0.987788248680122\\\\\n",
       "\t5 & AT1 & AT1        & ITGB6 & ITGB1 & 0.987788248680122\\\\\n",
       "\t6 & AT2 & Monocyte   & ITGB6 & ITGB1 & 0.987206853721975\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | edgeWeight &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AT2 | Fibroblast | ITGB6 | ITGB1 | 1                 |\n",
       "| 2 | AT1 | Fibroblast | ITGB6 | ITGB1 | 1                 |\n",
       "| 3 | AT1 | AT2        | ITGB1 | ITGB6 | 0.987788248680122 |\n",
       "| 4 | AT1 | AT1        | ITGB1 | ITGB6 | 0.987788248680122 |\n",
       "| 5 | AT1 | AT1        | ITGB6 | ITGB1 | 0.987788248680122 |\n",
       "| 6 | AT2 | Monocyte   | ITGB6 | ITGB1 | 0.987206853721975 |\n",
       "\n"
      ],
      "text/plain": [
       "  source target     ligand receptor edgeWeight       \n",
       "1 AT2    Fibroblast ITGB6  ITGB1    1                \n",
       "2 AT1    Fibroblast ITGB6  ITGB1    1                \n",
       "3 AT1    AT2        ITGB1  ITGB6    0.987788248680122\n",
       "4 AT1    AT1        ITGB1  ITGB6    0.987788248680122\n",
       "5 AT1    AT1        ITGB6  ITGB1    0.987788248680122\n",
       "6 AT2    Monocyte   ITGB6  ITGB1    0.987206853721975"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>edgeWeight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Endothelial</td><td>Monocyte</td><td>CCL2 </td><td>IL1B  </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT2        </td><td>Monocyte</td><td>ITGB1</td><td>IL1B  </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT2        </td><td>AT2     </td><td>ITGB1</td><td>ITGB6 </td><td>0.995656734127857</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT2        </td><td>AT2     </td><td>ITGB6</td><td>ITGB1 </td><td>0.995656734127857</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Fibroblast </td><td>Monocyte</td><td>CCL2 </td><td>IL1B  </td><td>0.982470893274612</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AT2        </td><td>AT2     </td><td>ITGB1</td><td>TGFBR2</td><td>0.979857454325993</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & edgeWeight\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Endothelial & Monocyte & CCL2  & IL1B   & 1                \\\\\n",
       "\t2 & AT2         & Monocyte & ITGB1 & IL1B   & 1                \\\\\n",
       "\t3 & AT2         & AT2      & ITGB1 & ITGB6  & 0.995656734127857\\\\\n",
       "\t4 & AT2         & AT2      & ITGB6 & ITGB1  & 0.995656734127857\\\\\n",
       "\t5 & Fibroblast  & Monocyte & CCL2  & IL1B   & 0.982470893274612\\\\\n",
       "\t6 & AT2         & AT2      & ITGB1 & TGFBR2 & 0.979857454325993\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | edgeWeight &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Endothelial | Monocyte | CCL2  | IL1B   | 1                 |\n",
       "| 2 | AT2         | Monocyte | ITGB1 | IL1B   | 1                 |\n",
       "| 3 | AT2         | AT2      | ITGB1 | ITGB6  | 0.995656734127857 |\n",
       "| 4 | AT2         | AT2      | ITGB6 | ITGB1  | 0.995656734127857 |\n",
       "| 5 | Fibroblast  | Monocyte | CCL2  | IL1B   | 0.982470893274612 |\n",
       "| 6 | AT2         | AT2      | ITGB1 | TGFBR2 | 0.979857454325993 |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target   ligand receptor edgeWeight       \n",
       "1 Endothelial Monocyte CCL2   IL1B     1                \n",
       "2 AT2         Monocyte ITGB1  IL1B     1                \n",
       "3 AT2         AT2      ITGB1  ITGB6    0.995656734127857\n",
       "4 AT2         AT2      ITGB6  ITGB1    0.995656734127857\n",
       "5 Fibroblast  Monocyte CCL2   IL1B     0.982470893274612\n",
       "6 AT2         AT2      ITGB1  TGFBR2   0.979857454325993"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>edgeWeight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Endothelial</td><td>Mast       </td><td>IL33  </td><td>IL1RL1</td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Endothelial</td><td>Monocyte   </td><td>IL33  </td><td>IL1B  </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT2        </td><td>Endothelial</td><td>ITGB6 </td><td>ITGB1 </td><td>0.988816200487422</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT1        </td><td>Endothelial</td><td>ITGB6 </td><td>ITGB1 </td><td>0.988816200487422</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Endothelial</td><td>Monocyte   </td><td>ITGB1 </td><td>IL1B  </td><td>0.988816200487422</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AT1        </td><td>Endothelial</td><td>TGFBR2</td><td>CTGF  </td><td>0.97781005272738 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & edgeWeight\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Endothelial & Mast        & IL33   & IL1RL1 & 1                \\\\\n",
       "\t2 & Endothelial & Monocyte    & IL33   & IL1B   & 1                \\\\\n",
       "\t3 & AT2         & Endothelial & ITGB6  & ITGB1  & 0.988816200487422\\\\\n",
       "\t4 & AT1         & Endothelial & ITGB6  & ITGB1  & 0.988816200487422\\\\\n",
       "\t5 & Endothelial & Monocyte    & ITGB1  & IL1B   & 0.988816200487422\\\\\n",
       "\t6 & AT1         & Endothelial & TGFBR2 & CTGF   & 0.97781005272738 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | edgeWeight &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Endothelial | Mast        | IL33   | IL1RL1 | 1                 |\n",
       "| 2 | Endothelial | Monocyte    | IL33   | IL1B   | 1                 |\n",
       "| 3 | AT2         | Endothelial | ITGB6  | ITGB1  | 0.988816200487422 |\n",
       "| 4 | AT1         | Endothelial | ITGB6  | ITGB1  | 0.988816200487422 |\n",
       "| 5 | Endothelial | Monocyte    | ITGB1  | IL1B   | 0.988816200487422 |\n",
       "| 6 | AT1         | Endothelial | TGFBR2 | CTGF   | 0.97781005272738  |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target      ligand receptor edgeWeight       \n",
       "1 Endothelial Mast        IL33   IL1RL1   1                \n",
       "2 Endothelial Monocyte    IL33   IL1B     1                \n",
       "3 AT2         Endothelial ITGB6  ITGB1    0.988816200487422\n",
       "4 AT1         Endothelial ITGB6  ITGB1    0.988816200487422\n",
       "5 Endothelial Monocyte    ITGB1  IL1B     0.988816200487422\n",
       "6 AT1         Endothelial TGFBR2 CTGF     0.97781005272738 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>edgeWeight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Mast       </td><td>Monocyte</td><td>IL1RL1</td><td>IL1RAP</td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Endothelial</td><td>Monocyte</td><td>IL1R1 </td><td>IL1RAP</td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Fibroblast </td><td>Tcell   </td><td>CTGF  </td><td>TGFB1 </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT2        </td><td>Tcell   </td><td>ITGB6 </td><td>TGFB1 </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AT1        </td><td>Tcell   </td><td>ITGB6 </td><td>TGFB1 </td><td>1                </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Endothelial</td><td>Tcell   </td><td>TGFBR2</td><td>TGFB1 </td><td>0.998704907678036</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & source & target & ligand & receptor & edgeWeight\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Mast        & Monocyte & IL1RL1 & IL1RAP & 1                \\\\\n",
       "\t2 & Endothelial & Monocyte & IL1R1  & IL1RAP & 1                \\\\\n",
       "\t3 & Fibroblast  & Tcell    & CTGF   & TGFB1  & 1                \\\\\n",
       "\t4 & AT2         & Tcell    & ITGB6  & TGFB1  & 1                \\\\\n",
       "\t5 & AT1         & Tcell    & ITGB6  & TGFB1  & 1                \\\\\n",
       "\t6 & Endothelial & Tcell    & TGFBR2 & TGFB1  & 0.998704907678036\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; | edgeWeight &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Mast        | Monocyte | IL1RL1 | IL1RAP | 1                 |\n",
       "| 2 | Endothelial | Monocyte | IL1R1  | IL1RAP | 1                 |\n",
       "| 3 | Fibroblast  | Tcell    | CTGF   | TGFB1  | 1                 |\n",
       "| 4 | AT2         | Tcell    | ITGB6  | TGFB1  | 1                 |\n",
       "| 5 | AT1         | Tcell    | ITGB6  | TGFB1  | 1                 |\n",
       "| 6 | Endothelial | Tcell    | TGFBR2 | TGFB1  | 0.998704907678036 |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target   ligand receptor edgeWeight       \n",
       "1 Mast        Monocyte IL1RL1 IL1RAP   1                \n",
       "2 Endothelial Monocyte IL1R1  IL1RAP   1                \n",
       "3 Fibroblast  Tcell    CTGF   TGFB1    1                \n",
       "4 AT2         Tcell    ITGB6  TGFB1    1                \n",
       "5 AT1         Tcell    ITGB6  TGFB1    1                \n",
       "6 Endothelial Tcell    TGFBR2 TGFB1    0.998704907678036"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(CCInxA)\n",
    "head(CCInxB)\n",
    "head(CCInxC)\n",
    "head(CCInxD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8e1e7d82",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:6\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:1485\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:24\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:2\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:3\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:20\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:46\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:12\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:44\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:24\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:969\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:27\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:2\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:305\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2236\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:27\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3111\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:10\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:3\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2081\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:46\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:16\"\n",
      "[1] \"XZRec_XZTF Num:4613\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:1190\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:41\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3117\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2424\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:20\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:12\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:41\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:7\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:1\"\n",
      "[1] \"XZRec_XZTF Num:10\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:5892\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:181\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:147\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:44\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:5\"\n",
      "[1] \"XZRec_XZTF Num:3487\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:2293\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:774\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:645\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:210\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3823\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 10413     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:299\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:34\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"Total predicted L-R pairs: 0\"\n"
     ]
    }
   ],
   "source": [
    "# scMLnet\n",
    "scMLnetA <- run_scMLnet(sampleA2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f90c9826",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:3795\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2261\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:47\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:51\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:3\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:51\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:59\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:3\"\n",
      "[1] \"XZRec_XZTF Num:4\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "Error in data.test[gene, cells.1] : incorrect number of dimensions\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:6\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:2\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:55\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:3\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:62\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:62\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2411\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:51\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:52\"\n",
      "[1] \"XZRec_XZTF Num:5705\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:55\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:2\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:91\"\n",
      "[1] \"XZRec_XZTF Num:7149\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:48\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:33\"\n",
      "[1] \"XZRec_XZTF Num:5254\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2933\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3101\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:48\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:57\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:55\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:47\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:23\"\n",
      "[1] \"XZRec_XZTF Num:4522\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:128\"\n",
      "[1] \"XZRec_XZTF Num:7691\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:3\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:122\"\n",
      "[1] \"XZRec_XZTF Num:7916\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:91\"\n",
      "[1] \"XZRec_XZTF Num:7211\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:51\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:40\"\n",
      "[1] \"XZRec_XZTF Num:5449\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:59\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:107\"\n",
      "[1] \"XZRec_XZTF Num:7317\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:62\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:4\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:135\"\n",
      "[1] \"XZRec_XZTF Num:7225\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:57\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:104\"\n",
      "[1] \"XZRec_XZTF Num:7550\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:2553\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2720\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:1548\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:62\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "Error in data.test[gene, cells.1] : incorrect number of dimensions\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:7622\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:62\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:104\"\n",
      "[1] \"XZRec_XZTF Num:7101\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:1153\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:3420\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:55\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:96\"\n",
      "[1] \"XZRec_XZTF Num:6809\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "Error in intI(i, n = x@Dim[1], dn[[1]], give.dn = FALSE) : \n",
      "  invalid character indexing\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:1693\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:1\"\n",
      "[1] \"XZRec_XZTF Num:4\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 20627     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:1405\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:1233\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 9\"\n"
     ]
    }
   ],
   "source": [
    "scMLnetB <- run_scMLnet(sampleB2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "39a2a8ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:12\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:190\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:27\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:342\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:22\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:54\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:75\"\n",
      "[1] \"XZRec_XZTF Num:6671\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:12\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:9\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:9\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:1038\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:41\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3196\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:7\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:22\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:1\"\n",
      "[1] \"XZRec_XZTF Num:3389\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:24\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:7\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:27\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:25\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:186\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:50\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:33\"\n",
      "[1] \"XZRec_XZTF Num:5164\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:31\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:35\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:32\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2438\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:22\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:22\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:779\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:16\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:46\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:24\"\n",
      "[1] \"XZRec_XZTF Num:4083\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:20\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:54\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:41\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:7\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:50\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:46\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:7\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:3225\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:96\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3496\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:24\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:360\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:2113\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:29\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:2829\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:20\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 28778     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:16507\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:152\"\n",
      "[1] \"gene:121\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:15\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2497\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"Total predicted L-R pairs: 0\"\n"
     ]
    }
   ],
   "source": [
    "scMLnetC <- run_scMLnet(sampleC2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ae806b1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:65\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:6\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:6\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:7\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:1015\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:6\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:97\"\n",
      "[1] \"XZRec_XZTF Num:7491\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:57\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:105\"\n",
      "[1] \"XZRec_XZTF Num:7440\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:1\"\n",
      "[1] \"XZRec_XZTF Num:3681\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:52\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:52\"\n",
      "[1] \"XZRec_XZTF Num:5876\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:61\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:133\"\n",
      "[1] \"XZRec_XZTF Num:7615\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:33\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:52\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:63\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:5\"\n",
      "[1] \"TF_Target Num:151\"\n",
      "[1] \"XZRec_XZTF Num:6966\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:47\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:29\"\n",
      "[1] \"XZRec_XZTF Num:4616\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "Error in intI(i, n = x@Dim[1], dn[[1]], give.dn = FALSE) : \n",
      "  invalid character indexing\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:107\"\n",
      "[1] \"XZRec_XZTF Num:7412\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "[1] \"calculate Cor between RecTF\"\n",
      "[1] \"calculate Cor between TFTar\"\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3948\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:65\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:161\"\n",
      "[1] \"XZRec_XZTF Num:7401\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:26\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:60\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:129\"\n",
      "[1] \"XZRec_XZTF Num:7591\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:48\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:36\"\n",
      "[1] \"XZRec_XZTF Num:5554\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:58\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:122\"\n",
      "[1] \"XZRec_XZTF Num:7128\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3404\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:37\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:2416\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:57\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:63\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:9\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:60\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:7\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3348\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:43\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:1\"\n",
      "[1] \"XZRec_XZTF Num:2795\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:47\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:10\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:48\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:7\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:30\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:19\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:2\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:1567\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:35\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:4126\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:14\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:52\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:4\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:8\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:56\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:7\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:58\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:7\"\n",
      "Error in seq.default(1, length(PairList), 1) : \n",
      "  wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:23\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:36\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:3\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Macrophage\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Macrophage\"\n",
      "[1] \"Macrophage:60483\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:42\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:2\"\n",
      "[1] \"XZRec_XZTF Num:3245\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Monocyte\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Monocyte\"\n",
      "[1] \"Monocyte:6793\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:28\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Tcell\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Tcell\"\n",
      "[1] \"Tcell:5088\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:18\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:38\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Mast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Mast\"\n",
      "[1] \"Mast:572\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:17\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:40\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Fibroblast\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Fibroblast\"\n",
      "[1] \"Fibroblast:1443\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:39\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:11\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:3152\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "[1] \"TF common in RecTF and TFTar:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:RecTF and TFTar don't have the same TF!\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"AT2\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in AT2\"\n",
      "[1] \"AT2:496\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:21\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:5\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:0\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "Error in seq.default(1, length(XZRec), 1) : wrong sign in 'by' argument\n",
      "[1] \"check table cell:\"\n",
      "[1] 78430     2\n",
      "[1] \"Rec Cluster:\"\n",
      "[1] \"Endothelial\"\n",
      "[1] \"Lig Cluster:\"\n",
      "[1] \"AT1\"\n",
      "[1] \"p val:\"\n",
      "[1] 0.05\n",
      "[1] \"logfc:\"\n",
      "[1] 0.15\n",
      "[1] \"get High Exp Gene in Endothelial\"\n",
      "[1] \"Endothelial:3379\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:34\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"get High Exp Gene in AT1\"\n",
      "[1] \"AT1:176\"\n",
      "[1] \"gene:119\"\n",
      "[1] \"logfc.threshold:\"\n",
      "[1] 0.15\n",
      "[1] \"do T-test\"\n",
      "[1] \"find high gene num:13\"\n",
      "[1] \"-----------------------\"\n",
      "[1] \"Lig_Rec Num:1\"\n",
      "[1] \"TF_Target Num:0\"\n",
      "[1] \"XZRec_XZTF Num:372\"\n",
      "[1] \"Rec common in LigRec and RecTF:\"\n",
      "Error in RunMLnet(mtx, annfile, RecClu, LigClu, pval, logfc, LigRecLib,  : \n",
      "  error:LigRec and RecTF don't have the same Rec!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (!netList[[i]] %in% NA) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Total predicted L-R pairs: 8\"\n"
     ]
    }
   ],
   "source": [
    "scMLnetD <- run_scMLnet(sampleD2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5eaaf5ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligrec</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & source & target & ligrec\\\\\n",
       "  & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 3\n",
       "\n",
       "| <!--/--> | source &lt;lgl&gt; | target &lt;lgl&gt; | ligrec &lt;lgl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  source target ligrec\n",
       "1 NA     NA     NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligrec</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Monocyte</td><td>AT2       </td><td>IL1A_IL1R1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Monocyte</td><td>AT2       </td><td>IL1B_IL1R1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Monocyte</td><td>Fibroblast</td><td>IL1A_IL1R1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Monocyte</td><td>Fibroblast</td><td>IL1B_IL1R1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Mast    </td><td>Macrophage</td><td>CCL4_CCR5 </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Mast    </td><td>Macrophage</td><td>CCL4_CCR1 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & source & target & ligrec\\\\\n",
       "  & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Monocyte & AT2        & IL1A\\_IL1R1\\\\\n",
       "\t2 & Monocyte & AT2        & IL1B\\_IL1R1\\\\\n",
       "\t3 & Monocyte & Fibroblast & IL1A\\_IL1R1\\\\\n",
       "\t4 & Monocyte & Fibroblast & IL1B\\_IL1R1\\\\\n",
       "\t5 & Mast     & Macrophage & CCL4\\_CCR5 \\\\\n",
       "\t6 & Mast     & Macrophage & CCL4\\_CCR1 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligrec &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | Monocyte | AT2        | IL1A_IL1R1 |\n",
       "| 2 | Monocyte | AT2        | IL1B_IL1R1 |\n",
       "| 3 | Monocyte | Fibroblast | IL1A_IL1R1 |\n",
       "| 4 | Monocyte | Fibroblast | IL1B_IL1R1 |\n",
       "| 5 | Mast     | Macrophage | CCL4_CCR5  |\n",
       "| 6 | Mast     | Macrophage | CCL4_CCR1  |\n",
       "\n"
      ],
      "text/plain": [
       "  source   target     ligrec    \n",
       "1 Monocyte AT2        IL1A_IL1R1\n",
       "2 Monocyte AT2        IL1B_IL1R1\n",
       "3 Monocyte Fibroblast IL1A_IL1R1\n",
       "4 Monocyte Fibroblast IL1B_IL1R1\n",
       "5 Mast     Macrophage CCL4_CCR5 \n",
       "6 Mast     Macrophage CCL4_CCR1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligrec</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & source & target & ligrec\\\\\n",
       "  & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 3\n",
       "\n",
       "| <!--/--> | source &lt;lgl&gt; | target &lt;lgl&gt; | ligrec &lt;lgl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  source target ligrec\n",
       "1 NA     NA     NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligrec</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Monocyte</td><td>Fibroblast </td><td>IL1B_IL1R1  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Monocyte</td><td>Endothelial</td><td>IL1B_IL1R1  </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Tcell   </td><td>Fibroblast </td><td>TGFB1_ITGAV </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Tcell   </td><td>Fibroblast </td><td>TGFB1_TGFBR1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Tcell   </td><td>Fibroblast </td><td>TGFB1_ITGB6 </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Tcell   </td><td>Fibroblast </td><td>TGFB1_TGFBR3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & source & target & ligrec\\\\\n",
       "  & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Monocyte & Fibroblast  & IL1B\\_IL1R1  \\\\\n",
       "\t2 & Monocyte & Endothelial & IL1B\\_IL1R1  \\\\\n",
       "\t3 & Tcell    & Fibroblast  & TGFB1\\_ITGAV \\\\\n",
       "\t4 & Tcell    & Fibroblast  & TGFB1\\_TGFBR1\\\\\n",
       "\t5 & Tcell    & Fibroblast  & TGFB1\\_ITGB6 \\\\\n",
       "\t6 & Tcell    & Fibroblast  & TGFB1\\_TGFBR3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligrec &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | Monocyte | Fibroblast  | IL1B_IL1R1   |\n",
       "| 2 | Monocyte | Endothelial | IL1B_IL1R1   |\n",
       "| 3 | Tcell    | Fibroblast  | TGFB1_ITGAV  |\n",
       "| 4 | Tcell    | Fibroblast  | TGFB1_TGFBR1 |\n",
       "| 5 | Tcell    | Fibroblast  | TGFB1_ITGB6  |\n",
       "| 6 | Tcell    | Fibroblast  | TGFB1_TGFBR3 |\n",
       "\n"
      ],
      "text/plain": [
       "  source   target      ligrec      \n",
       "1 Monocyte Fibroblast  IL1B_IL1R1  \n",
       "2 Monocyte Endothelial IL1B_IL1R1  \n",
       "3 Tcell    Fibroblast  TGFB1_ITGAV \n",
       "4 Tcell    Fibroblast  TGFB1_TGFBR1\n",
       "5 Tcell    Fibroblast  TGFB1_ITGB6 \n",
       "6 Tcell    Fibroblast  TGFB1_TGFBR3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(scMLnetA)\n",
    "head(scMLnetB)\n",
    "head(scMLnetC)\n",
    "head(scMLnetD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "22ecfab8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# CellPhoneDB\n",
    "CellPhoneDB_process(sampleA)\n",
    "CellPhoneDB_process(sampleB)\n",
    "CellPhoneDB_process(sampleC)\n",
    "CellPhoneDB_process(sampleD)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29d9fab0",
   "metadata": {},
   "source": [
    "### Python kernel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6330f175",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cpdb:\r\n",
      "sampleA  sampleB  sampleC  sampleD\r\n",
      "\r\n",
      "cpdb/sampleA:\r\n",
      "barcodes.tsv  features.tsv  matrix.mtx\tmeta.tsv\r\n",
      "\r\n",
      "cpdb/sampleB:\r\n",
      "barcodes.tsv  features.tsv  matrix.mtx\tmeta.tsv\r\n",
      "\r\n",
      "cpdb/sampleC:\r\n",
      "barcodes.tsv  features.tsv  matrix.mtx\tmeta.tsv\r\n",
      "\r\n",
      "cpdb/sampleD:\r\n",
      "barcodes.tsv  features.tsv  matrix.mtx\tmeta.tsv\r\n"
     ]
    }
   ],
   "source": [
    "!ls -AR cpdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "58d3fcd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "!source activate cpdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0451347d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ ][APP][25/04/22-16:52:37][WARNING] Latest local available version is `v2.0.0`, using it\n",
      "[ ][APP][25/04/22-16:52:37][WARNING] User selected downloaded database `v2.0.0` is available, using it\n",
      "[ ][CORE][25/04/22-16:52:37][INFO] Initializing SqlAlchemy CellPhoneDB Core\n",
      "[ ][CORE][25/04/22-16:52:37][INFO] Using custom database at /home/bioinformatics/.cpdb/releases/v2.0.0/cellphone.db\n",
      "[ ][APP][25/04/22-16:52:37][INFO] Launching Method cpdb_statistical_analysis_local_method_launcher\n",
      "[ ][APP][25/04/22-16:52:37][INFO] Launching Method _set_paths\n",
      "[ ][APP][25/04/22-16:52:37][INFO] Launching Method _load_meta_counts\n",
      "[ ][APP][25/04/22-16:52:37][INFO] Launching Method _check_counts_data\n",
      "[ ][CORE][25/04/22-16:52:37][INFO] Launching Method cpdb_statistical_analysis_launcher\n",
      "[ ][CORE][25/04/22-16:52:37][INFO] Launching Method _counts_validations\n",
      "[ ][CORE][25/04/22-16:52:37][INFO] Launching Method get_interactions_genes_complex\n",
      "[ ][CORE][25/04/22-16:52:38][INFO] [Cluster Statistical Analysis] Threshold:0.1 Iterations:1000 Debug-seed:-1 Threads:4 Precision:3\n",
      "[ ][CORE][25/04/22-16:52:39][INFO] Running Real Analysis\n",
      "[ ][CORE][25/04/22-16:52:39][INFO] Running Statistical Analysis\n",
      "[ ][CORE][25/04/22-17:34:43][INFO] Building Pvalues result\n",
      "[ ][CORE][25/04/22-17:34:43][INFO] Building results\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!cellphonedb method statistical_analysis cpdb/sampleA/meta.tsv cpdb/sampleA/ --counts-data hgnc_symbol --output-path cpdb/sampleA/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5d22dd59",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ ][APP][25/04/22-17:34:45][WARNING] Latest local available version is `v2.0.0`, using it\n",
      "[ ][APP][25/04/22-17:34:45][WARNING] User selected downloaded database `v2.0.0` is available, using it\n",
      "[ ][CORE][25/04/22-17:34:45][INFO] Initializing SqlAlchemy CellPhoneDB Core\n",
      "[ ][CORE][25/04/22-17:34:45][INFO] Using custom database at /home/bioinformatics/.cpdb/releases/v2.0.0/cellphone.db\n",
      "[ ][APP][25/04/22-17:34:45][INFO] Launching Method cpdb_statistical_analysis_local_method_launcher\n",
      "[ ][APP][25/04/22-17:34:45][INFO] Launching Method _set_paths\n",
      "[ ][APP][25/04/22-17:34:45][INFO] Launching Method _load_meta_counts\n",
      "[ ][APP][25/04/22-17:34:45][INFO] Launching Method _check_counts_data\n",
      "[ ][CORE][25/04/22-17:34:45][INFO] Launching Method cpdb_statistical_analysis_launcher\n",
      "[ ][CORE][25/04/22-17:34:45][INFO] Launching Method _counts_validations\n",
      "[ ][CORE][25/04/22-17:34:45][INFO] Launching Method get_interactions_genes_complex\n",
      "[ ][CORE][25/04/22-17:34:46][INFO] [Cluster Statistical Analysis] Threshold:0.1 Iterations:1000 Debug-seed:-1 Threads:4 Precision:3\n",
      "[ ][CORE][25/04/22-17:34:46][INFO] Running Real Analysis\n",
      "[ ][CORE][25/04/22-17:34:46][INFO] Running Statistical Analysis\n",
      "[ ][CORE][25/04/22-18:55:31][INFO] Building Pvalues result\n",
      "[ ][CORE][25/04/22-18:55:31][INFO] Building results\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!cellphonedb method statistical_analysis cpdb/sampleB/meta.tsv cpdb/sampleB/ --counts-data hgnc_symbol --output-path cpdb/sampleB/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b4a10b00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ ][APP][25/04/22-18:55:33][WARNING] Latest local available version is `v2.0.0`, using it\n",
      "[ ][APP][25/04/22-18:55:33][WARNING] User selected downloaded database `v2.0.0` is available, using it\n",
      "[ ][CORE][25/04/22-18:55:33][INFO] Initializing SqlAlchemy CellPhoneDB Core\n",
      "[ ][CORE][25/04/22-18:55:33][INFO] Using custom database at /home/bioinformatics/.cpdb/releases/v2.0.0/cellphone.db\n",
      "[ ][APP][25/04/22-18:55:33][INFO] Launching Method cpdb_statistical_analysis_local_method_launcher\n",
      "[ ][APP][25/04/22-18:55:33][INFO] Launching Method _set_paths\n",
      "[ ][APP][25/04/22-18:55:33][INFO] Launching Method _load_meta_counts\n",
      "[ ][APP][25/04/22-18:55:33][INFO] Launching Method _check_counts_data\n",
      "[ ][CORE][25/04/22-18:55:33][INFO] Launching Method cpdb_statistical_analysis_launcher\n",
      "[ ][CORE][25/04/22-18:55:33][INFO] Launching Method _counts_validations\n",
      "[ ][CORE][25/04/22-18:55:34][INFO] Launching Method get_interactions_genes_complex\n",
      "[ ][CORE][25/04/22-18:55:34][INFO] [Cluster Statistical Analysis] Threshold:0.1 Iterations:1000 Debug-seed:-1 Threads:4 Precision:3\n",
      "[ ][CORE][25/04/22-18:55:35][INFO] Running Real Analysis\n",
      "[ ][CORE][25/04/22-18:55:35][INFO] Running Statistical Analysis\n",
      "[ ][CORE][25/04/22-20:48:21][INFO] Building Pvalues result\n",
      "[ ][CORE][25/04/22-20:48:21][INFO] Building results\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!cellphonedb method statistical_analysis cpdb/sampleC/meta.tsv cpdb/sampleC/ --counts-data hgnc_symbol --output-path cpdb/sampleC/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "63dad869",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ ][APP][25/04/22-20:48:22][WARNING] Latest local available version is `v2.0.0`, using it\n",
      "[ ][APP][25/04/22-20:48:22][WARNING] User selected downloaded database `v2.0.0` is available, using it\n",
      "[ ][CORE][25/04/22-20:48:22][INFO] Initializing SqlAlchemy CellPhoneDB Core\n",
      "[ ][CORE][25/04/22-20:48:22][INFO] Using custom database at /home/bioinformatics/.cpdb/releases/v2.0.0/cellphone.db\n",
      "[ ][APP][25/04/22-20:48:22][INFO] Launching Method cpdb_statistical_analysis_local_method_launcher\n",
      "[ ][APP][25/04/22-20:48:22][INFO] Launching Method _set_paths\n",
      "[ ][APP][25/04/22-20:48:22][INFO] Launching Method _load_meta_counts\n",
      "[ ][APP][25/04/22-20:48:23][INFO] Launching Method _check_counts_data\n",
      "[ ][CORE][25/04/22-20:48:23][INFO] Launching Method cpdb_statistical_analysis_launcher\n",
      "[ ][CORE][25/04/22-20:48:23][INFO] Launching Method _counts_validations\n",
      "[ ][CORE][25/04/22-20:48:23][INFO] Launching Method get_interactions_genes_complex\n",
      "[ ][CORE][25/04/22-20:48:23][INFO] [Cluster Statistical Analysis] Threshold:0.1 Iterations:1000 Debug-seed:-1 Threads:4 Precision:3\n",
      "[ ][CORE][25/04/22-20:48:24][INFO] Running Real Analysis\n",
      "[ ][CORE][25/04/22-20:48:24][INFO] Running Statistical Analysis\n",
      "[ ][CORE][26/04/22-02:01:40][INFO] Building Pvalues result\n",
      "[ ][CORE][26/04/22-02:01:40][INFO] Building results\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!cellphonedb method statistical_analysis cpdb/sampleD/meta.tsv cpdb/sampleD/ --counts-data hgnc_symbol --output-path cpdb/sampleD/out"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "98ad4c93",
   "metadata": {},
   "source": [
    "### R kernel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9ee4144f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cellA</th><th scope=col>cellB</th><th scope=col>geneA</th><th scope=col>geneB</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1        </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1        </td><td>Fibroblast </td><td>PDGFA</td><td>PDGFRA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Endothelial</td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>PDGFRA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>PDGFRB</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>LRP1  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & cellA & cellB & geneA & geneB\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1         & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t2 & AT1         & Fibroblast  & PDGFA & PDGFRA\\\\\n",
       "\t3 & Endothelial & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t4 & Endothelial & Fibroblast  & PDGFB & PDGFRA\\\\\n",
       "\t5 & Endothelial & Fibroblast  & PDGFB & PDGFRB\\\\\n",
       "\t6 & Endothelial & Fibroblast  & PDGFB & LRP1  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | cellA &lt;chr&gt; | cellB &lt;chr&gt; | geneA &lt;chr&gt; | geneB &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1         | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 2 | AT1         | Fibroblast  | PDGFA | PDGFRA |\n",
       "| 3 | Endothelial | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 4 | Endothelial | Fibroblast  | PDGFB | PDGFRA |\n",
       "| 5 | Endothelial | Fibroblast  | PDGFB | PDGFRB |\n",
       "| 6 | Endothelial | Fibroblast  | PDGFB | LRP1   |\n",
       "\n"
      ],
      "text/plain": [
       "  cellA       cellB       geneA geneB \n",
       "1 AT1         Endothelial TGFB1 TGFBR3\n",
       "2 AT1         Fibroblast  PDGFA PDGFRA\n",
       "3 Endothelial Endothelial TGFB1 TGFBR3\n",
       "4 Endothelial Fibroblast  PDGFB PDGFRA\n",
       "5 Endothelial Fibroblast  PDGFB PDGFRB\n",
       "6 Endothelial Fibroblast  PDGFB LRP1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CellPhoneDBA <- run_CellPhoneDB_output(\"sampleA\")\n",
    "head(CellPhoneDBA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c5372529",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cellA</th><th scope=col>cellB</th><th scope=col>geneA</th><th scope=col>geneB</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT2        </td><td>Endothelial</td><td>TGFB1 </td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT2        </td><td>Fibroblast </td><td>PDGFA </td><td>PDGFRA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Endothelial</td><td>Endothelial</td><td>TGFB1 </td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Fibroblast </td><td>AT2        </td><td>IGF1  </td><td>IGF1R </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Fibroblast </td><td>AT2        </td><td>PDGFRA</td><td>PDGFC </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Fibroblast </td><td>Endothelial</td><td>TGFB1 </td><td>TGFBR3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & cellA & cellB & geneA & geneB\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT2         & Endothelial & TGFB1  & TGFBR3\\\\\n",
       "\t2 & AT2         & Fibroblast  & PDGFA  & PDGFRA\\\\\n",
       "\t3 & Endothelial & Endothelial & TGFB1  & TGFBR3\\\\\n",
       "\t4 & Fibroblast  & AT2         & IGF1   & IGF1R \\\\\n",
       "\t5 & Fibroblast  & AT2         & PDGFRA & PDGFC \\\\\n",
       "\t6 & Fibroblast  & Endothelial & TGFB1  & TGFBR3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | cellA &lt;chr&gt; | cellB &lt;chr&gt; | geneA &lt;chr&gt; | geneB &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT2         | Endothelial | TGFB1  | TGFBR3 |\n",
       "| 2 | AT2         | Fibroblast  | PDGFA  | PDGFRA |\n",
       "| 3 | Endothelial | Endothelial | TGFB1  | TGFBR3 |\n",
       "| 4 | Fibroblast  | AT2         | IGF1   | IGF1R  |\n",
       "| 5 | Fibroblast  | AT2         | PDGFRA | PDGFC  |\n",
       "| 6 | Fibroblast  | Endothelial | TGFB1  | TGFBR3 |\n",
       "\n"
      ],
      "text/plain": [
       "  cellA       cellB       geneA  geneB \n",
       "1 AT2         Endothelial TGFB1  TGFBR3\n",
       "2 AT2         Fibroblast  PDGFA  PDGFRA\n",
       "3 Endothelial Endothelial TGFB1  TGFBR3\n",
       "4 Fibroblast  AT2         IGF1   IGF1R \n",
       "5 Fibroblast  AT2         PDGFRA PDGFC \n",
       "6 Fibroblast  Endothelial TGFB1  TGFBR3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CellPhoneDBB <- run_CellPhoneDB_output(\"sampleB\")\n",
    "head(CellPhoneDBB)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "905125af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cellA</th><th scope=col>cellB</th><th scope=col>geneA</th><th scope=col>geneB</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1        </td><td>Fibroblast </td><td>PDGFA</td><td>PDGFRA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT2        </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Endothelial</td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>PDGFRA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>PDGFRB</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>LRP1  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & cellA & cellB & geneA & geneB\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1         & Fibroblast  & PDGFA & PDGFRA\\\\\n",
       "\t2 & AT2         & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t3 & Endothelial & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t4 & Endothelial & Fibroblast  & PDGFB & PDGFRA\\\\\n",
       "\t5 & Endothelial & Fibroblast  & PDGFB & PDGFRB\\\\\n",
       "\t6 & Endothelial & Fibroblast  & PDGFB & LRP1  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | cellA &lt;chr&gt; | cellB &lt;chr&gt; | geneA &lt;chr&gt; | geneB &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1         | Fibroblast  | PDGFA | PDGFRA |\n",
       "| 2 | AT2         | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 3 | Endothelial | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 4 | Endothelial | Fibroblast  | PDGFB | PDGFRA |\n",
       "| 5 | Endothelial | Fibroblast  | PDGFB | PDGFRB |\n",
       "| 6 | Endothelial | Fibroblast  | PDGFB | LRP1   |\n",
       "\n"
      ],
      "text/plain": [
       "  cellA       cellB       geneA geneB \n",
       "1 AT1         Fibroblast  PDGFA PDGFRA\n",
       "2 AT2         Endothelial TGFB1 TGFBR3\n",
       "3 Endothelial Endothelial TGFB1 TGFBR3\n",
       "4 Endothelial Fibroblast  PDGFB PDGFRA\n",
       "5 Endothelial Fibroblast  PDGFB PDGFRB\n",
       "6 Endothelial Fibroblast  PDGFB LRP1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CellPhoneDBC <- run_CellPhoneDB_output(\"sampleC\")\n",
    "head(CellPhoneDBC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "591b55f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cellA</th><th scope=col>cellB</th><th scope=col>geneA</th><th scope=col>geneB</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1        </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1        </td><td>Fibroblast </td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AT2        </td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AT2        </td><td>Fibroblast </td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Endothelial</td><td>Endothelial</td><td>TGFB1</td><td>TGFBR3</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Endothelial</td><td>Fibroblast </td><td>PDGFB</td><td>PDGFRB</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & cellA & cellB & geneA & geneB\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1         & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t2 & AT1         & Fibroblast  & TGFB1 & TGFBR3\\\\\n",
       "\t3 & AT2         & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t4 & AT2         & Fibroblast  & TGFB1 & TGFBR3\\\\\n",
       "\t5 & Endothelial & Endothelial & TGFB1 & TGFBR3\\\\\n",
       "\t6 & Endothelial & Fibroblast  & PDGFB & PDGFRB\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | cellA &lt;chr&gt; | cellB &lt;chr&gt; | geneA &lt;chr&gt; | geneB &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1         | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 2 | AT1         | Fibroblast  | TGFB1 | TGFBR3 |\n",
       "| 3 | AT2         | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 4 | AT2         | Fibroblast  | TGFB1 | TGFBR3 |\n",
       "| 5 | Endothelial | Endothelial | TGFB1 | TGFBR3 |\n",
       "| 6 | Endothelial | Fibroblast  | PDGFB | PDGFRB |\n",
       "\n"
      ],
      "text/plain": [
       "  cellA       cellB       geneA geneB \n",
       "1 AT1         Endothelial TGFB1 TGFBR3\n",
       "2 AT1         Fibroblast  TGFB1 TGFBR3\n",
       "3 AT2         Endothelial TGFB1 TGFBR3\n",
       "4 AT2         Fibroblast  TGFB1 TGFBR3\n",
       "5 Endothelial Endothelial TGFB1 TGFBR3\n",
       "6 Endothelial Fibroblast  PDGFB PDGFRB"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CellPhoneDBD <- run_CellPhoneDB_output(\"sampleD\")\n",
    "head(CellPhoneDBD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "18686be6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# NATMI\n",
    "NATMI_process(sampleA)\n",
    "NATMI_process(sampleB)\n",
    "NATMI_process(sampleC)\n",
    "NATMI_process(sampleD)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33d3db65",
   "metadata": {},
   "source": [
    "### Python kernel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a90f3405",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "natmi:\r\n",
      "sampleA  sampleB  sampleC  sampleD\r\n",
      "\r\n",
      "natmi/sampleA:\r\n",
      "ann.csv  em.csv\r\n",
      "\r\n",
      "natmi/sampleB:\r\n",
      "ann.csv  em.csv\r\n",
      "\r\n",
      "natmi/sampleC:\r\n",
      "ann.csv  em.csv\r\n",
      "\r\n",
      "natmi/sampleD:\r\n",
      "ann.csv  em.csv\r\n"
     ]
    }
   ],
   "source": [
    "!ls -AR natmi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "927d1b95",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "Species to obtain abundance data: human\n",
      "The expression matrix file: /opt/NATMI/workdir/em.csv\n",
      "The metafile: /opt/NATMI/workdir/ann.csv\n",
      "The ligand-receptor interaction database: lrc2p.csv\n",
      "Species to obtain ligand-receptor interactions: human\n",
      "The number of cores to use: 1\n",
      "The folder to save all results: /opt/NATMI/workdir/out\n",
      "===================================================\n",
      "#### the folder \"/opt/NATMI/workdir/out\" has been created to save the analysis results\n",
      "#### cluster the expression matrix\n",
      "#### generate cell-to-cell interactions\n",
      "#### extract signaling factors\n",
      "#### construct signaling interactions\n",
      "#### collected 1512 LR-mediated edges\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleA/:/opt/NATMI/workdir asrhou/natmi python ExtractEdges.py --emFile /opt/NATMI/workdir/em.csv --annFile /opt/NATMI/workdir/ann.csv --interDB lrc2p --out /opt/NATMI/workdir/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "73bd44b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "The source dataset folder: /opt/NATMI/workdir/out\n",
      "The ligand-receptor interaction database: lrc2p\n",
      "The weight type of cell-to-cell signaling: mean\n",
      "The detection threshold for interactions to draw is: 0.2\n",
      "The specicificity threshold for interactions to draw is: 0\n",
      "The expression threshold for interactions to draw is: 0\n",
      "The network layout: kk\n",
      "The font size for cluster labels: 8\n",
      "The edge width: 0\n",
      "The maximum radius of the clusters: 0\n",
      "The relative distance between clusters: 1\n",
      "The plot width: 12\n",
      "The plot height: 10\n",
      "The plot format: pdf\n",
      "===================================================\n",
      "#### start to construct the cell-to-cell communication network\n",
      "#### 51 edges are loaded\n",
      "#### plotting the weighted directed cell-to-cell communication network\n",
      "#### the folder \"/opt/NATMI/workdir/out/Network_exp_0_spe_0_det_0.2_top_0_signal_lrc2p_weight_mean\" has been created to save the analysis results\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleA/:/opt/NATMI/workdir asrhou/natmi python VisInteractions.py --sourceFolder /opt/NATMI/workdir/out --interDB lrc2p --weightType mean --detectionThreshold 0.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "447f7ebf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "Species to obtain abundance data: human\n",
      "The expression matrix file: /opt/NATMI/workdir/em.csv\n",
      "The metafile: /opt/NATMI/workdir/ann.csv\n",
      "The ligand-receptor interaction database: lrc2p.csv\n",
      "Species to obtain ligand-receptor interactions: human\n",
      "The number of cores to use: 1\n",
      "The folder to save all results: /opt/NATMI/workdir/out\n",
      "===================================================\n",
      "#### the folder \"/opt/NATMI/workdir/out\" has been created to save the analysis results\n",
      "#### cluster the expression matrix\n",
      "#### generate cell-to-cell interactions\n",
      "#### extract signaling factors\n",
      "#### construct signaling interactions\n",
      "#### collected 1856 LR-mediated edges\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleB/:/opt/NATMI/workdir asrhou/natmi python ExtractEdges.py --emFile /opt/NATMI/workdir/em.csv --annFile /opt/NATMI/workdir/ann.csv --interDB lrc2p --out /opt/NATMI/workdir/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "341220ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "The source dataset folder: /opt/NATMI/workdir/out\n",
      "The ligand-receptor interaction database: lrc2p\n",
      "The weight type of cell-to-cell signaling: mean\n",
      "The detection threshold for interactions to draw is: 0.2\n",
      "The specicificity threshold for interactions to draw is: 0\n",
      "The expression threshold for interactions to draw is: 0\n",
      "The network layout: kk\n",
      "The font size for cluster labels: 8\n",
      "The edge width: 0\n",
      "The maximum radius of the clusters: 0\n",
      "The relative distance between clusters: 1\n",
      "The plot width: 12\n",
      "The plot height: 10\n",
      "The plot format: pdf\n",
      "===================================================\n",
      "#### start to construct the cell-to-cell communication network\n",
      "#### 85 edges are loaded\n",
      "#### plotting the weighted directed cell-to-cell communication network\n",
      "#### the folder \"/opt/NATMI/workdir/out/Network_exp_0_spe_0_det_0.2_top_0_signal_lrc2p_weight_mean\" has been created to save the analysis results\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleB/:/opt/NATMI/workdir asrhou/natmi python VisInteractions.py --sourceFolder /opt/NATMI/workdir/out --interDB lrc2p --weightType mean --detectionThreshold 0.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95df6342",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "Species to obtain abundance data: human\n",
      "The expression matrix file: /opt/NATMI/workdir/em.csv\n",
      "The metafile: /opt/NATMI/workdir/ann.csv\n",
      "The ligand-receptor interaction database: lrc2p.csv\n",
      "Species to obtain ligand-receptor interactions: human\n",
      "The number of cores to use: 1\n",
      "The folder to save all results: /opt/NATMI/workdir/out\n",
      "===================================================\n",
      "#### the folder \"/opt/NATMI/workdir/out\" has been created to save the analysis results\n",
      "#### cluster the expression matrix\n",
      "#### generate cell-to-cell interactions\n",
      "#### extract signaling factors\n",
      "#### construct signaling interactions\n",
      "#### collected 1521 LR-mediated edges\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleC/:/opt/NATMI/workdir asrhou/natmi python ExtractEdges.py --emFile /opt/NATMI/workdir/em.csv --annFile /opt/NATMI/workdir/ann.csv --interDB lrc2p --out /opt/NATMI/workdir/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f591e049",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "The source dataset folder: /opt/NATMI/workdir/out\n",
      "The ligand-receptor interaction database: lrc2p\n",
      "The weight type of cell-to-cell signaling: mean\n",
      "The detection threshold for interactions to draw is: 0.2\n",
      "The specicificity threshold for interactions to draw is: 0\n",
      "The expression threshold for interactions to draw is: 0\n",
      "The network layout: kk\n",
      "The font size for cluster labels: 8\n",
      "The edge width: 0\n",
      "The maximum radius of the clusters: 0\n",
      "The relative distance between clusters: 1\n",
      "The plot width: 12\n",
      "The plot height: 10\n",
      "The plot format: pdf\n",
      "===================================================\n",
      "#### start to construct the cell-to-cell communication network\n",
      "#### 132 edges are loaded\n",
      "#### plotting the weighted directed cell-to-cell communication network\n",
      "#### the folder \"/opt/NATMI/workdir/out/Network_exp_0_spe_0_det_0.2_top_0_signal_lrc2p_weight_mean\" has been created to save the analysis results\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleC/:/opt/NATMI/workdir asrhou/natmi python VisInteractions.py --sourceFolder /opt/NATMI/workdir/out --interDB lrc2p --weightType mean --detectionThreshold 0.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "460ed22e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "Species to obtain abundance data: human\n",
      "The expression matrix file: /opt/NATMI/workdir/em.csv\n",
      "The metafile: /opt/NATMI/workdir/ann.csv\n",
      "The ligand-receptor interaction database: lrc2p.csv\n",
      "Species to obtain ligand-receptor interactions: human\n",
      "The number of cores to use: 1\n",
      "The folder to save all results: /opt/NATMI/workdir/out\n",
      "===================================================\n",
      "#### the folder \"/opt/NATMI/workdir/out\" has been created to save the analysis results\n",
      "#### cluster the expression matrix\n",
      "#### generate cell-to-cell interactions\n",
      "#### extract signaling factors\n",
      "#### construct signaling interactions\n",
      "#### collected 1674 LR-mediated edges\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleD/:/opt/NATMI/workdir asrhou/natmi python ExtractEdges.py --emFile /opt/NATMI/workdir/em.csv --annFile /opt/NATMI/workdir/ann.csv --interDB lrc2p --out /opt/NATMI/workdir/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a2b77533",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===================================================\n",
      "Input data:\n",
      "The source dataset folder: /opt/NATMI/workdir/out\n",
      "The ligand-receptor interaction database: lrc2p\n",
      "The weight type of cell-to-cell signaling: mean\n",
      "The detection threshold for interactions to draw is: 0.2\n",
      "The specicificity threshold for interactions to draw is: 0\n",
      "The expression threshold for interactions to draw is: 0\n",
      "The network layout: kk\n",
      "The font size for cluster labels: 8\n",
      "The edge width: 0\n",
      "The maximum radius of the clusters: 0\n",
      "The relative distance between clusters: 1\n",
      "The plot width: 12\n",
      "The plot height: 10\n",
      "The plot format: pdf\n",
      "===================================================\n",
      "#### start to construct the cell-to-cell communication network\n",
      "#### 172 edges are loaded\n",
      "#### plotting the weighted directed cell-to-cell communication network\n",
      "#### the folder \"/opt/NATMI/workdir/out/Network_exp_0_spe_0_det_0.2_top_0_signal_lrc2p_weight_mean\" has been created to save the analysis results\n",
      "#### DONE\n"
     ]
    }
   ],
   "source": [
    "!docker run --rm -it --name natmi -v /home/bioinformatics/Documents/Benchmark/Python/natmi/sampleD/:/opt/NATMI/workdir asrhou/natmi python VisInteractions.py --sourceFolder /opt/NATMI/workdir/out --interDB lrc2p --weightType mean --detectionThreshold 0.2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7d9ce7b",
   "metadata": {},
   "source": [
    "### R kernel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f1089005",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1       </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AT1       </td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Fibroblast</td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Fibroblast</td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Macrophage</td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Macrophage</td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & source & target & ligand & receptor\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1        & Macrophage & CCL2 & CCR1\\\\\n",
       "\t2 & AT1        & Monocyte   & CCL2 & CCR1\\\\\n",
       "\t3 & Fibroblast & Macrophage & CCL2 & CCR1\\\\\n",
       "\t4 & Fibroblast & Monocyte   & CCL2 & CCR1\\\\\n",
       "\t5 & Macrophage & Macrophage & CCL2 & CCR1\\\\\n",
       "\t6 & Macrophage & Monocyte   & CCL2 & CCR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1        | Macrophage | CCL2 | CCR1 |\n",
       "| 2 | AT1        | Monocyte   | CCL2 | CCR1 |\n",
       "| 3 | Fibroblast | Macrophage | CCL2 | CCR1 |\n",
       "| 4 | Fibroblast | Monocyte   | CCL2 | CCR1 |\n",
       "| 5 | Macrophage | Macrophage | CCL2 | CCR1 |\n",
       "| 6 | Macrophage | Monocyte   | CCL2 | CCR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  source     target     ligand receptor\n",
       "1 AT1        Macrophage CCL2   CCR1    \n",
       "2 AT1        Monocyte   CCL2   CCR1    \n",
       "3 Fibroblast Macrophage CCL2   CCR1    \n",
       "4 Fibroblast Monocyte   CCL2   CCR1    \n",
       "5 Macrophage Macrophage CCL2   CCR1    \n",
       "6 Macrophage Monocyte   CCL2   CCR1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NATMIA <- run_NATMI_output(\"sampleA\")\n",
    "head(NATMIA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f2bf1e19",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Endothelial</td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Fibroblast </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Macrophage </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Monocyte   </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Macrophage </td><td>Macrophage</td><td>CCL3</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Monocyte   </td><td>Macrophage</td><td>CCL3</td><td>CCR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & source & target & ligand & receptor\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Endothelial & Macrophage & CCL2 & CCR1\\\\\n",
       "\t2 & Fibroblast  & Macrophage & CCL2 & CCR1\\\\\n",
       "\t3 & Macrophage  & Macrophage & CCL2 & CCR1\\\\\n",
       "\t4 & Monocyte    & Macrophage & CCL2 & CCR1\\\\\n",
       "\t5 & Macrophage  & Macrophage & CCL3 & CCR1\\\\\n",
       "\t6 & Monocyte    & Macrophage & CCL3 & CCR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | Endothelial | Macrophage | CCL2 | CCR1 |\n",
       "| 2 | Fibroblast  | Macrophage | CCL2 | CCR1 |\n",
       "| 3 | Macrophage  | Macrophage | CCL2 | CCR1 |\n",
       "| 4 | Monocyte    | Macrophage | CCL2 | CCR1 |\n",
       "| 5 | Macrophage  | Macrophage | CCL3 | CCR1 |\n",
       "| 6 | Monocyte    | Macrophage | CCL3 | CCR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target     ligand receptor\n",
       "1 Endothelial Macrophage CCL2   CCR1    \n",
       "2 Fibroblast  Macrophage CCL2   CCR1    \n",
       "3 Macrophage  Macrophage CCL2   CCR1    \n",
       "4 Monocyte    Macrophage CCL2   CCR1    \n",
       "5 Macrophage  Macrophage CCL3   CCR1    \n",
       "6 Monocyte    Macrophage CCL3   CCR1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NATMIB <- run_NATMI_output(\"sampleB\")\n",
    "head(NATMIB)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "82d84484",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Endothelial</td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Endothelial</td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Fibroblast </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Fibroblast </td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Macrophage </td><td>Macrophage</td><td>CCL2</td><td>CCR1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Macrophage </td><td>Monocyte  </td><td>CCL2</td><td>CCR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & source & target & ligand & receptor\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & Endothelial & Macrophage & CCL2 & CCR1\\\\\n",
       "\t2 & Endothelial & Monocyte   & CCL2 & CCR1\\\\\n",
       "\t3 & Fibroblast  & Macrophage & CCL2 & CCR1\\\\\n",
       "\t4 & Fibroblast  & Monocyte   & CCL2 & CCR1\\\\\n",
       "\t5 & Macrophage  & Macrophage & CCL2 & CCR1\\\\\n",
       "\t6 & Macrophage  & Monocyte   & CCL2 & CCR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | Endothelial | Macrophage | CCL2 | CCR1 |\n",
       "| 2 | Endothelial | Monocyte   | CCL2 | CCR1 |\n",
       "| 3 | Fibroblast  | Macrophage | CCL2 | CCR1 |\n",
       "| 4 | Fibroblast  | Monocyte   | CCL2 | CCR1 |\n",
       "| 5 | Macrophage  | Macrophage | CCL2 | CCR1 |\n",
       "| 6 | Macrophage  | Monocyte   | CCL2 | CCR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target     ligand receptor\n",
       "1 Endothelial Macrophage CCL2   CCR1    \n",
       "2 Endothelial Monocyte   CCL2   CCR1    \n",
       "3 Fibroblast  Macrophage CCL2   CCR1    \n",
       "4 Fibroblast  Monocyte   CCL2   CCR1    \n",
       "5 Macrophage  Macrophage CCL2   CCR1    \n",
       "6 Macrophage  Monocyte   CCL2   CCR1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NATMIC <- run_NATMI_output(\"sampleC\")\n",
    "head(NATMIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "b7985d7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>source</th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>receptor</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AT1        </td><td>Macrophage</td><td>CCL2</td><td>CCR1 </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Endothelial</td><td>Macrophage</td><td>CCL2</td><td>CCR1 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Fibroblast </td><td>Macrophage</td><td>CCL2</td><td>CCR1 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Macrophage </td><td>Macrophage</td><td>CCL2</td><td>CCR1 </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Tcell      </td><td>Macrophage</td><td>CCL4</td><td>CCR1 </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Fibroblast </td><td>Macrophage</td><td>CSF1</td><td>CSF1R</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & source & target & ligand & receptor\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AT1         & Macrophage & CCL2 & CCR1 \\\\\n",
       "\t2 & Endothelial & Macrophage & CCL2 & CCR1 \\\\\n",
       "\t3 & Fibroblast  & Macrophage & CCL2 & CCR1 \\\\\n",
       "\t4 & Macrophage  & Macrophage & CCL2 & CCR1 \\\\\n",
       "\t5 & Tcell       & Macrophage & CCL4 & CCR1 \\\\\n",
       "\t6 & Fibroblast  & Macrophage & CSF1 & CSF1R\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | source &lt;chr&gt; | target &lt;chr&gt; | ligand &lt;chr&gt; | receptor &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AT1         | Macrophage | CCL2 | CCR1  |\n",
       "| 2 | Endothelial | Macrophage | CCL2 | CCR1  |\n",
       "| 3 | Fibroblast  | Macrophage | CCL2 | CCR1  |\n",
       "| 4 | Macrophage  | Macrophage | CCL2 | CCR1  |\n",
       "| 5 | Tcell       | Macrophage | CCL4 | CCR1  |\n",
       "| 6 | Fibroblast  | Macrophage | CSF1 | CSF1R |\n",
       "\n"
      ],
      "text/plain": [
       "  source      target     ligand receptor\n",
       "1 AT1         Macrophage CCL2   CCR1    \n",
       "2 Endothelial Macrophage CCL2   CCR1    \n",
       "3 Fibroblast  Macrophage CCL2   CCR1    \n",
       "4 Macrophage  Macrophage CCL2   CCR1    \n",
       "5 Tcell       Macrophage CCL4   CCR1    \n",
       "6 Fibroblast  Macrophage CSF1   CSF1R   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NATMID <- run_NATMI_output(\"sampleD\")\n",
    "head(NATMID)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95ce5b93",
   "metadata": {},
   "source": [
    "## 3. SSP Calculation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4a4a4786",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load functions\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/CellChat_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/iTalk_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/SingleCellSignalR_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/CCInx_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/scMLnet_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/CellPhoneDB_SSP_calculation.R\")\n",
    "source(\"https://raw.githubusercontent.com/mora-lab/cell-cell-interactions/main/benchmark-workflow/R/NATMI_SSP_calculation.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "17fa49ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "CellChat_SSPA <- run_CellChat_SSP(CellChatA,sampleA,gold)\n",
    "iTalk_SSPA <- run_iTalk_SSP(iTalkA,sampleA,gold)\n",
    "SingleCellSignalR_SSPA <- run_SingleCellSignalR_SSP(SCSRA,sampleA,gold)\n",
    "CCInx_SSPA <- run_CCInx_SSP(CCInxA,sampleA,gold)\n",
    "scMLnet_SSPA <- run_scMLnet_SSP(scMLnetA,sampleA,gold)\n",
    "CellPhoneDB_SSPA <- run_CellPhoneDB_SSP(CellPhoneDBA,sampleA,gold)\n",
    "NATMI_SSPA <- run_NATMI_SSP(NATMIA,sampleA,gold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "292d750a",
   "metadata": {},
   "outputs": [],
   "source": [
    "CellChat_SSPB <- run_CellChat_SSP(CellChatB,sampleB,gold)\n",
    "iTalk_SSPB <- run_iTalk_SSP(iTalkB,sampleB,gold)\n",
    "SingleCellSignalR_SSPB <- run_SingleCellSignalR_SSP(SCSRB,sampleB,gold)\n",
    "CCInx_SSPB <- run_CCInx_SSP(CCInxB,sampleB,gold)\n",
    "scMLnet_SSPB <- run_scMLnet_SSP(scMLnetB,sampleB,gold)\n",
    "CellPhoneDB_SSPB <- run_CellPhoneDB_SSP(CellPhoneDBB,sampleB,gold)\n",
    "NATMI_SSPB <- run_NATMI_SSP(NATMIB,sampleB,gold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "15fa571b",
   "metadata": {},
   "outputs": [],
   "source": [
    "CellChat_SSPC <- run_CellChat_SSP(CellChatC,sampleC,gold)\n",
    "iTalk_SSPC <- run_iTalk_SSP(iTalkC,sampleC,gold)\n",
    "SingleCellSignalR_SSPC <- run_SingleCellSignalR_SSP(SCSRC,sampleC,gold)\n",
    "CCInx_SSPC <- run_CCInx_SSP(CCInxC,sampleC,gold)\n",
    "scMLnet_SSPC <- run_scMLnet_SSP(scMLnetC,sampleC,gold)\n",
    "CellPhoneDB_SSPC <- run_CellPhoneDB_SSP(CellPhoneDBC,sampleC,gold)\n",
    "NATMI_SSPC <- run_NATMI_SSP(NATMIC,sampleC,gold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "5214c5fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "CellChat_SSPD <- run_CellChat_SSP(CellChatD,sampleD,gold)\n",
    "iTalk_SSPD <- run_iTalk_SSP(iTalkD,sampleD,gold)\n",
    "SingleCellSignalR_SSPD <- run_SingleCellSignalR_SSP(SCSRD,sampleD,gold)\n",
    "CCInx_SSPD <- run_CCInx_SSP(CCInxD,sampleD,gold)\n",
    "scMLnet_SSPD <- run_scMLnet_SSP(scMLnetD,sampleD,gold)\n",
    "CellPhoneDB_SSPD <- run_CellPhoneDB_SSP(CellPhoneDBD,sampleD,gold)\n",
    "NATMI_SSPD <- run_NATMI_SSP(NATMID,sampleD,gold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "ea48cdc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CellChat_SSPA</th><th scope=col>CellChat_SSPB</th><th scope=col>CellChat_SSPC</th><th scope=col>CellChat_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>1</td><td>0.2857143</td><td>0.5</td><td>0.6</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.004166667</td><td>0.008438819</td><td>0.01666667</td><td>0.03846154</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>1</td><td>0.9974962</td><td>0.998</td><td>0.996991</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & CellChat\\_SSPA & CellChat\\_SSPB & CellChat\\_SSPC & CellChat\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 1 & 0.2857143 & 0.5 & 0.6\\\\\n",
       "\tSensitivity & 0.004166667 & 0.008438819 & 0.01666667 & 0.03846154\\\\\n",
       "\tSpecificity & 1 & 0.9974962 & 0.998 & 0.996991\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | CellChat_SSPA &lt;dbl&gt; | CellChat_SSPB &lt;dbl&gt; | CellChat_SSPC &lt;dbl&gt; | CellChat_SSPD &lt;dbl&gt; | CellChat_SSPA &lt;dbl&gt; | CellChat_SSPB &lt;dbl&gt; | CellChat_SSPC &lt;dbl&gt; | CellChat_SSPD &lt;dbl&gt; | CellChat_SSPA &lt;dbl&gt; | CellChat_SSPB &lt;dbl&gt; | CellChat_SSPC &lt;dbl&gt; | CellChat_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 1 | 0.2857143 | 0.5 | 0.6 |\n",
       "| Sensitivity | 0.004166667 | 0.008438819 | 0.01666667 | 0.03846154 |\n",
       "| Specificity | 1 | 0.9974962 | 0.998 | 0.996991 |\n",
       "\n"
      ],
      "text/plain": [
       "            CellChat_SSPA CellChat_SSPB CellChat_SSPC CellChat_SSPD\n",
       "Accuracy    1             0.2857143     0.5           0.6          \n",
       "Sensitivity 0.004166667   0.008438819   0.01666667    0.03846154   \n",
       "Specificity 1             0.9974962     0.998         0.996991     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>iTalk_SSPA</th><th scope=col>iTalk_SSPB</th><th scope=col>iTalk_SSPC</th><th scope=col>iTalk_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0.3365385</td><td>0.2788462</td><td>0.2644231</td><td>0.375</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.3465347</td><td>0.2815534</td><td>0.2696078</td><td>0.3786408</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.9493392</td><td>0.9446903</td><td>0.943459</td><td>0.9525027</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & iTalk\\_SSPA & iTalk\\_SSPB & iTalk\\_SSPC & iTalk\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0.3365385 & 0.2788462 & 0.2644231 & 0.375\\\\\n",
       "\tSensitivity & 0.3465347 & 0.2815534 & 0.2696078 & 0.3786408\\\\\n",
       "\tSpecificity & 0.9493392 & 0.9446903 & 0.943459 & 0.9525027\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | iTalk_SSPA &lt;dbl&gt; | iTalk_SSPB &lt;dbl&gt; | iTalk_SSPC &lt;dbl&gt; | iTalk_SSPD &lt;dbl&gt; | iTalk_SSPA &lt;dbl&gt; | iTalk_SSPB &lt;dbl&gt; | iTalk_SSPC &lt;dbl&gt; | iTalk_SSPD &lt;dbl&gt; | iTalk_SSPA &lt;dbl&gt; | iTalk_SSPB &lt;dbl&gt; | iTalk_SSPC &lt;dbl&gt; | iTalk_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0.3365385 | 0.2788462 | 0.2644231 | 0.375 |\n",
       "| Sensitivity | 0.3465347 | 0.2815534 | 0.2696078 | 0.3786408 |\n",
       "| Specificity | 0.9493392 | 0.9446903 | 0.943459 | 0.9525027 |\n",
       "\n"
      ],
      "text/plain": [
       "            iTalk_SSPA iTalk_SSPB iTalk_SSPC iTalk_SSPD\n",
       "Accuracy    0.3365385  0.2788462  0.2644231  0.375     \n",
       "Sensitivity 0.3465347  0.2815534  0.2696078  0.3786408 \n",
       "Specificity 0.9493392  0.9446903  0.943459   0.9525027 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SingleCellSignalR_SSPA</th><th scope=col>SingleCellSignalR_SSPB</th><th scope=col>SingleCellSignalR_SSPC</th><th scope=col>SingleCellSignalR_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0.8333333</td><td>0.3333333</td><td>0.5833333</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.02525253</td><td>0.005050505</td><td>0.03535354</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.9995524</td><td>0.9991047</td><td>0.9977619</td><td>0.9995526</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & SingleCellSignalR\\_SSPA & SingleCellSignalR\\_SSPB & SingleCellSignalR\\_SSPC & SingleCellSignalR\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0.8333333 & 0.3333333 & 0.5833333 & 0\\\\\n",
       "\tSensitivity & 0.02525253 & 0.005050505 & 0.03535354 & 0\\\\\n",
       "\tSpecificity & 0.9995524 & 0.9991047 & 0.9977619 & 0.9995526\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | SingleCellSignalR_SSPA &lt;dbl&gt; | SingleCellSignalR_SSPB &lt;dbl&gt; | SingleCellSignalR_SSPC &lt;dbl&gt; | SingleCellSignalR_SSPD &lt;dbl&gt; | SingleCellSignalR_SSPA &lt;dbl&gt; | SingleCellSignalR_SSPB &lt;dbl&gt; | SingleCellSignalR_SSPC &lt;dbl&gt; | SingleCellSignalR_SSPD &lt;dbl&gt; | SingleCellSignalR_SSPA &lt;dbl&gt; | SingleCellSignalR_SSPB &lt;dbl&gt; | SingleCellSignalR_SSPC &lt;dbl&gt; | SingleCellSignalR_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0.8333333 | 0.3333333 | 0.5833333 | 0 |\n",
       "| Sensitivity | 0.02525253 | 0.005050505 | 0.03535354 | 0 |\n",
       "| Specificity | 0.9995524 | 0.9991047 | 0.9977619 | 0.9995526 |\n",
       "\n"
      ],
      "text/plain": [
       "            SingleCellSignalR_SSPA SingleCellSignalR_SSPB\n",
       "Accuracy    0.8333333              0.3333333             \n",
       "Sensitivity 0.02525253             0.005050505           \n",
       "Specificity 0.9995524              0.9991047             \n",
       "            SingleCellSignalR_SSPC SingleCellSignalR_SSPD\n",
       "Accuracy    0.5833333              0                     \n",
       "Sensitivity 0.03535354             0                     \n",
       "Specificity 0.9977619              0.9995526             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CCInx_SSPA</th><th scope=col>CCInx_SSPB</th><th scope=col>CCInx_SSPC</th><th scope=col>CCInx_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0.01923077</td><td>0.04807692</td><td>0.02403846</td><td>0.01442308</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.02116402</td><td>0.05235602</td><td>0.02645503</td><td>0.01587302</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.9407321</td><td>0.9425921</td><td>0.9411253</td><td>0.9406485</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & CCInx\\_SSPA & CCInx\\_SSPB & CCInx\\_SSPC & CCInx\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0.01923077 & 0.04807692 & 0.02403846 & 0.01442308\\\\\n",
       "\tSensitivity & 0.02116402 & 0.05235602 & 0.02645503 & 0.01587302\\\\\n",
       "\tSpecificity & 0.9407321 & 0.9425921 & 0.9411253 & 0.9406485\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | CCInx_SSPA &lt;dbl&gt; | CCInx_SSPB &lt;dbl&gt; | CCInx_SSPC &lt;dbl&gt; | CCInx_SSPD &lt;dbl&gt; | CCInx_SSPA &lt;dbl&gt; | CCInx_SSPB &lt;dbl&gt; | CCInx_SSPC &lt;dbl&gt; | CCInx_SSPD &lt;dbl&gt; | CCInx_SSPA &lt;dbl&gt; | CCInx_SSPB &lt;dbl&gt; | CCInx_SSPC &lt;dbl&gt; | CCInx_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0.01923077 | 0.04807692 | 0.02403846 | 0.01442308 |\n",
       "| Sensitivity | 0.02116402 | 0.05235602 | 0.02645503 | 0.01587302 |\n",
       "| Specificity | 0.9407321 | 0.9425921 | 0.9411253 | 0.9406485 |\n",
       "\n"
      ],
      "text/plain": [
       "            CCInx_SSPA CCInx_SSPB CCInx_SSPC CCInx_SSPD\n",
       "Accuracy    0.01923077 0.04807692 0.02403846 0.01442308\n",
       "Sensitivity 0.02116402 0.05235602 0.02645503 0.01587302\n",
       "Specificity 0.9407321  0.9425921  0.9411253  0.9406485 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>scMLnet_SSPA</th><th scope=col>scMLnet_SSPB</th><th scope=col>scMLnet_SSPC</th><th scope=col>scMLnet_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0</td><td>0.1111111</td><td>0</td><td>0.5</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0</td><td>0.005291005</td><td>0</td><td>0.02116402</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.9995413</td><td>0.9963336</td><td>0.9995413</td><td>0.9981643</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & scMLnet\\_SSPA & scMLnet\\_SSPB & scMLnet\\_SSPC & scMLnet\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0 & 0.1111111 & 0 & 0.5\\\\\n",
       "\tSensitivity & 0 & 0.005291005 & 0 & 0.02116402\\\\\n",
       "\tSpecificity & 0.9995413 & 0.9963336 & 0.9995413 & 0.9981643\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | scMLnet_SSPA &lt;dbl&gt; | scMLnet_SSPB &lt;dbl&gt; | scMLnet_SSPC &lt;dbl&gt; | scMLnet_SSPD &lt;dbl&gt; | scMLnet_SSPA &lt;dbl&gt; | scMLnet_SSPB &lt;dbl&gt; | scMLnet_SSPC &lt;dbl&gt; | scMLnet_SSPD &lt;dbl&gt; | scMLnet_SSPA &lt;dbl&gt; | scMLnet_SSPB &lt;dbl&gt; | scMLnet_SSPC &lt;dbl&gt; | scMLnet_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0 | 0.1111111 | 0 | 0.5 |\n",
       "| Sensitivity | 0 | 0.005291005 | 0 | 0.02116402 |\n",
       "| Specificity | 0.9995413 | 0.9963336 | 0.9995413 | 0.9981643 |\n",
       "\n"
      ],
      "text/plain": [
       "            scMLnet_SSPA scMLnet_SSPB scMLnet_SSPC scMLnet_SSPD\n",
       "Accuracy    0            0.1111111    0            0.5         \n",
       "Sensitivity 0            0.005291005  0            0.02116402  \n",
       "Specificity 0.9995413    0.9963336    0.9995413    0.9981643   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CellPhoneDB_SSPA</th><th scope=col>CellPhoneDB_SSPB</th><th scope=col>CellPhoneDB_SSPC</th><th scope=col>CellPhoneDB_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0.3793103</td><td>0.2291667</td><td>0.25</td><td>0.4177215</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.1195652</td><td>0.05978261</td><td>0.06989247</td><td>0.1823204</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.9829868</td><td>0.9824894</td><td>0.981569</td><td>0.9781991</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & CellPhoneDB\\_SSPA & CellPhoneDB\\_SSPB & CellPhoneDB\\_SSPC & CellPhoneDB\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0.3793103 & 0.2291667 & 0.25 & 0.4177215\\\\\n",
       "\tSensitivity & 0.1195652 & 0.05978261 & 0.06989247 & 0.1823204\\\\\n",
       "\tSpecificity & 0.9829868 & 0.9824894 & 0.981569 & 0.9781991\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | CellPhoneDB_SSPA &lt;dbl&gt; | CellPhoneDB_SSPB &lt;dbl&gt; | CellPhoneDB_SSPC &lt;dbl&gt; | CellPhoneDB_SSPD &lt;dbl&gt; | CellPhoneDB_SSPA &lt;dbl&gt; | CellPhoneDB_SSPB &lt;dbl&gt; | CellPhoneDB_SSPC &lt;dbl&gt; | CellPhoneDB_SSPD &lt;dbl&gt; | CellPhoneDB_SSPA &lt;dbl&gt; | CellPhoneDB_SSPB &lt;dbl&gt; | CellPhoneDB_SSPC &lt;dbl&gt; | CellPhoneDB_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0.3793103 | 0.2291667 | 0.25 | 0.4177215 |\n",
       "| Sensitivity | 0.1195652 | 0.05978261 | 0.06989247 | 0.1823204 |\n",
       "| Specificity | 0.9829868 | 0.9824894 | 0.981569 | 0.9781991 |\n",
       "\n"
      ],
      "text/plain": [
       "            CellPhoneDB_SSPA CellPhoneDB_SSPB CellPhoneDB_SSPC CellPhoneDB_SSPD\n",
       "Accuracy    0.3793103        0.2291667        0.25             0.4177215       \n",
       "Sensitivity 0.1195652        0.05978261       0.06989247       0.1823204       \n",
       "Specificity 0.9829868        0.9824894        0.981569         0.9781991       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 3 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>NATMI_SSPA</th><th scope=col>NATMI_SSPB</th><th scope=col>NATMI_SSPC</th><th scope=col>NATMI_SSPD</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Accuracy</th><td>0.3333333</td><td>0.2705882</td><td>0.280303</td><td>0.3895349</td></tr>\n",
       "\t<tr><th scope=row>Sensitivity</th><td>0.0923913</td><td>0.125</td><td>0.201087</td><td>0.3641304</td></tr>\n",
       "\t<tr><th scope=row>Specificity</th><td>0.983463</td><td>0.9698444</td><td>0.9537938</td><td>0.94893</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 3 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & NATMI\\_SSPA & NATMI\\_SSPB & NATMI\\_SSPC & NATMI\\_SSPD\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAccuracy & 0.3333333 & 0.2705882 & 0.280303 & 0.3895349\\\\\n",
       "\tSensitivity & 0.0923913 & 0.125 & 0.201087 & 0.3641304\\\\\n",
       "\tSpecificity & 0.983463 & 0.9698444 & 0.9537938 & 0.94893\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 3 × 4\n",
       "\n",
       "| <!--/--> | NATMI_SSPA &lt;dbl&gt; | NATMI_SSPB &lt;dbl&gt; | NATMI_SSPC &lt;dbl&gt; | NATMI_SSPD &lt;dbl&gt; | NATMI_SSPA &lt;dbl&gt; | NATMI_SSPB &lt;dbl&gt; | NATMI_SSPC &lt;dbl&gt; | NATMI_SSPD &lt;dbl&gt; | NATMI_SSPA &lt;dbl&gt; | NATMI_SSPB &lt;dbl&gt; | NATMI_SSPC &lt;dbl&gt; | NATMI_SSPD &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| Accuracy | 0.3333333 | 0.2705882 | 0.280303 | 0.3895349 |\n",
       "| Sensitivity | 0.0923913 | 0.125 | 0.201087 | 0.3641304 |\n",
       "| Specificity | 0.983463 | 0.9698444 | 0.9537938 | 0.94893 |\n",
       "\n"
      ],
      "text/plain": [
       "            NATMI_SSPA NATMI_SSPB NATMI_SSPC NATMI_SSPD\n",
       "Accuracy    0.3333333  0.2705882  0.280303   0.3895349 \n",
       "Sensitivity 0.0923913  0.125      0.201087   0.3641304 \n",
       "Specificity 0.983463   0.9698444  0.9537938  0.94893   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cbind(CellChat_SSPA,CellChat_SSPB,CellChat_SSPC,CellChat_SSPD)\n",
    "cbind(iTalk_SSPA,iTalk_SSPB,iTalk_SSPC,iTalk_SSPD)\n",
    "cbind(SingleCellSignalR_SSPA,SingleCellSignalR_SSPB,SingleCellSignalR_SSPC,SingleCellSignalR_SSPD)\n",
    "cbind(CCInx_SSPA,CCInx_SSPB,CCInx_SSPC,CCInx_SSPD)\n",
    "cbind(scMLnet_SSPA,scMLnet_SSPB,scMLnet_SSPC,scMLnet_SSPD)\n",
    "cbind(CellPhoneDB_SSPA,CellPhoneDB_SSPB,CellPhoneDB_SSPC,CellPhoneDB_SSPD)\n",
    "cbind(NATMI_SSPA,NATMI_SSPB,NATMI_SSPC,NATMI_SSPD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "f7388a22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzde5wXdb0/8NkLe+ESV0kERSUDMrzrgpq6LrCCLFqKHC+kFWV5I4527JiGWZl2\nys1UzFJTMTBMLdDdI5eDhrpYmQoHhUMKntIF5a7I7sL3u78/+P44hQS7y35ndmafzz/2Ad/v\njPPmi66vnZnPa3IaGxsDAADiLzfqAQAAaB2CHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2\nAAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcA\nkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJ\nIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCC\nHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgB\nACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBA\nQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSE\nYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2\nAAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcA\nkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACREftQDxMCmTZsefPDBrVu3\nRj0IANAmFBcXX3zxxV27do16kF0Jdnv3q1/9atKkSVFPAQC0Ifn5+ZdddlnUU+xKsNu7bdu2\nBUFw3333HXnkkVHPAgBE7NVXX/3Sl760Ix60NYJdUw0cOPDYY4+NegoAIGJ1dXVRj/BPWTwB\nAJAQgh0AQEIIdgAACSHYAQAkhGAHAJAQgh0AQELEte5kQ+3K5ctXrFm/ecuHdflFnbr23P+w\nQYMP7dMt6rkAACITs2DXmNo0s/I7P71v+gvL1nz03f0HDb1g4qQbJo3vlp8T/mwAANGKU7BL\nNbz9heOPnLZ4XV6HHiWnjz1i8IA+vboVFuZvr6/fuHb1WyuWvrDwxduuOf+h6U++WvPQAQWu\nMgMA7Uucgl3N1WdMW7zu5Ctun3HLZf067WbydMO6GbdePmHK9BFXTlx6z2mhDwgAEKU4nda6\nbtqKzn2+uvCOq3ab6oIgyC3oeeENj9xd8vE3Hrk+5NkAACIXp2C3ZMu2zgdV7HWzY0/pve3D\npSHMAwDQpsQp2J3Vs3jDsltWN6T3tFF66/0zVxV1Lw9rKACAtiJOwe5bt5bXb1r46aHnPfz0\nS1tSjbu+3Vj/2sInJo4YfPeqzadNmRLFgAAAUYrT4onDLn70F38ceenUxyec8VheQddDDxtw\nwH7dCgs7pBrqN62tfXPFG+vrtufk5JRedtesywdHPSwAQNjiFOyCIHfinfNGTfjtXb+cUbVg\n0bLXX16xNHPeLie3sN+Aw0eUlp8/8aqzju8b7ZQAAJGIV7ALgiDoW3L2zSVn3xwEjdu3btz4\n/patDQXFHbt0616slBgAaN/iF+x2yskv7t6ruHvUYwAAtBFxWjwBAMAexPiM3W41bH6+/8Bz\ngyCora1tyvapVKqqqqqurm4P27z88stBEGzbtq1VJiTbmvJ3ulvpdHrJkiVDhgzJzW3eDzxF\nRUWjR4/Oy8tr7hEBoHUlLdg1NjasXr266dsvWLBg7NixTdly+vTpp512WgvHIkRN/zttRXPn\nzh0+fHjIBwWAXSQt2BV0Pm7RokVN3760tHTWrFl7PrszderUZ555pl+/fvs8HWFoyt/pbtXU\n1FRWVk6ePHnYsGHN2rGoqKi0tLS5hwOAVpe0YJeT16WkpKTp2+fl5VVU7OUxZVVVVUEQNPfy\nHFFpyt/pP1NZWTls2LBx48a17kgAEA5hBQAgIQQ7yCguLt75FQDiKGmXYqHFRo0a9fTTT5eV\nlUU9CAC0UJyC3cbVtVtS6SZu3LevB4vRPHl5eSNHjox6CgBouTgFu28c/cl7V3/QxI0bGxuz\nOgzJk0ql5s+fX1ZWppEOgJiKU7D73rzqgQ/c9e3KX29NNXYfctpJ/TtHPRGJUl1dXVFRMXv2\n7DFjxkQ9CwC0RJyC3ccPP/ma/zi5tMebx133h8GX3z370kFRT0SibN26dedXAIij+K2KHXL5\nj6MeAQCgLYpfsCv42MnH9Nu/a5G7oAAA/kGcLsXu9NJfa6MegQTSYwdA3MUy2EE26LEDIO4E\nO8jQYwdA3MXvHjvIklQqNWfOnFQqFfUgANBCgh1kVFdXl5eXV1dXRz0IALSQYAcZeuwAiDvB\nDgAgIQQ7AICEEOwgQ48dAHGn7gQy9NgBEHeCHWTosQMg7lyKhQw9dgDEnWAHGXrsAIg7wQ4y\n9NgBEHeCHQBAQgh2AAAJIdhBhh47AOJO3Qlk6LEDIO4EO8jQYwdA3LkUCxl67ACIO8EOMvTY\nARB3gh1k6LEDIO4EOwCAhBDsAAASQrCDDD12AMSduhPI0GMHQNwJdpChxw6AuHMpFjL02AEQ\nd4IdZOixAyDuBDvI0GMHQNwJdgAACSHYAQAkhGAHGXrsAIg7dSeQoccOgLgT7CBDjx0AcedS\nLGTosQMg7gQ7yNBjB0DcCXaQoccOgLgT7AAAEkKwAwBICMEOMvTYARB36k4gQ48dAHEn2EGG\nHjsA4s6lWMjQYwdA3Al2kKHHDoC4E+wgQ48dAHEn2AEAJIRgBwCQEIIdZOixAyDu1J1Ahh47\nAOJOsIMMPXYAxJ1LsZChxw6AuBPsIEOPHQBxJ9hBhh47AOJOsAMASAjBDgAgIQQ7yNBjB0Dc\nqTuBDD12AMSdYAcZeuwAiDuXYiFDjx0AcSfYQYYeOwDiTrCDDD12AMSdYAcAkBCCHQBAQgh2\nkKHHDoC4U3cCGXrsAIg7wQ4y9NgBEHcuxUKGHjsA4k6wgww9dgDEnWAHGXrsAIg7wQ4AICEE\nOwCAhBDsIEOPHQBxp+4EMvTYARB3gh1k6LEDIO5cioUMPXYAxJ1gBxl67ACIO8EOMvTYARB3\ngh0AQEIIdgAACSHYQYYeOwDiTt0JZOixAyDuBDvI0GMH8M+kUqmqqqq6urrm7phOp5csWTJk\nyJDc3OZdJCwqKho9enReXl5zj9jOCXaQkUql5s+fX1ZW5vsIwC4WLFgwduzYkA86d+7c4cOH\nh3zQuBPsIKO6urqiomL27NljxoyJehaAtqW0tHTWrFktOGNXU1NTWVk5efLkYcOGNWvHoqKi\n0tLS5h4OwQ4y9NgB/DN5eXkVFRUt27eysnLYsGHjxo1r3ZHYLatiAQASQrADALJFk1TIXIqF\nDN99AFqdJqmQJSHYpba+/dtHq/7y9oYeBw4e+dlR/Tsl4Q9F+Hz3AWh1mqRCFrNLsRuW/m7C\nmaf279mxe59PXv7j+UEQrP3T/YN7H3ruxV/55nXXfmXC2MN6f2LKb5ZHPSaxtOO7j64TgFaU\nSqXmzJmTSqWiHqS9iFOw+3DNU0OOPefhqt+/l9Mjte6NqdcM/+Kv5o4s/dqb2/b72r9/774H\n7ply9cVdtv31e/9yzL2r3o96WOLHdx+AVlddXV1eXl5dXR31IO1FnILdrIu++k5D+psz/vzh\n2r9t/KB2yqgDf3nRyMX1H/vdimVTb/7WFy/+yo0/euCNVx8oaNx6wwWPRz0s8eO7D0Cr0yQV\nsjgFux/UvNvloBt+8C9HB0GQW9D72mk/CYKg9wlTzzyw885tug2e8MPDuq9b/OPIpiS2fPcB\nIO7iFOzeqNve8ePH7/xt4cc+EwRB10/13WWzQQd2StWtDHUyAIA2IE7B7qSPFWxeOW3nDVCb\nV94fBMG7zy3aZbPZr28s6HJCuKMBALuhSSpkcQp2N1w44MP3ZpZefvsfl/7lT888dsHI7+cX\nd92w7N+u/83inds8e88X73j7/QPHfDPCOYkp330AWt2OJqlRo0ZFPUh7EafKt2E/qhpbNWTW\n1K+fMPXrQRDkduhxz+Klz5056PvjjnzixBHHDuz91yXPPfOntwo6f/rhqadGPSzxo8cOoNXp\nsQtZnIJdXuFBj7/2+oN3/Pz3L770focD/mXy98YN2u/iVxYGZ5334IK5r70QBEFw6Enj73r4\n3hO6FEQ9LPHjuw9Aq0ulUvPnzy8rK9MSGo44BbsgCPIKD/jiNTd+8e9e6dBlyAP/9fqP3lq+\n4m8bu/cbOKh/t8iGI+Z89wFoddXV1RUVFbNnzx4zZkzUs7QLcbrHbg969R847KQSqY59occO\noNVpkgpZzM7YtbpUKlVVVVVXV7eHbVatWhUEQTqdDmkmIuK7DwBxl7Rg17D5+f4Dzw2CoLa2\ntinbL1iwYOzYsU3ZcuVK3Xhhe+edd84777w9x+5WtGnTpiAIrrvuultvvTWcI954442uTQDQ\nipIW7BobG1avXt307UtLS2fNmrXn6DB16tRnnnnmkEMO2efpaJ7XX3/9+eef79OnT/fu3UM4\nXEFBQf/+/QsKCkI4abdt27YVK1Y8++yzgh2QbJqkQpa0YFfQ+bhFi3atLN6DvLy8ioqKPW9T\nVVUVBEFubkLuR4ydL3/5y+PGjYt6ilZWW1trBS7QHmiSClnSgl1OXpeSkpKopwAAgkCTVOji\nGuw21K5cvnzFmvWbt3xYl1/UqWvP/Q8bNPjQPlbFAkAbokkqZDELdo2pTTMrv/PT+6a/sGzN\nR9/df9DQCyZOumHS+G75OeHPBgDsQo9dyOIU7FINb3/h+COnLV6X16FHyeljjxg8oE+vboWF\n+dvr6zeuXf3WiqUvLHzxtmvOf2j6k6/WPHRAgVviACBimqRCFqdgV3P1GdMWrzv5ittn3HJZ\nv067mTzdsG7GrZdPmDJ9xJUTl95zWugDAgBEKU6nta6btqJzn68uvOOq3aa6IAhyC3peeMMj\nd5d8/I1Hrg95NgCAyMUp2C3Zsq3zQXupJgmC4NhTem/7cGkI8wAAe6bHLmRxCnZn9SzesOyW\n1Q17fLRXeuv9M1cVdS8PaygA4J/a0WM3atSoqAdpL+IU7L51a3n9poWfHnrew0+/tCXVuOvb\njfWvLXxi4ojBd6/afNqUKVEMCAD8gx09drpOQhOnxROHXfzoL/448tKpj08447G8gq6HHjbg\ngP26FRZ2SDXUb1pb++aKN9bXbc/JySm97K5Zlw+OelgAQI9d2OIU7IIgd+Kd80ZN+O1dv5xR\ntWDRstdfXrE0c94uJ7ew34DDR5SWnz/xqrOO7xvtlADADnrsQhavYBcEQdC35OybS86+OQga\nt2/duPH9LVsbCoo7dunWvVgpMQC0MXrsQha/YLdTTn5x917F3aMeAwCgjYjT4gkAAPZAsAMA\nskWPXchifCkWAGjjdvTYlZWVRT1IeyHYAQDZsqPHLuop2hGXYgGAbEmlUnPmzEmlUlEP0l4I\ndgBAtlRXV5eXl1dXV0c9SHsh2AEA2aLHLmSCHQBAQgh2AAAJIdgBANmixy5k6k4AgGzRYxcy\nwQ4AyBY9diFzKRYAyBY9diET7ACAbNFjFzLBDgDIFj12IRPsAAASQrADAEgIwQ4AyBY9diFT\ndwIAZIseu5AJdgBAtuixC5lLsQBAtuixC5lgBwBkix67kAl2AEC26LELmWAHAJAQgh0AQEII\ndgBAtuixC5m6EwAgW/TYhUywAwCyRY9dyFyKBQCyRY9dyAQ7ACBb9NiFTLADALJFj13IBDsA\ngIQQ7AAAEkKwAwCyRY9dyNSdAADZoscuZIIdAJAteuxCJtgBQPvyyiuvrF27NpxjpdPpl19+\n+eijj87NDenur6OPPrpnz57hHKsNEuwAoB157733jj322HQ6HfUg2XLhhRc+/PDDUU8RGcEO\nANqRrVu3ptPpYcOGhXOFNJ1Or1y58pBDDgnnjN33vve9LVu2hHCgNkuwA4B2Z+DAgeeee27U\nU7S+m2++OeoRIqbuBAAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjB\nDgAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwA\nABJCsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjBDgAg\nIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEEOwAABJC\nsAMASAjBDgAgIQQ7AICEEOwAABJCsAMASAjBDgAgIQQ7AICEyI96gBZofO+vH+x3YJf//9v0\nq88+9fuXXvsgXXjIp44fXX7ix/JyopwOACAiMQt2q+ZM/fxV317a+B/rln8hCIKt7z57Ufn4\nx19Zs3ODjn2OuW3Gk5ee2ie6GQEAohGnYLf25R8PHvWNhpxOI750YBAEjan3xx995ux3thwx\n6pLzyo7r97H0f//x6Tvvq7p8xJHdV60874BOUc8LABCqOAW7O8d/vyGn472L3vzCcfsFQVD7\n3MTZ72w55t+efOnWMzNbfPnKb3zproNOvPLr4x8/b+GEKGcFAAhdnBZP3LVqc/dP3r4j1QVB\nsGr64iAI7vv2yL/fpnfJ5T8e2GPtn2+JYD4AgEjFKdj1yM/NK9y5ZiLILcgNguCgwl1POh66\nX1GqoTbUyQAA2oA4BbuvH959/evfeHFTw47fDrjkM0EQ3PTSu3+/TeP2Dd9/ZW1xzzERzAcA\nEKk4BbsLfvX9Dtv/evrg0+96bOGm7en9jr3rGyft/7PyMb985s0dG3xY+8fJY49+fnP9qd/+\n92hHBQAIX5wWT3T95MSXH33n9PNvuuLcUyYVdvvEoE/u3/WA+k1/+mLpgKv2O6hfp/oVb72b\namw86cs/+d3XBkc9LABA2OJ0xi4IgoGf/fabtYtvu/7yEwf1fuf1l579/Z93vP7Be/9bu7Wo\n7LxLpz3zl+d+PilfRTEA0P7E6YzdDoXdPzX5u3dO/m4QNG5bv3btlq3b8gqKOnXu3rVzh6hH\nAwCIUvyC3f/J6dBjvz49op4CAKCNiHOwaw2pVKqqqqqurm4P26xatSoIgnQ6HdJMAAAtkrRg\n17D5+f4Dzw2CoLa2SVV2CxYsGDt2bFO2XLly5T5NBgCQZUkLdo2NDatXr2769qWlpbNmzdrz\nGbupU6c+88wzhxxyyD5PBwCQRUkLdgWdj1u0aFHTt8/Ly6uoqNjzNlVVVUEQ5ObGbAUxANDe\nJC3Y5eR1KSkpiXoKAIAIxDXYbahduXz5ijXrN2/5sC6/qFPXnvsfNmjwoX26RT0XAEBkYhbs\nGlObZlZ+56f3TX9h2ZqPvrv/oKEXTJx0w6Tx3TQUAwDtT5yCXarh7S8cf+S0xevyOvQoOX3s\nEYMH9OnVrbAwf3t9/ca1q99asfSFhS/eds35D01/8tWahw4ocEscANC+xCnY1Vx9xrTF606+\n4vYZt1zWr9NuJk83rJtx6+UTpkwfceXEpfecFvqAAABRitNpreumrejc56sL77hqt6kuCILc\ngp4X3vDI3SUff+OR60OeDQAgcnEKdku2bOt80F6qSYIgOPaU3ts+XBrCPAAAbUqcgt1ZPYs3\nLLtldcMeH+2V3nr/zFVF3cvDGgoAoK2IU7D71q3l9ZsWfnroeQ8//dKWVOOubzfWv7bwiYkj\nBt+9avNpU6ZEMSAAQJTitHjisIsf/cUfR1469fEJZzyWV9D10MMGHLBft8LCDqmG+k1ra99c\n8cb6uu05OTmll9016/LBUQ8LABC2OAW7IMideOe8URN+e9cvZ1QtWLTs9ZdXLM2ct8vJLew3\n4PARpeXnT7zqrOP7RjslAEAk4hXsgiAI+pacfXPJ2TcHQeP2rRs3vr9la0NBcccu3boXKyUG\nANq3+AW7nXLyi7v3Ku4e9RgAAG1EnBZPAACwB00Ndudeet0Tv1+ayuosAADsg6YGu8d+/oPP\nnfrpbgce+eVrf7hg8dtZnQkAgBZoarB77blZ377igr4NK+794bWnH9mvz6dP/cYPfv7y/27O\n6nAAADRdU4Pd4JMqvnPHr5at3vjneY/+28Rziv626EfXXXrswT0+dfJZ37975psbG7I6JQAA\ne9XMxRM5BUeXnXvrL37z5vr1NU9Nu+qiMza+XHX9ZeMP69X9xDEX3zVjzto9P+8LAICsafmq\n2Ny8/KLiTl06dQiCIJ368MWqaVdcUH5A9/6X/vDp1hsPAICmal6wS9W/91+P3XvFhWce2LV7\nyRnn3/rzmR/0PvbKb1c+8+pf33978S9uueaIj63/+bVnfGW+1RUAAGFrakHx7x74yRNPPDH7\nP59f35DKycnpf9TpV59zzrnnnjN0YO//v0m/idf+x4RLK4q6n/rklFeDMs/1AgAIVVOD3dlf\nmJyTk/fJkpFfPeecc8753DGH7P6JD/nFh+y///4HH9Wz9SYEAKBJmhrsbrx92jnnnP3pvp33\nvFle4YG1tbX7PBUAAM3W1GA35aqLgsb6uQ/dPvf1vj/8wbk7Xhx9zkWfKT/vqokVnXJzsjYh\n7d1NN9100003RT0FAMRAUxdPpLe9+5WhB428+Os/e/gPO198+onp11161sHHX7J6m5YTAICI\nNTXY/fnG0b/4w7vHf/57c+b9284XN/51yY++8pm1f35o5LU12RkPAICmamqwu+me1zr1vqjm\nwW8NHdhr54td+h5+9T3PfLVP5xUPuFIGABCxpga7ZzfV9zjq83m7+yeMO75Xw+YXWnMoAACa\nr6mLJwZ37PA/q14JghEffWvxivfzOw5s1ang/3z7298eN25c1FO0stra2pEjR0Y9BQBJ09Qz\ndjeOPnDjim9eO3PJLq8vn/Xta5at73Pq9a09GAAAzdPUM3Zl9848ad5JPxx/xJNTPze2rKRf\nr851m9b8+ZknZ8x5Ob/j4Q8+PCqrUwIAsFdNDXYdOh09b9nz1375irt/98Qtzz6+8/VPnX7R\nHffffWrXwuyMBwBAUzU12AVBUNjz6J88/vwta1ct+uOS1es3F3TpMeiooZ86aPfPFgMAIGTN\nCHY7FPU6+LRRB//9K8t+dvqJU95fv+aPrTYUAADN14xg99bcB+58YsGq9z78x5fTS59+fnN9\nt9YdCwCA5mpqsHtnwTcHnvHD+nTjR9/q0Hn/s7/xUKtOBQBAszW17uTnX/rZtrzuD734lw/f\nf+9bQ3r2LX2krq7u/fdW/fjznyruXXrPjWVZnRIAgL1qarD7Ze2WHgN/POGEAcWde11y7eHr\nXn2gsLCwc6/+k+9fdPz6Jypu3bXfDgCAkDU12L23LdWp/4E7ft3zhIH1GxdsSTcGQZCT12XK\nmANf+cl3sjUgAABN09Rgd1Sngs3LF+/4dVH34Y3p+ofXZFZRFPcprt8wLyvTAQDQZE0Ndlef\n+PGNb1573bT567eli3qc2acg76ffXxgEQdC4/ZEn/je/+LAszggAQBM0NdiNfmhq/4LGH3x+\n+IUvrM7J7VQ56sDXp44eWv65kUMP/vFfNvY/+7tZnRIAgL1qat1J8X5nLn1j4a3/cX/RfsVB\nEJwzo/rCkWc+POeJnNyCY87999/eW57NIQEA2LtmFBR3PGDodyqHZnYrHjRt4Rt3vfe37Z37\n9CjOy85sAAA0Q5Muxaa3vTd58uRbH3trl9c/tl8/qQ4AoI1oUrDL7bBf9c/vuvPu17I9DQAA\nLdbUxRMPfOMza2omv/bh9qxOAwBAizX1HruhN86fnnvR6UPKv/HtK0qPHdyjS3HOP27Qv3//\nVh8OAICma2qw69ChQxAEjanUNZf81243aGxsbLWhAABovqYGu4kTJ2Z1DgAA9lFTg93dd9+d\n1TkAANhHTV08AQBAG9fUM3aHHHLInjdYuXLlPg8DAEDLNTXYde7ceZdXtm1Z98aq1dsbGwu7\nHVUx/BOtPRgARCaVSlVVVdXV1TVrr3Q6vWTJkiFDhuTmNvuCWFFR0ejRo/Py1P6zT5oa7JYs\nWfLRFxs2/c+Prplw/X0vFZ70i1adCgCitGDBgrFjx4Z80Llz5w4fPjzkg5IwzXhW7EcVdP3k\ndb+oqZ3Tfeo3hn//a+v6F/o5A4AkKC0tnTVrVnPP2NXU1FRWVk6ePHnYsGHNPWJRUVFpaWlz\n94Jd7FOwC4IgCHIv/peD7/zh4mUfbhfsAEiGvLy8ioqKFuxYWVk5bNiwcePGtfpI0BStsCr2\nncUbc/M6De9euO//KAAAWqypZ+zq6+s/+mJ6+wev/ud9E+b9rbjXBCfrAGjniouLd36FSDQ1\n2BUVFf2zt3Jy8r5y142tMw4AxNaoUaOefvrpsrKyqAeh/WpqsDv33HN3+3rHXged+rkrvjji\n4FabCADiKS8vb+TIkVFPQbvW1GD36KOPZnUOAIi7VCo1f/78srIydXREpTmLJxrr5z50+7/9\n+292vjD6nIt+8PNZW9KNrT8XAMRNdXV1eXl5dXV11IPQfjU12KW3vfuVoQeNvPjrP3v4Dztf\nfPqJ6dddetbBx1+yels6O+MBQGxs3bp151eIRFOD3Z9vHP2LP7x7/Oe/N2fev+18ceNfl/zo\nK59Z++eHRl5bk53xAABoqqYGu5vuea1T74tqHvzW0IG9dr7Ype/hV9/zzFf7dF7xwE3ZGQ8A\ngKZqarB7dlN9j6M+v7t7QXPHHd+rYfMLrTkUAMSQHjsi19RgN7hjhw9WvbLbtxaveD+/48DW\nGwkAYmlHj92oUaOiHoT2q6nB7sbRB25c8c1rZy7Z5fXls759zbL1fU69vrUHA4CY2dFjp+uE\nCDW1x67s3pknzTvph+OPeHLq58aWlfTr1blu05o/P/PkjDkv53c8/MGH/ehB9UwAACAASURB\nVHQCQHunx47INTXYdeh09Lxlz1/75Svu/t0Ttzz7+M7XP3X6RXfcf/epXQuzMx4AxEZ1dXVF\nRcXs2bPHjBkT9Sy0U00NdkEQFPY8+iePP3/L2lWL/rhk9frNBV16DDpq6KcO6p694QAgRvTY\nEblmBLugsX7utJ/Nfb3vD3+QeW7s6HMu+kz5eVdNrOiUm5OV6QAAaDJPngAASAhPngCA1qHH\njsg19VLszidP/P06nx1PnvjL7K4PPHBTcNvT2ZgPAOJiR49dWVlZ1IPs3QMPPPDAAw9EPUXr\nsx7ZkycAoHXosSNynjwBAK0jlUrNmTMnlUpFPQjtlydPAEDrqK6uLi8vr66ujnoQ2i9PngCA\n1qHHjsh58gQAtDuXXHLJ1VdfHfUUre+YY46JeoSIefIEAEBCNOfJE0EQBEFRr4NPG3XwLi++\n9ed5/Y8Z3joTAUA86bEjck1dPLFb61a8eOdNXz9pUO+Djx3RWgMBQEzt6LEbNcp950Sm2Wfs\ngiD4sHbpozNmzJgx4+k/vbnjlaJen2zVqQAgfnb02EU9Be1aM4Ldts1vzf71I9NnzJj1zOJt\njY1BEOR37HPGuf9ywQUXfG7kcVmbEADiIZVKzZ8/v6ysTEcxUdl7sEs3rJ3/+Mzp06c/VlXz\nfiodBEF+0X5B3XsfP+Gnrz9/eff8fbqYCwCJUV1dXVFRMXv27DFjxkQ9C+3UnoJdzVPTZkyf\n/uvH5r5bnwqCIL+o94hzP3feuPM+O/bUXgV5hT0GS3UAsJMeOyK3p2B34pjPB0GQX9R75Phz\nxp037nMVp/boIMkBALRRew9qBw097YzRZ44ZdYpUBwDQlu0pq93+7StO+ESPN5+Z+a8Xj+nX\ntfeI8Zc9MOv5LenG0IYDgBjRY0fk9hTsrvrOHS+uWLdiUdWNV144oOuH82be/YWzTu7R49Dz\nvvqt0OYDgLjQY0fk9n519RMlo6b89OHl727843/+6usTRveo/9uj99wcBMHbCyaMufjqR+a+\n3OAUHgD8/x47XSdEqMm3zeUUHld+QeVDT729qXbOjDsvHjOsaPu7Tz102/kjj+l+wOGXXP2D\nbA4JADGQSqXmzJmTSqWiHoT2q9nrIXILeo34l8sfmP3C+rUrfj31u2NPGlS35vUHb7suG8MB\nQIxUV1eXl5dXV1dHPQjtV8sXuhZ0O/S8r13/u+de37DyTz/7/r+24kwAEEd67IhcS54Vu4uP\n9T/m0uuO2fd/DgAA+yI51XQTJkyYdPOSqKcAAIhMcoLdww8//Njcd6KeAoD2S48dkWuFS7Gh\nefNXP5n2l0172OD9Vb/6zncW7fj1lClTQhkKADJ29NiVlZVFPQjtV5yC3f8+fseNj7+5hw02\nr5p2442ZXwt2sFMqlaqqqqqrq2vWXul0esmSJUOGDMnNbd6p/aKiotGjR6vyoh3a0WMX9RS0\na3EKdqfMeP6Wy8Z/877fF/U46nt3XP+JTv8w/Nlnn93z01Pu+97RUY0HbdaCBQvGjh0b5hHn\nzp07fPjwMI8IbUEqlZo/f35ZWZkfbIhKnIJdbsH+19777OjRt55z8Q3XT7r5tumPfm3EoX+/\nQVGvYWedVR7VeNBmlZaWzpo1q7ln7GpqaiorKydPnjxs2LBm7VhUVFRaWtqsXSAZqqurKyoq\nZs+ePWbMmKhnoZ2KU7DbYcjnrl1y6sivj//c5eWfnHXFbQ/fdkXP/OQsAYFsyMvLq6ioaMGO\nlZWVw4YNGzduXKuPBImkx47IxTISFfY8+u55b/zuR19+4e7JAwaP+s0ra6OeCAAgerEMdkEQ\nBEFuxb/e/dYrT5yY94fxx/W/5Pu/jnoeSBrFDQCxE99gFwRB0O3wsU/99xs/ueyUaTecH/Us\nkDQ7ihtGjRoV9SAQG34cInLxu8duFzn5Pa78afXoioeefG1D536Dox4HkkNxAzSXHjsiF/tg\nt8OAEZ+fNCLqISBZFDdAc/lxiMjF+1IskD3V1dXl5eXV1dVRDwKxkUql5syZk0qloh6E9ish\nZ+x2atj8fP+B5wZBUFtb25Ttm9LIv2rVqiAI0ul0awwIsaG4AZpLjx2RS1qwa2xsWL16ddO3\nb3oj/8qVK1s6FLSaX/7yl6GdQvvb3/4WBEFlZeWjjz4azhGPP/74b3zjG+EcC7LBj0NELmnB\nrqDzcYsWLWr69k1p5J86deozzzxzyCGH7PN0sK9uu+22//7v/w7ziDU1NaEda9asWYIdwL5I\nWrDLyetSUlLS9O2b0shfVVUVBEFzn4MO2dDY2PiJT3ziiSeeCOFY6XR60aJFQ4cODedf/kmT\nJj333HMhHAggweIa7DbUrly+fMWa9Zu3fFiXX9Spa8/9Dxs0+NA+3aKei9b30ksv5eTkRD1F\nK9u0aVPUI+xdbm7uiSeeGPUUECd67IhczIJdY2rTzMrv/PS+6S8sW/PRd/cfNPSCiZNumDS+\nW37SckD71KVLlyAInnrqqaeeeirqWbKic+fOUY8AtCY9dkQuTsEu1fD2F44/ctridXkdepSc\nPvaIwQP69OpWWJi/vb5+49rVb61Y+sLCF2+75vyHpj/5as1DBxS4chp7J5xwQk1NzQcffBDO\n4RYtWnTDDTd897vfHTp0aAiHy8nJcT6MtqMpFQEflU6nlyxZMmTIkOZery8qKho9enTyKhL1\n2BG5OAW7mqvPmLZ43clX3D7jlsv6ddrN5OmGdTNuvXzClOkjrpy49J7TQh+Q1hdOxtphw4YN\nQRAMHDhw+PDhoR0U2oimVwS0lrlz5ybvvzW13kQuTsHuumkrOvf56sI7rvpnG+QW9Lzwhkc+\nqHp20iPXB/e4CxugqZpSEfBRNTU1lZWVkydPHjZsWLN2LCoqKi0tbdYusaDHjsjFKdgt2bKt\n86C9rGANguDYU3pv+9PSEOYBSIymVATsVmVl5bBhw8aNG9fqI8WRHjsiF6cb0c7qWbxh2S2r\nG/b4BIj01vtnrirqXh7WUCSH5WwAxF2cgt23bi2v37Tw00PPe/jpl7akGnd9u7H+tYVPTBwx\n+O5Vm0+bMiWKAYm3HcvZRo0aFfUgEBt+HIK2Jk6XYg+7+NFf/HHkpVMfn3DGY3kFXQ89bMAB\n+3UrLOyQaqjftLb2zRVvrK/bnpOTU3rZXbMuHxz1sMSP5WzQXNo9diHpErk4BbsgyJ1457xR\nE3571y9nVC1YtOz1l1cszZy3y8kt7Dfg8BGl5edPvOqs4/tGOyUxZTkbNJcfh3Yh6RK5eAW7\nIAiCviVn31xy9s1B0Lh968aN72/Z2lBQ3LFLt+7FSonZN5aztR8t62wL1LZ9hB+HdiHpErn4\nBbudcvKLu/cq7h71GCSG5WztR/idbUFCa9v8OLQLSZfIxTjYAbRMyzrbArVtH+HHoV1IukRO\nsAPanRZ3tgVq29gjSZfIxanuBLLKcjYA4s4ZO8iwnI29kv534QOBtkawgwzL2dgr6X8XPpBd\nSLpEzqVYyEilUnPmzEmlUlEPQtu1I/1b8LiTD2QXHmBD5JyxgwzL2dirWJRZ/P73v1+2bFk4\nx0qn08uWLRs0aFBzi/1arKSk5MgjjwznWC3gxD+RE+wgw3I29ioW6f/MM8/84IMPop4iW44+\n+ug///nPUU/xT8Ui+pNsgh1AU8Ui/Tc0NBx11FGXX355CMdKp9Ovv/764MGDwzljN2XKlIaG\nhhAO1GKxiP4km2AHxNtf//rXbdu2hXOsNWvW7Pj65ptvhnC4Dh06HHjggS3YsUePHkOHDm31\neXbrxBNPDOdAQRB07NgxtGO1TCyiP8km2EGG5Wxx9MQTT3zuc58L+aBXXnnllVdeGc6xHn/8\n8c9+9rPhHAtIAMEOMhQ3xNG7774bBMGoUaP69OkTzhHffvvtvn37hnCg2tra6urqHX9AgCYS\n7CDDcrb4Ouecc0pKSqKeopW9+OKL1dXVUU9B8zjxT+T02EGGHjtgH+mxI3KCHWRUV1eXl5c7\nRwK0mMZmIifYQYblbMA+cuKfyAl2ANA6nPgncoIdALQOJ/6JnGAHGZazARB36k4gQ48dAHEn\n2EGGHjtgHznxT+RcioUMy9mAfaTHjsgJdpBhORuwj/TYETnBDjIsZwP2kRP/RE6wA4DW4cQ/\nkRPsAKB1OPFP5AQ7yLCcDYC4U3cCGXrsAIg7wQ4y9NgB+8iJfyLnUixkWM4G7CM9dkROsIMM\ny9mAfaTHjsgJdpBhORuwj5z4J3LusYOYWb9+/ZAhQ6KeIiuc5yDuqqurKyoqZs+ePWbMmKhn\noZ1yxg4AWocT/0ROsIMMy9kAiDuXYiFDjx0AcSfYQYYeu/iaOHFi1CO0IalU6r/+67+SeiPm\nfvvtF/UIe+LEP5ET7CAjlUrNnz+/rKysjd/C36NHj2effTbqKVrfpEmTnnvuuaingH3ixD+R\nE+wgw3I2SKQNGzbMnz+/sbExtCM+/vjjoR2ruLj4zDPPzMnJCe2ItHGCHWRYzgaJ9OMf//j7\n3/9+1FNk0XPPPXfSSSdFPQVthWAHQJLV19cHQXD99dd379496llaWU1NzW9+85u6urqoB6EN\nEeyA2Lv33ntLSkqinqKVvfjiiy1bFJKXl3fqqafefvvtrT5S5D772c8WFha2bN9TTjmlT58+\nrTtP5DZt2hT1CLQ5euwgw3I2AOLOGTvIsJwNgLgT7CBDjx0AcedSLGSkUqk5c+akUqmoBwGA\nFhLsIKO6urq8vLy6ujrqQQCghQQ7yNBjB0DcCXYAAAlh8QRJk0qlqqqqWtDYWVNTs/NrsxQV\nFY0ePbqNP2EWgPZAsCNpFixYMHbs2BbvXllZWVlZ2dy95s6dO3z48BYfFABahWBH0pSWls6a\nNasFZ+zS6fSSJUuGDBmSm9u8WxSKiopKS0ubezgAaHWCHUmTl5dXUVHRsn3Hjx/fusMAQJgs\nngAASAjBDgAgIVyKBUiabdu2bd68OYQDpdPpP/3pT8cdd1xz70xtGQ+Ggb0S7AASJTc3d+HC\nhSeddFLUg2TFkCFDoh4B2jTBDiBRbrvttldeeSWcY7355pvz5s0bPnz4oYceGsLhcnNzR44c\nGcKBIL4EOyDGcnJygiCYNGlSaAXR27dvz88P4zvnjsuOO/6AzfK1r32tZYdrQbN3TU3NvHnz\nhgwZMmzYsGbtqNYbskSwA2Ls5JNPLi8vD+3Wq7Vr177yyitHHXVUr169QjhcXl7eySefHMKB\ngn1r9lbrDW2HYAfE2Kc+9an//M//DO1wjz766HnnnXfdddeNGzcutIOGo2XN3mq9oa0R7ABo\nebO3Wm9oUwQ7oN1p2f1kQRDU1NTs/NosbikDwiHYAe3OvtxPFrilDGjDBDug3WnZ/WSBW8qA\nNk+wA9qdFt9PFrilDGjbPCsWACAhBDsAgIQQ7AAAEkKwAwBICMEOACAhBDsAgIQQ7AAAEkKw\nAwBICMEOACAhBDsAgITwSDGImXXr1rXgCfRt3xtvvBH1CACxJ9hBnPTu3Xvp0qX3339/1INk\nRb9+/aIeASDeBDuIkyeffHL16tXhHCuVSr3wwgsnnnhiXl5eOEfs2bNnOAcCSCrBDuKkY8eO\nhx56aDjHevLJJy+55JLZs2ePGTMmnCMCsI8sngB2b+vWrTu/AhALgh0AQEK4FAvJl0qlqqqq\n6urqmrVXTU3Nzq/NUlRUNHr06NDuzANgJ8EOkm/BggVjx45t2b6VlZUtaFeZO3fu8OHDW3ZE\nAFpMsIPkKy0tnTVrVnPP2KXT6SVLlgwZMiQ3t3n3bBQVFZWWljZrFwBahWAHyZeXl1dRUdGC\nHcePH9/qwwCQPfELdg2b/nfRC39Y/D/v9fnE4aNHfaY4N2eXDZb+7tFXPmi48MILIxkPACAq\nMQt2i35+1dlXTl3TkNrx2879S+7+XdVFR/b4+21+9/Uvf2vVJsEOAGhv4hTs3v3DjSd99c4g\nr9uEr182dND+//unp+/6ZdUlJ3yq4C9/Oe/AzlFPBwAQsTgFu/s+/9Mgt9ODr75x0ae6B0EQ\nXHrFVRfd/smyf/3yKZdWvPHwR6/JAgC0K3EqKL571fs9P317JtUFQRAEB5w6af53hm1eNf2c\ne5dHOBgAQFsQp2D3QSpdtN+Bu7x4wjefOqNX8byvj33tw+2RTAUA0EbEKdid3q3ovZd++EGq\n8e9fzMnr+uCT16Xq/nLGuXc0/rM9AQDagTgFu29OHFS3Yd6x59/43+9s+fvXe5dc/5uJg/9a\n/a8nT7pnU0q6AwDaqTgFu2Nuqj7/iB7/8+hNR/TresAhn3xi3dadb501deF1Ywa88NOv7r//\nJ+5dvWUP/xAAgKSKU7DL7dD74ZeW33vTlScf/cmGDbWbtv/fybnc/B7fn/XaQ9+99OC81Svr\n3GwHALRHcQp2QRDk5vf60g0//f1Lr63d+P4lH+/4D+/lFEy4/mevr978t/95dcGcqogGBACI\nTJx67Jomr+9hR/Q97IioxwAACFvMztgBAPDPJO2MXcPm5/sPPDcIgtra2qZsn0qlqqqq6urq\n9rDNqlWrgiBIp9OtMSAAQLYkLdg1NjasXr266dsvWLBg7NixTdly5cqVLR0KACAMSQt2BZ2P\nW7RoUdO3Ly0tnTVr1p7P2E2dOvWZZ5455JBD9nk6AIAsSlqwy8nrUlJS0vTt8/LyKioq9rxN\nVVVVEAS5ue5HBADatLgGuw21K5cvX7Fm/eYtH9blF3Xq2nP/wwYNPrRPt6jnAgCITMyCXWNq\n08zK7/z0vukvLFvz0Xf3HzT0gomTbpg0vlt+TvizAQBEK07BLtXw9heOP3La4nV5HXqUnD72\niMED+vTqVliYv72+fuPa1W+tWPrCwhdvu+b8h6Y/+WrNQwcUuHIKALQvcQp2NVefMW3xupOv\nuH3GLZf167SbydMN62bcevmEKdNHXDlx6T2nhT4gAECU4nRa67ppKzr3+erCO67abaoLgiC3\noOeFNzxyd8nH33jk+pBnAwCIXJyC3ZIt2zoftJcVrEEQHHtK720fLg1hHgCANiVOwe6snsUb\nlt2yumGPT4BIb71/5qqi7uVhDQUA0FbEKdh969by+k0LPz30vIeffmlLqnHXtxvrX1v4xMQR\ng+9etfm0KVOiGBAAIEpxWjxx2MWP/uKPIy+d+viEMx7LK+h66GEDDtivW2Fhh1RD/aa1tW+u\neGN93facnJzSy+6adfngqIcFAAhbnIJdEOROvHPeqAm/veuXM6oWLFr2+ssrlmbO2+XkFvYb\ncPiI0vLzJ1511vF9o50SACAS8Qp2QRAEfUvOvrnk7JuDoHH71o0b39+ytaGguGOXbt2LlRID\nAO1b/ILdTjn5xd17FXePegwAgDYiTosnAADYA8EOACAhBDsAgISI8T12AEDLzJkzZ9myZVFP\n0fpSqVTUI0RMsAOAdqRXr14DBw589913ly9fHvUsra9r167HHnts1FNESbADgHakY8eOiTxX\nxw7usQMAsiWVSs2ZM8cV0tAIdgBAtlRXV5eXl1dXV0c9SHsh2AEA2bJ169adXwmBYAcAkBCC\nHQBAQgh2AEC2FBcX7/xKCNSdAADZMmrUqKeffrqsrCzqQdoLwQ4AyJa8vLyRI0dGPUU74lIs\nAJAteuxCJtgBANmixy5kgh0AkC167EIm2AEAJIRgBwCQEIIdAJAteuxCpu4EAMgWPXYhE+wA\ngGzRYxcyl2IBgGzRYxcywQ4AyBY9diET7ACAbNFjFzLBDgAgIQQ7AICEEOwAgGzRYxcydScA\nQLbosQuZYAcAZIseu5C5FAsAZIseu5AJdgBAtuixC5lgBwBkix67kAl2AAAJIdgBACSEYAcA\nZIseu5CpOwEAskWPXcgEOwAgW/TYhcylWAAgW/TYhUywAwCyRY9dyAQ7ACBb9NiFTLADAEgI\nwQ4AICEEOwAgW/TYhUzdCQCQLXrsQibYAQDZoscuZC7FAgDZoscuZIIdAJAteuxCJtgBANmi\nxy5kgh0AQEIIdgAACSHYAQDZoscuZOpOAIBs0WMXMsEOAMgWPXYhcykWAMgWPXYhE+wAgGzR\nYxcywQ4AyBY9diET7AAAEkKwAwBICMEOAMgWPXYhU3cCAGSLHruQCXYAQLbosQuZS7EAQLbo\nsQuZYAcAZIseu5AJdgBAtuixC5l77ABIPrd50U44YwcAkBCCHQCQLXrsQuZSLACQLXrsQibY\nAQDZoscuZIIdAMk3Z86cPn36RD1FK3v00UdvuummqKfYi1QqNX/+/LKysry8vKhnaRcEu9hL\npVJVVVV1dXXN2iudTi9ZsmTIkCG5uc27z7KoqGj06NH++wSgKaqrqysqKmbPnj1mzJioZ2kX\nBLvYW7BgwdixY8M84ty5c4cPHx7mEQGIKT12IRPsYq+0tHTWrFnNPWNXU1NTWVk5efLkYcOG\nNWvHoqKi0tLSZu0CAIRDsIu9vLy8ioqKFuxYWVk5bNiwcePGtfpIAEAk9NgBANmixy5kzti1\nU/5LAyAEeuxCJti1U/5LAyAEeuxCJti1IVOnTr3tttsaGxujHiQr+vfvX1VVVVRUFPUgAIRH\nj13IBLs2ZO7cuW+++ebgwYOjHqT1rVmzZsGCBe++++5BBx0U9SwAhEePXcgEu7YlPz//17/+\nddRTtL4f//jHDzzwQNRTABA2PXYhsyoWACAhBDsAgIQQ7ACAbNGuFTL32AEA2aJdK2SCXduS\nTqeHDBkS9RQA0Dr02IUsrsFuQ+3K5ctXrFm/ecuHdflFnbr23P+wQYMP7dMt6rkAgP+jxy5k\nMQt2jalNMyu/89P7pr+wbM1H391/0NALJk66YdL4bvk54c8GAOxCj13I4hTsUg1vf+H4I6ct\nXpfXoUfJ6WOPGDygT69uhYX52+vrN65d/daKpS8sfPG2a85/aPqTr9Y8dECBdSEAEDE9diGL\nU7CrufqMaYvXnXzF7TNuuaxfp91Mnm5YN+PWyydMmT7iyolL7zkt9AFbQSqVinoEACCu4hTs\nrpu2onOfry6846p/tkFuQc8Lb3jkg6pnJz1yfXDPc2HO1irKysoWL17c3L0aGxu3bt3a3CfM\n1tfXb9y4sVu3boWFhc3aMScnp7i4OCen2Re7Dz744N69ezd3LwCg6eIU7JZs2dZ5UMVeNzv2\nlN7b/rQ0hHla3RVXXHHFFVc0d6958+aNGDGiZUfcuHFjC/aaO3fu8OHDW3ZEANoVPXYhi1Ow\nO6tn8SPLblndcMb+e7h/Lr31/pmrirqPCnGuiJWWls6aNauurq5Ze6XT6SVLlgwZMiQ3t3k3\nIxYVFZWWljZrFwDaLT12IYtTsPvWreUPXvL4p4ee95Mf/Ptnhx/TKe8frwY21r/2XNVtN06+\nb9Xm0XdOiWjGCOTl5VVU7P1E5keNHz++1YcBgL+nxy5kcQp2h1386C/+OPLSqY9POOOxvIKu\nhx424ID9uhUWdkg11G9aW/vmijfW123PyckpveyuWZcPjnpYAECPXdjiFOyCIHfinfNGTfjt\nXb+cUbVg0bLXX16xNLNiICe3sN+Aw0eUlp8/8aqzju8b7ZQAwA567EIWr2AXBEHQt+Tsm0vO\nvjkIGrdv3bjx/S1bGwqKO3bp1r1YKTEAtDF67EIWv2C3U05+cfdexd2jHgMAoI2IcbBrFalU\nqqqqas9LSletWhUEQTqdDmkmAFpbbW1tsxrg0+n0H/7wh23btjXrKOl0etWqVQcffHBzCweC\nIOjQocMJJ5zQrB03bNjQ3KO0WFP+d7lbNTU1O782S1FR0ejRo92Z11xJC3YNm5/vP/DcIAhq\na2ubsv2CBQvGjh3blC3/9re/7dNkAERhRzK4+OKLox4kW8KJPk3/3+VuVVZWVlZWNncvtakt\nkLRg19jYsHr16qZv35QSuKeeeurBBx+84IIL9nk6AMJ26aWXdujQobnn3hobG994443t27c3\nd6/33ntvv/32a8HjefLz8wcMGNDcHXv06DFs2LDmHqsFWtaZGqhNDV3Sgl1B5+MWLVrU9O2b\nUgL3zjvvPPjggx06dNi30QCIwCGHHPLd73436ilir8WdqYHa1HAlLdjl5HUpKSmJegoAgAjE\nNdhtqF25fPmKNes3b/mwLr+oU9ee+x82aPChfbpFPRcAQGRiFuwaU5tmVn7np/dNf2HZmo++\nu/+goRdMnHTDpPHddNoBAO1PnIJdquHtLxx/5LTF6/I69Cg5fewRgwf06dWtsDB/e339xrWr\n31qx9IWFL952zfkPTX/y1ZqHDiho9lJzAIBYi1Owq7n6jGmL1518xe0zbrmsX6fdTJ5uWDfj\n1ssnTJk+4sqJS+85LfQBAQCiFKfTWtdNW9G5z1cX3nHVblNdEAS5BT0vvOGRu0s+/sYj14c8\nGwBA5OIU7JZs2db5oL2vtT72lN7bPlwawjwAAG1KnILdWT2LNyy7ZXXDHh/tld56/8xVRd3L\nwxoKAKCtiFOw+9at5fWbFn566HkPP/3SllTjrm831r+28ImJIwbfvWrzaVOmRDEgAECU4rR4\n4rCLH/3FH0deOvXxCWc8llfQ9dDDBhywX7fCwg6phvpNa2vfXPHG+rrtOTk5pZfdNevywVEP\nCwAQtjgFuyDInXjnvFETfnvXL2dULVi07PWXVyzNnLfLyS3sN+DwEaXl50+86qzj+0Y7JQBA\nJOIV7IIgCPqWnH1zydk3B0Hj9q0bN76/ZWtDQXHHLt26FyslBgDat/gFu51y8ou79yruHvUY\nAABtRJwWTwAAsAeCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQsS47iRky5cvLyoqinqKVrNt\n27YHHnigf//+ubnCfUY6nf7LX/7yiU98wmeygw/ko3wmu/CB7MIH8lHpdPqtt9665JJLOnTo\nEPUsrWb58uVRj/BPCXZ7t+PfxS996UtRDwIAsXTPPfdEPULra5tRVbDbuwsvvHD79u1bt26N\nepDWtHjx4unTp5988sn9+/ePepa24q233nruued8Jjv5QD7KZ7ILhkbSKQAAFHNJREFUH8gu\nfCAfteMzueCCC4444oioZ2lNxcXFF154YdRT7E4j7dLMmTODIJg5c2bUg7QhPpNd+EA+ymey\nCx/ILnwgH+UzCZmbAAAAEkKwAwBICMEOACAhBDsAgIQQ7AAAEkKwAwBICMEOACAhBDsAgIQQ\n7AAAEkKwa6eKi4t3fmUHn8kufCAf5TPZhQ9kFz6Qj/KZhCynsbEx6hmIQCqVmj9/fllZWV5e\nXtSztBU+k134QD7KZ7ILH8gufCAf5TMJmWAHAJAQLsUCACSEYAcAkBCCHQBAQgh2AAAJIdgB\nACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBAQgh2AAAJIdgBACSEYAcAkBCCHQBA\nQgh2AAAJIdjFz9t/eOLKz589uP8BXYo7dOzSfdCxp171vZ/XNqT3ZeMDCvPzOvTI/uwt1Kw/\n8j8zb1T/nJz/196dx0VR/38Af8/eLNcCcoskiHhgipqgUXlBkOaZt6SERt+v4X0GdomKeOdZ\nqZl9tfInpalkXniEV5pHHmSEIIooolwuLOzu/P5YXRF22WVZDsfX8y9m5rPzec/M8uH12M8w\ny5wsKtO5WKO+jo3wYRgmuUBRm4OChrLH35lhmAyFip5cyuolPSw1ZreV3lHjXa0t7HrX4WHU\nROqXrzIMI5EF3S5TVd2qOQnzs4qqbjr/SUfNSfjwwn2drzLm1F1ZGahZDP/9js7yCjMXaRp4\nDUzWrmw8g5IJ9WcfDWMYpt2UM/VYJsBjgoYuAGpme2y/4Qv2EJGH3yvd/NoX3M64fjll1Z/H\nvvnyxwN/JQXKxCY3brTq8yjqra/CzFjPDmu6rj2bNMLbXPsEEzh0Dh5Q6qddzEn+9VSBwjek\nb2vp07HRWchviNLMT1GQEjIx6cr6t41/ydzVqZofdk49tPrwsIqbTDh1B6fvo1MRVXu5OO8b\n40tqQI2qfowhoBcLz48LawYTkW2L/rsv3dOuLH90a8UHHYmoif9skxu7ivg8gV1d12+CGh1F\n9Q6ENiOiE4UKnYs16uvo8BZEdDi/1OTjyk+fRkSvbb5u8h7AZMUZaampqeVqHZt2d3AiopiM\nAhN2W+kdNc7FSiLrVZs6zeja+m5ExBPwGEa4JvVhpa2a93PczcJK6wuzlhORzGemq4jPF7nc\nLVNV00U1p+7yigAisnSz4Antbymq7kQVZCsWSF2JqPmAw9q1jWdQMqH+20dCichv8um6qwpj\nCOiDqdjnRnnxnz0n/yyy6pByYUffdo7a9QKp+6S1JyNcre6fj195q9iExtVTK2o45Wk+ZjyK\nRtUXNCxLT29fX18B09B11LuAL2fxSTmr9weP1Kwx7c/FriOi7ismLA10VpXlTErJqU3vHRaG\nqcsfTE2pPJtZdHPp7wWKDnP71Wbn9aDu6m/AMRY4CcHuuXF58bgH5epuy35oa1llAp0RxSwa\nExoamnEuz4TGVWnunlGWXJ/ydhepVCLgSzx82oXPXFeoYolIfvcXBxHf1iuitMJotLaPJ48n\nXHZR7z5NUNOjYFUFWxdO7NbG08ZC7OTRInj0tP2pBXXUFxGx6vId8RM6NneViqQePu1Gz1ij\nOT8axZlHpoe/7evuKBEKrWydOr4xYOXPlzWb1vnYy7yWEtHxsS0Zhllz55GRRYJZ/NrVTXuP\nXY1Uc00NepS9v7NMIrTw3nY1v6b9mot920nbI3yLb/0YFn/WcGtWMTkxgye0/6KXe89FIUR0\neNqu2vTu6J/QTCw4MmNvpfUX521iGGHCAM/a7NwYx79bEBboZ2dtIbKwatH+tTmr91aMt2UF\nVz+PGtLS3VEssmzaolNU7Fe55c/ELfPWX/0Yq1HNgIYxBKrT0B8ZgrGmNbUmoqP5CsNNa9iY\nrTLroZmamdbFSWjlMyTiwxnR49raiYmoTWSSpsG5Jb2JqNfiPzWLtw9NJ6L20XtrckBmPgq1\nqvjDIBcism/ddfjYyP7B3cQ8hi9yXnLkjqZB9VOxNepLc34ih7QSWrUc+l70jOjINg4SImoz\nbp+mgfzeLy9JBAwj7Bw6KDLq/eEDe9kJeAzDm30ih2XZK99/szyuNxG1GPP5+vXrLz8qr9Fp\ngVpKCnQlohulyqqbqplPrP6astVOxcpzDgfYSQSSZpsv5tXNMRmgmYrtcypHWZoZYCPmCWS7\n78q1W3VOxd7/ayoRuQZtYVlWVZ7rIuLz+FZ/y/W+Vw1OxQ66ej8x2IMnsM145syrXrMVy7w+\nLrq1hOpyKvb0/DeJyMKp7bDwyHHhw3ztxUTUe+HjEUxReDrI0YJheH5dgyMiw3u87EREjq98\nUKo2sX6DU7EGx9jqBzSMIVANBLvnhqdEIBB71EVjVk+ws3DodfpeiWZNaX6Ks4gvtGz3uIVa\nEeUr4wntk3LlytKMQBuxpUu/vPLqbsExQY2O4mJ8EBF1mrJF8eTeqZzT/3MT80VW/nnlatZQ\nsKtRX5rzI7HveerJ+SkruughFoisO2sWT0/yI6LhW//WvuT+hSVE5P7G4+SH+2MakGnBzuA1\n1RfsSnKPBzWx4IvdN5y7b/6DMY422LEse+vAJCJq0mGm9tdVZ7DbEeJBRFF/3NUsbn3NjYjC\nEtP1dWFMsMs9P4GIBv2Wpd1UeHMJEfX83z91HOzUXhKByLqz9qIrCs/aC3kSu96axa96uhPR\nxB+vPGmv/GqENxGNOJBlWv1GBrtqxliDAxrGENAHU7HPCbb8pkLFF3uYv7F+PTdt7OIo0fws\ntu023sVSpbj1eBsjWnp4ozWb/27wp4lRIaeLadmRb+0FZn071fAoJsb/IbZ5NXnxaNGTe6ec\nu4zaPs63rPh8fKahCVmTzlj3DRsDnpwfodXLY52lKsVtzaJ78NzNmzevHtpC21jWaggRKXJL\natQFNB6mXVPFw9Nv+b2Z8oBWHT0f2dGhzqs0gnvvFct6ud+/kDBya5q+Nury3IlHs4UWPkv8\nH99v2ishhIhOzf6uNl07tIv3shAcm7lTu+ZS3EaGJ0zoX7fzsKxaflOh4gudtWOUyLrTmT/O\nphxcSkRK+dXoI9ky749WDm3z5BX88FWLAwMDlSnP3H1h9vqrGWNrNaDBiw2PO3lOMEIXIS+3\n7Lb5G+s3LNCx4mKl3GbpNuhg3BuvzE4YdoEC5iS/7yurZXeV1eQoyovPHc1XWLm23r55U8X1\n+ZY8IjpzNo+8qy3PpDM26lWniosS3tO78d37DB1DxKrkN65dT8/IyEj/9/jutTXaOTQ2JlxT\nVVl2P79eyXflRJRWoqyXMo0y4efEFU5BieNDzw+45m8prNog58SkbIXKa/AXVvzH72rHTotd\nRN/dTYs7VTQ70FpkWr8M32p5T/eB++aklf6nhYRPpI7Znm7bPKaTlbC4LrMKw7OM7+E2/fBe\nD9/Xxo7s/8ar3QK7dvFu76/ZWpy9RqFm24x+p+JLJA4DT54cWNf16xtjazugwYsNn9g9N8Ls\nJcrSzOOFlR+oq6HIPzRw4MCRkRtMaKyPg9DA26PDxPXWfB7DMLMmBxgq3xTGH4Wy5DoRFd/Z\nMO5Z0xMuE1FJtuHPyUw4Y24ivY83U8pTY8eEOEhtvNr4B/cdOnfpxnu23Q3WAI2ZCde0XH4t\nucBz86FVYh6zduDIe+WN5X8fRdYB+1f3VZb8O2Co7kHg5ykHiSg9MUz7qGG+yDGnTMWy5TO2\npdem66BFg9Sq4qkHbxNR0a0VR/MVnT4fXZsdGmnqb5c2xUW34qV9MW/m4NCgpnZW7XsN//Fc\nLhEpHmYSkU1rG2P2Y9769Y2xtR/Q4EWGYPfcGD/Ik4hit/6rc2vO8SU7d+48dsPFhMYm2/5+\n32KWL2Eoqk9cXfzJMv4o+CJ3InLp8ovOGw5OT/HTuQfT+tJi9D8vI6Zr0PwtB3pMXvL7xbRi\nhSI7/erebcsM1gCNmQnXlC9y2nb+xJieH/4U1aY0/0hY7PH6KdUYvpE7otvY30z675xjlR9i\noixJnXUpjy9yrZQqxox4hYgufL6qNv3atZ7fSio8MXsHEf01fwPDEy7u16w2OzQSI7CPiPni\n9PWc/Kxre77/evK7If8e/b9R3fyOF5YJbeyJSH5Tbsx+6qf+2g9o8CJDsHtu+M9fIeXzTk4f\nfLagyqdKrCJuQgoRhS18xYTGpsk5Pnf01jS/Cb/smeqfe3bBiC3/1GZvOhl/FCLboDZSYWH6\n5kr5Mu27+VOmTEnR8zmcaX0ZpJRfSbiUJ/NenLho8qsve0sFDBGpy3ONeS00TqZdU6G03Ts+\ntkT05vKkztaiC0v7/pRjVHqoF/yFB7604vOW9x9a6XvGMndNfKRSu/VY/fWzNm3Z11QsKM5e\nuyvPqO9Y04nhWSwLafogNfaavDz2hxu2XnP9rXTMBZtXad6uOXPmLEvMJCLbpq36DB+37Jvd\nxz7zV5Xdi7/ywMplHMMw6d/uq/iSsqKTfB7Pqf3WBqm/9gMavMgQ7J4bYllw0sc9yuXXerTr\nl/jH07vBlPLbiyICNmQV2bwUvraLswmNTaAqTevfJ0Fi3/vQ0pDuC/eHNrH4KSr4pLmHm5oc\nBW/de77y+z+FfvaLdigsurEnLOrTdZtOdzBi5DXnGWMEPIZRyv9RPnkilbo8d/WEQURE9Mxf\nULWysczNgQFGX1Od+OJmiVvD1ariqNB5Rj0auF5Yur2z76MuivzjY/dlVVy/4aMzRDRiWfdK\n7XkC+y+6uxHR50++Z8w0gQuHsqqS9zfMSM4v7Rw3sja7MhobHx//cXRs3tPfOPbM+QdE1M7Z\nQmT7+id+9g+uzorZrf3Ant0xJVLNsgGxXRuofmMHNIwhoENd/sstmJ3qq+iemgvn3jbgrX4D\nQrp3ayLiE5Gl++uH78lNbqzzcSd7H5RUbLPMS6Zts2Fwc4bhLTibq1nMPbeAxzCury9swENW\nKW4P9pURkWPLTkMiPggfEioT8Hh8q3nJ2ZoGBr9SzPi+dH6l2PyXbPkiV83PC4NciKh50Duz\nYj+ZOH50R2epS5fhHmKB0LLtghVfsixbmJVARLKWgz797JOUAmMfNwhmYdrjTgxeU4NfKRbT\n0ZGIxvx0w/yHZEjFx51UpFbm93ex1LznNY87KX14kM8wIuvOCl1fuZZ3eRoRWTQZWHWTMY87\nedJraTtLIcMwDE90vrhMs66un2O3oIcbEVm6dxg88r3/jh/bw8+ZiJy7TdV8s5z8blJbaxHD\n8Dt17zP+P5GhAU2JyN4vokipNq1+zeNObH16vlPF8FGTWSPGWIMDGsYQ0AfB7vmTemhLxODg\nl1wcJAK+1NquVeceE+M23NT1V8r4xjUKdrd+m0pEPuE/VNy6baQ3EUXWzR8tI49CqchaNSvC\n38vVQih0atayR/9xieeefuurEcHO2L4MBjtlaea8qP7NnWxEFnYvB/aamLBDoWYPxQySWQit\nXTqyLMuqSmKGdJVJhSKp3bd3H5nhHIHRTAt2Bq+pwWD36O5Oaz5PZNUhvUT3b2vd0RfsWJbN\nu7iEzzDaYHcpoQsRtX7/uO4dqUvbW4mIaF12caUtNQh2LLt/qDcR2fnM066p62CnKstdMyfS\nv2VTqYgvkFh6tesaPe+bvApfGPwo+9TM8L7Nne2EArGjp1/4jOV3nnwtrAn1a4KdTkJpG9aI\nYMcaGtAwhoA+DMs2npkBAAAAADAd7rEDAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLAD\nAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAA\nAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA\n4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACO\nQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADgBfOsRE+\njCFJD0vN0tfBME+GYU4WlZllbwAA1RM0dAEAAPXNoXPwgFI/7WJO8q+nChS+IX1bS58Oic5C\nfkOUBgBQKwzLsg1dAwBAQ9rj7/z2hXsxGQVxnjZm3/nBMM/gfTdPFCq6WovMvnMAgEowFQsA\nAADAEQh2AADVKS/+e1H0SD9PFwuh2MGl+VujphxJL6pRAwCAeoNgBwCgl1L+V0jLTrNXf18g\n8x4YPqKTt83+71cGt/Xb8m+hkQ0AAOoTgh0AgF67RvU7cudRyPx9WRdTtm3avD/lYuquGLUi\nK7rXdCMbAADUJwQ7AADdWFXB+D03Jfahe+e8qV3Z4u15K/0dCzO//iG3xGCDhqgaAF5oCHYA\nALrJc7c/VKqdu04TMM+sD4luSURb0woMNqi/WgEAiAjBDgBAH5Uik4isfSo/A8WmtQ0RFWfJ\nDTaojyoBACpAsAMA0I0v9iSion8q/4trcVoxEUndLAw2qI8qAQAqQLADANBN2mSITMC7d3K5\n6tn1h1b9TUTDWtoabFBflQIAPIZgBwCgGyOQfRXmUfJgb//FydqV6UmfTjhzz6bZuHedpAYb\nNETVAPBCw3fFAgDoNeD7Xa97d9s7s2fz7d3f6ORz/+8/9x39kxF7rk1eamQDAID6hE/sAAD0\nElq2P/jP2fkThknvXvlx4+aUK/eDh086ePWvUV42RjYAAKhPDMuyDV0DAAAAAJgBPrEDAAAA\n4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACO\nQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgE\nOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLAD\nAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACOQLADAAAA4AgEOwAAAACO+H+XLAvZN7Z6nwAA\nAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzde3yWdf348Wu7dwR0HBQBQQQkQRmaJ5h5mkNOObQU8QBf9St99ecJUcvykKUd\n1MrlOVMTpdAwtbC2AAnLw7AiFcQgUvFAgAdOAhsb973fHyMywHHvZrsP157PP/YYu697n/du\nab24r/v63FkNDQ0BAACZLzvVAwAA0DKEHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJ\nYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0A\nQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBASAg7AICQEHYAACEh\n7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMA\nCAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSE\nHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAA\nISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISw\nAwAICWEHABASwg4AICSEHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAg\nJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2\nAAAhIewAAEJC2AEAhISwAwAICWEHABASOakeIAOsW7fukUceqampSfUgAEBaKCwsPPfcc4uK\nilI9yPaE3a794he/mDRpUqqnAADSSE5OzsUXX5zqKbYn7Hatvr4+CIKHHnrokEMOSfUsAECK\nvfbaaxdccEFjHqQbYRevAw888PDDD0/1FABAitXW1qZ6hM/k4gkAgJAQdgAAISHsAABCQtgB\nAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABAS\nwg4AICSEHQBASAg7AICQEHYAACEh7AAAQiIn1QMAJFs0Gq2srKytrW3uHWOx2MKFC4uLi7Oz\nm/ev4oKCgtGjR0cikeauCNAswg5oc+bOnTtmzJgkLzp79uxhw4YleVGgrRF2QJtTWlo6Y8aM\nBJ6xq66urqiomDx5cklJSbPuWFBQUFpa2tzlAJpL2AFtTiQSKS8vT+y+FRUVJSUlY8eObdmR\nAFqEiycAAEJC2AHEq7CwcNtHgDTkVCxAvEaNGjVz5syysrJUDwKwc8IOIF6RSGT48OGpngLg\nMzkVCxCvaDQ6a9asaDSa6kEAdk7YAcSrqqpqxIgRVVVVqR4EYOeEHUC8ampqtn0ESEPCDgAg\nJIQdAEBICDuAeNnHDkhztjsBiJd97IA0J+wA4mUfOyDNORULEC/72AFpTtgBxMs+dkCaE3YA\n8bKPHZDmhB0AQEgIOwCAkBB2APGyjx2Q5mx3AhCvEO9jF41GKysra2trm3WvWCy2cOHC4uLi\n7OzmPU1QUFAwevToSCTSrHsBuyTsAOIV4n3s5s6dO2bMmGSuOHv27GHDhiVzRWgLhB2QwRoa\nGl5//fW6urrkLBeLxf785z8fddRRzX2CKjF5eXmDBg3KyspKwlqlpaUzZsxo7jN21dXVFRUV\nkydPLikpadYdCwoKSktLm3UXIB7CDshgv/rVr84444xUT9GKpk+fPnbs2CQsFIlEysvLE7hj\nRUVFSUlJcoYEdknYARls9erVQRCcdtpp++23XxKWi8Vi7733Xq9evZLwjN2777775JNPNv6A\nAHESdkDGGzVq1JAhQ1I9RQt7+eWXn3zyyVRPsQsuE4Z0I+wASFCILxOGDCXsAEhQiC8Thgxl\ng2IAEhSNRmfNmhWNRlM9CLCVsAMgQVVVVSNGjKiqqkr1IMBWwg6ABNXU1Gz7CKQDYQcAEBLC\nDgAgJIQdAAmyjx2kG9udAJAg+9hBuhF2ACTIPnaQbpyKBSBB9rGDdCPsAEiQfewg3Qg7ABJk\nHztIN8IOACAkhB0AQEgIOwASZB87SDe2OwEgQfaxg3Qj7ABIkH3sIN04FQtAguxjB+lG2AGQ\nIPvYQboRdgAkyD52kG6EHQBASAg7AICQEHYAJMg+dpBubHcCQILsYwfpRtgBkCD72EG6cSoW\ngATZxw7SjbADIEH2sYN0I+wASJB97CDdCDsAgJAQdgAAISHsAEiQfewg3djuBIAE2ccO0o2w\nAyBB9rGDdONULAAJso8dpBthB0CC7GMH6UbYAZAg+9hBuhF2AAAhIewAAEJC2AGQIPvYQbqx\n3QkACbKPHaQbYQdAguxjB+nGqVgAEmQfO0g3wg6ABNnHDtKNsAMgQfaxg3Qj7AAAQkLYAQCE\nhLADIEH2sYN0Y7sTABJkHztIN8IOgATZxw7SjVOxACTIPnaQboQdAAmyjx2kG2EHQILsYwfp\nRtgBAISEsAMACAlhB0CC7GMH6cZ2JwAkyD52kG6EHQAJso8dpBunYgFIkH3sIN0IOwASZB87\nSDfCDoAE2ccO0o2wAwAICWEHABASwg6ABNnHDtKN7U4ASJB97CDdCDsAEmQfO0g3TsUCkCD7\n2EG6EXYAJMg+dpBuhB0ACbKPHaQbr7EDMt7EiRNTPQJAWvCMHQBASAg7ABJkHztIN07FApAg\n+9hBuhF2ACTIPnaQboQdkPEefPDBIUOGpHqKFvbyyy+n/0Uh0Wh0zpw5ZWVlkUgk1bMAQeA1\ndgAkzD52kG6EHQAJso8dpBthBwAQEqF4jV1D3fwXnl+2Ym1RjwO+cPTgwuysVA8EAJACGfaM\n3Sdvzbl03Ih++3Xvsf+BX7r45pV1sc1rXj6pf9cjjht2+rjTTzr20L17HXX3c8tTPSZAm2Af\nO0g3mfSMXc2Hvx98cPmy2i1Z2YUd263+9X3f/MuSPa7dfNuctzYcP/a8kgH7rFz8t2lPPjtp\n+OBu7753erd2qZ4XIOTsYwfpJpOesZtxzv8tq91y3u0z1m/euPqTmuqply3/w+TLX1r5pUde\nfW76w9+/6ZaHp89a9vwdkeiaK8+tTPWwAOHXuI+dvU4gfWRS2N1WvapjvxsfnlzeIScrCLKH\njr/zjL3bZeX1fGz8oG3HdD/6su/37fjBvB+mcE6ANiIajc6aNSsajaZ6EGCrTAq7v9fUFw08\n9tNfObVLYW6Hw/L++2KJwX33qN+4KKmTAbRJ9rGDdJNJr7EbUJj71qI/BcF/Xsxx1DU3fGtN\nr+0Oe2PZJzmFByR3NIC2yD52kG4y6Rm7a0q6rl928yU/ezH276/0O2/S1yZ/+dPHfDT/ga//\nc23nQZOSPx4AQGplUtiNmfbgAYU5915wTFHPAcPP/N12t77x09suPGtEnyEX1WW1/97UL6Vk\nQgCAFMqksCvca8Sr/5hz1YRRe21ePv+VD7a7dUHFbT99fFZunyH3zFp0/gFFKZkQoE2xjx2k\nm0x6jV0QBO32PfaHjx77wyCI1ce2u+kLP5764j6fKzm0n/edAEgO+9hBusmwsNsmO3f75xp7\njRi1/WUUALSmxn3sUj0F8B+ZdCoWgLRiHztIN5n6jN1nqVv/Yu8DTw+CYMWKFfEcH41GKysr\na2trmzjmlVdeCYKgvr6+RSYECI2qqqry8vJnnnnm5JNPTvUsQBCEL+waGupWrlwZ//Fz584d\nM2ZMPEdOmzbthBNOSHAsgDCyjx2km7CFXV6HI+bNmxf/8aWlpTNmzGj6Gbt77733ueee69mz\n525PBwDQisIWdlmRPYYMGRL/8ZFIpLy8vOljKisrgyDIzvZ6RAAgrWVq2K1Z8faSJUtXrV6/\ncVNtTkH7oi7d+g8Y2Ld7x1TPBdCG2McO0k2GhV1DdN30im/f+dC0lxav2vHWbgOGnj1x0g2T\nxnXMsZkdQKuzjx2km0wKu2jd8vOPPGTqgo8juZ2HnDhm8MB+3ffqmJ+fs2Xz5rUfrXxn6aKX\nnn/59qvPenTab1+rfrRHnjOnAK3LPnaQbjIp7KqvGjl1wcfHXHrHY7dc3LP9TiaP1X382K2X\nTLhx2kmXTVx0/wlJHxCgbYlGo3PmzCkrK4tEIqmeBQiCzNqg+NqpSzt0v+j5uy7fadUFQZCd\n1+WcGx6/b8g+bz5+fZJnA2iDqqqqRowYUVVVlepBgK0yKewWbqzvsN8urmANguDw47rWb1qU\nhHkA2jj72EG6yaSwO6VL4ZrFt6ysizV1UKzmZ9OXFXQakayhAADSRSaF3XW3jti87vlBQ8/4\n+cz5G6MN29/csPmN55+eeNLA+5atP+HGG1MxIABAKmXSxRP9z33igb8Mv/DepyaMfDKSV9S3\nf78ee3fMz8+N1m1e99GKt5a+ubp2S1ZWVunF98y4ZGCqhwUIP/vYQbrJpLALguyJdz87asKv\n73n4scq58xb//ZWli7Y+b5eVnd+z38EnlY44a+Llpxy5b2qnBGgj7GMH6Sazwi4IgmDfIad+\nb8ip3wuChi01a9d+srGmLq+w3R4dOxXalBgguexjB+kmk15jt52snMJOe3Xt2atn1706qzqA\n5ItGo7NmzYpGo6keBNgqg8MOgNSyjx2kG2EHQILsYwfpRtgBAISEsAMACAlhB0CC7GMH6Sbz\ntjsBIE3Yxw7SjbADIEH2sYN041QsAAmyjx2kG2EHQILsYwfpRtgBkCD72EG68Ro7gFDZvHnz\npk2bkrPWxo0bGz+uWbMmOSvuscceOTn+nws+k/95AITKwIED33777WSueP755ydtreHDh8+c\nOTNpy0HGEXYAobJ8+fIePXocffTRyVlu1apV++yzT3LWmjVr1vLly5OzFmQoYQcQNgMGDLjx\nxhtTPUXLe/XVV1M9AqQ7F08AAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhYR87\nIONVVVUtWrQoCQvFYrH33nuvV69e2dmt/q/id999t7WXAMJH2AEZrGPHjkEQPPnkk6kepLU0\n/oAAcRJ2QAYbO3bsgAEDtmzZkpzlnn/++cmTJ1dUVBx77LFJWC4nJ6e4uDgJCwGhIeyADJad\nnX3IIYckbblDDz30oIMOKisri0QiSVsUIH7CDiBekUhk+PDhqZ4C4DO5KhYgXtFodNasWdFo\nNNWDAOycsAOIV1VV1YgRI6qqqlI9CMDOCTuAeNXU1Gz7CJCGhB0AQEi4eAJoc6LRaGVlZW1t\nbXPvWF1dve1jsxQUFIwePdq1tEBrE3ZAmzN37twxY8YkfPeKioqKiorm3mv27NnDhg1LeFGA\neAg7oM0pLS2dMWNGAs/YxWKxhQsXFhcXN/ctxQoKCkpLS5u7HEBzCTugzYlEIuXl5Yndd9y4\ncS07DEALcvEEAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgI\nOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBASAg7AICQEHYAACEh7AAA\nQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlh\nBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBA\nSAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHs\nAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAI\nCWEHABASwg4AICSEHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQd\nAEBIxBt2p1947dN/WhRt1VkAANgN8Ybdkz/9/pePH9Sx1yFfuea2uQuWt+pMAAAkIN6we+OF\nGd+89Ox965Y+eNs1Jx7Ss/ug47/6/Z++8u76Vh0OAID4xRt2A79Q/u27frF45dq/PfvE1yae\nVvD+vB9ee+Hh+3c+6JhTvnvf9LfW1rXqlAAA7FIzL57Iyvt82em3PvCrt1avrv7d1MvHj1z7\nSuX1F4/rv1eno08+957HZn1UF2udOQEA2IXEr4rNjuQUFLbfo31uEASx6KaXK6deevaIHp16\nX3jbzJYbDwCAeDUv7KKbP/zDkw9ees4XexV1GjLyrFt/On1D18Mv+2bFc6+998nyBQ/ccvXg\nPVf/9JqR/zfH1RUAAMmWE+dxv5ny46effvqZ37+4ui6alZXV+9ATrzrttNNPP23ogV3/fUjP\nidf8YMKF5QWdjv/tja8FZfu20sQAAOxUvGF36vmTs7Iinxsy/KLTTjvttC8f1qfTzr9dYZ9u\n3brtf2iXlpsQAIC4xBt237pj6mmnnTpo3w5NHxbJ77VixYrdngoAgGaL9zV2J39hYH6HvJ3e\ntPGdRa+8trTlRgIAIBHxht0RRxxx0XP/2ulNSx4Yf+RRx7fcSAAAJGIXp2Kn3HPXui1bt6Z7\n75mH71jWefsjGra8+PjbQZDfGsMBABC/XYTdzVdf+VbtlsbPlz500xWfcdj+o3/aolMBANBs\nuwi7qZUza2INQRAMGzbs89/++Q++0G0n36JdlyFDDm2V6QAAiNsuwu7o0hMbPxk5cuShJw0r\nK9mn9UcCACAR8W53UlVV1apzxK9u3bvzXvrzgn982P2Ag0ePOrYwO2u7Axb95olXN9Sdc845\nKRkPACBVmgq7devWBUHQfs+inKytnzehqKioJef6DPN+evmpl927qi7a+McOvYfc95vK8Yf8\n1yUdv7niK9ctWyfsAIC2pqmw69ixYxAET3606ctdChs/b0JDQ0NLzrUzH/z5W1+46O4g0nHC\nFRcPHdDt3b/OvOfhyvOOOijvn/88o9cudk4GAAi9psLuzDPPDIKgZ15OEATjx49P0kSf7aH/\nuTPIbv/Ia2+OP6hTEATBhZdePv6Oz5Vd+ZXjLix/8+c7npMFAGhTmgq7xx57bNvnU6dObf1h\nduG+ZZ90GXT/1qoLgiAIehw/ac63nxh6/bTTHryh8v8GpHA2AICUi/edJw467ss/+NmMlTXR\nVp2maRuisYK9e233xaO+/ruRexU+e8WYNzZtSclUAABpIt6wW/zCr792wSk9O3X/4v9c+cQf\nFqSk707sWPDh/Ns2RP/rxXxZkaJHfntttPafI0+/q9Vf5QcAkMbiDbuPl/75vu9efeyBBZVT\nK84oO6RT78Mv+eadf3lrbasOt52vTxxQu+bZw8/61uv/2vjpr3cdcv2vJg58r+rKYybdvy6q\n7gCANiresOvU74iLrv3B3NfeXfX3l+781qTBHf51782ThhzQ5eAknqI97KaqswZ3/scTNw3u\nWdSjz+ee/rhm202n3Pv8tSf3e+nOi7p1O+DBlRub+CYAAGEVb9ht03VAyWU3/viFRSuWL/zT\nj677f3user7xFG1rDLed7NyuP5+/5MGbLjvm85+rW7Ni3Zb/PDmXndP5uzPeePTmC/ePrHy7\n1ovtAIC2qNlh92+xjZtq6urrG3evi27+sOVGakp2zl4X3HDnn+a/8dHaT87bp91/3ZaVN+H6\nn/x95fr3//Ha3FmVyZkHACB9xPuWYls11C34U9VTTz351FO/Xvj+J0EQ7NHrkP+9+sozx41r\nlekSEdm3/+B9+w9O9RgAAMkWb9i9/PvHn3rqyaee/u0/P6oNgqBwnwHjLxs37swzRx89IOEn\n/dJBNBqtrKysra1t4phly5YFQRCLxZI0EwBAQuINu6GjzgqCIL9Tn7H/N27cmWeOOeGQ3LR8\no4e69S/2PvD0IAhWrFgRz/Fz584dM2ZMPEe+/fbbuzUZAEArizfsxpw7edyZZ35p+JFp/s5d\nDQ11K1eujP/40tLSGTNmNP2M3b333vvcc8/16dNnt6cDAGhF8Ybdb6bc3qpztJS8DkfMmzcv\n/uMjkUh5eXnTx1RWVgZBkJ2d0eecAYDwayrs1q1bFwRB+z2LcrK2ft6EoqKilpwrUVmRPYYM\nGZLqKQAAUqCpsOvYsWMQBE9+tOnLXQobP29CQ0NS3/JhzYq3lyxZumr1+o2banMK2hd16dZ/\nwMC+3XcxJABAiDUVdmeeeWYQBD3zcoIgGD9+fJImalJDdN30im/f+dC0lxav2vHWbgOGnj1x\n0g2TxnXMSesXAgIAtIamwu6xxx7b9vnUqVNbf5hdiNYtP//IQ6Yu+DiS23nIiWMGD+zXfa+O\n+fk5WzZvXvvRyneWLnrp+Zdvv/qsR6f99rXqR3vkeUkcANC2xHvxxPz58/c8oLh/Ud6ON218\nZ9E/1uZ9/pD+LTrYTlRfNXLqgo+PufSOx265uGf7nUweq/v4sVsvmXDjtJMum7jo/hNaex4A\ngLQS79NaRxxxxEXP/WunNy15YPyRRx3fciN9pmunLu3Q/aLn77p8p1UXBEF2Xpdzbnj8viH7\nvPn49UmYBwAgreziGbsp99y1bsvWd1x475mH71jWefsjGra8+PjbQZDfGsNtZ+HG+g4DdrE1\nSRAEhx/Xtf6vi5IwDwBAWtlF2N189ZVv1W5p/HzpQzdd8RmH7T/6py061c6d0qXw8cW3rKwb\n2a2J18/Fan42fVlBp1FJmAcAIK3sIuymVs6siTUEQTBs2LDPf/vnP/hCt518i3Zdhgw5tFWm\n+2/X3TrikfOeGjT0jB9//xtfGnZY+8h/X/rasPmNFypv/9bkh5atH333jUmYBwAgrewi7I4u\nPbHxk5EjRx560rCykn1af6TP1P/cJx74y/AL731qwsgnI3lFffv367F3x/z83Gjd5nUfrXhr\n6Zura7dkZWWVXnzPjEsGpnBOAICUiPeq2KqqqladIz7ZE+9+dtSEX9/z8GOVc+ct/vsrSxdt\n3RU5Kzu/Z7+DTyodcdbEy085ct/UTgkAkBKZ95Zi+w459XtDTv1eEDRsqVm79pONNXV5he32\n6Nip0KbEAEDblqlvKRYEQVZOYae9CjsleVUAgHSVYW8pBgDAZ8mktxQDAKAJib+hau2HC2f8\n8rHn/rpkS7LPwQIAsBPxh13Dr75/0dDifg+s3BgEwSfvPHrgfoedcubZpUcO6HvC5WvEHQBA\nqsUbdkseOGXstff/9R+rC7OzgiD4SfmV79fnX/7diq9OOOy9P91VfvvrrTkkAAC7Fm/Yff+G\nP+S1H/zXVavGd20X3bzsW2+s6Tl86h3XXnHbo389u2u7VysqWnVKAAB2Kd6we/rjmr0Ou+XQ\njnlBEKx/5/ZN0dhR15cEQRAEWecftlfNx79ptQkBAIhLvGGXn5UV/Pt1dG8+9MesrKwrizs3\n/jG6pSFo2NIawwEAEL94w+5/urX/6LVvvrM52hBdf+ODS9t1nVCyR14QBLG6f1338qr8jmWt\nOSQAALsWb9hd+uNT6j7560F9iocc3Ltydc1R3/haEATv/+4H5UcOnv9J3cALvtGaQwIAsGtN\nbVD8aft/+dE5d7a/+NbH579Zf8TY63596UFBEPzr2UcrF3x80KgrZ958eGsOCeyWaDRaWVlZ\nW1vbrHvFYrGFCxcWFxdnZzdvw8uCgoLRo0dHIpFm3QuA3Rdv2AVBcOJl9y2+7L76hiA3a+tX\nDvzKT/560QGHH7hPq4wGtJC5c+eOGTMmmSvOnj172LBhyVwRgKBZYddoW9UFQVB00Bc8Uwfp\nr7S0dMaMGc19xq66urqiomLy5MklJSXNumNBQUFpaWmz7gJAi4g37BqiGx74+v/e8cTct1Zt\n2OkBNTU1LTcV0JIikUh5eXkCd6yoqCgpKRk7dmyLjwRAa4g37F64+tgLf/xqJL/rYUeVFOV7\n6QwAQNqJN+yu+dnf8zoc+uJb1UfsXdCqAwFporCwcNtHADJCXGHXEKv5yyd1fc6+U9VB2zFq\n1KiZM2eWldmlEiBjxBd20Y0NQdAQi7X2NED6iEQiw4cPT/UUADRDXNtTZefuddNRXd+dcfnr\nG+pbeyAgTUSj0VmzZkWj0VQPAkC84n2N3df+MPcfw8qGDjzxm9++/AuDB3brtP3Lbvr169fS\nswGpVFVVVV5e/swzz5x88smpngWAuMQbdrntDwqCIAhWXHPBCzs9oKGhoYVGAtJC4x5GdjIC\nyCDxht2ll17aqnMAALCb4g27u+66q1XnAABgNzXvLcViW1a/NHvugn8sW7eh5hvXXb9x2TuF\n+/du3tuDAxnCPnYAGacZVbZi7r1De/U6dvTpl1xx9bXX3xAEwavfHtG5z5F3znq31cYDUqZx\nH7tRo0alehAA4hVv2G14/5efH3n5/I/yzr7i+u9e2XghRbDv6NM6f/Da5C8WP/z2+labEEiN\nxn3sIhFvIQiQMeINu+njrvgwWvDIgrd/UXHzhOH7Nn5x/7Hffe31X+0ZbLj27OmtNiGQGvax\nA8g48Ybdra983PngO8YP7Ljd1/foM+buQXt9vOBHLT0YkGJVVVUjRoyoqqpK9SAAxCvesFtV\nH23fc/+d3tR9v3bRun+12ERAerCPHUDGiTfsRnYq+Gj+Izvbgzg25eUP84uOb8mhAABovnjD\n7torP79x1dRh1/xsY+xTdddQ//S3Rk1dtfFz/3tdq0wHAEDc4t3Hrvirv7v0NwfefdsFXafe\nesT+a4Ig+Mr557z+wu/m/XNdUf+xv/3OEa05JJAC9rEDyDjxPmOXFSm684V/Trn5kn45H/yp\n+sMgCB6cMu3VNZ3OvvJHb7z+eM88GyJA2NjHDiDjNOOdJ7IiHc69/u5zr7979b/eWbV6Q/6e\nnfffr7u3nYCwatzHLtVTANAMiYRZ5x69Bw46uK+qg1Czjx1AxomrzdYsW7Jk1X+2PNiyadld\nN1x11pfP+NrNP5796oetNhuQSvaxA8g4uwi72g//cuGwgZ37DBg/ZWnjV3+HFZoAACAASURB\nVLbULB62/0GXf+f2x59+4gffnDzi8F7n3vli688JJJt97AAyTlNhF6tbfuqgE346Z/H+hx47\nvLhT4xfnXFD+xw9repxw8e+ee2n6gz8c0iV76hUnPPDOJ0mZFgCAz9TUxROL7vjyzA82jan4\n42+uOK7xKw2xjf/39LKcwgP+WHXHAQU5wfElo0fuV9Rr3Hcv/eNXnjk5KQMD0JRoNPqHP/yh\nuLg41YO0ir333jvVI0Baa+oZu5/++I38ouOe+nfVBUGwYfmd79Zu6TnsjgMKthZh+33Hnt21\n3QfVD7TumEDS2ccOIOM0FXaz127ec/9LP71D3bLHnw6C4As3/Nd2xCV75NWtr26V6YDUsY8d\nQMZpKuze3xzN67Tnp78y64F/ZmVFvn5w509/sSA7q6FhS6tMB6RO4z52kYjtxwEyRlNhV9w+\nd8Oyv2z7Y6z+w1uXrS/ce9ygdv/1yrxn19bmdTi0tQYEUsQ+dgAZp6mLJy4e1Pnc6m9Pf++K\nM3p1CILg3Wcu+bA+2v+kCz99zIblP//lhzXdjrukdccEkq6qqqq8vPyZZ545+WSXRmWSSCRy\n/PHH33HHHakepOV96Utfys/PT/UUkNaaCruTH5iUNegb/zP4+MXf/H89c9777jVPB0Hwle/8\n58m51a9Xjh/5/6INDZfdXdrqkwLJZR87gIzTVNh1Ouia39/81y/e8OSNV36l8SsHn/PgV/ff\nMwiChuiGQwYetOTN9+tiDcdNnn7NoM5NfB8AAJKgqbALguCk6554/4vPPvKbuSs35Qw4ctTE\n04f++5boorc+2L/4xP/39RuvPvPY1p4SAIBd2kXYBUHQ9dBhXz102HZfzIoU1W+pjeuNZoHM\nZB87gIyz67D7LKoOwq1xH7uysrJUDwJAvBIPOyDcGvexS/UUADSD592AnbOPHUDGEXbAzlVV\nVY0YMaKqqirVgwAQL2EH7Jx97AAyjrADAAgJYQcAEBLCDtg5+9gBZBzbnQA7Zx87gIwj7ICd\ns48dQMYRdpBJKioqKisrk7NWQ0PD6tWrO3funJWVlZwVDznkkB/+8IfJWQsglIQdZJKHHnpo\n8eLF7du3T8Ja0Wh048aN7du3j0QiSVhu06ZNf/rTn4QdwO4QdpBh+vTp8/TTTydhoVgsNm/e\nvKFDh2ZnJ+Mqq0mTJr3wwgtJWAggxIQdsHPZ2dlHH310qqcAoBlsdwIAEBLCDgAgJIQdAEBI\nCDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewA\nAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJ\nYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0A\nQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBASAg7AICQEHYAACEh\n7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMA\nCAlhBwAQEsIOACAkhB0AQEgIOwCAkMhJ9QAAtLBXXnnlK1/5SnLWWrduXVFRUXLWWr58ed++\nfZOzFmQoYQcQKscdd9z8+fOXLFmShLXq6+s3bNjQoUOH3NzcJCzXrl274447LgkLQeYSdgCh\nMnv27KStFY1G58yZU1ZWFolEkrYo0IRMCru1K1dsjMbiPHjfffdt1WEAiEQiw4cPT/UUwH9k\nUth99fOfe3DlhjgPbmhoaNVhAPCMHaSbTAq77zxbdeCUe75Z8cuaaEOn4hO+0LtDqicCaNOq\nqqrKy8ufeeaZk08+OdWzAEGQWWG3z8HHXP2DY0o7v3XEtX8eeMl9z1w4INUTAbRpNTU12z4C\n6SDz9rErvuRHqR4BACAdZV7Y5e15zGE9uxUVeD0HAMB/yaRTsdvMf29FqkcAICgsLNz2EUgH\nGRl2AKSDUaNGzZw5s6ysLNWDAFsJOwASZB87SDeZ9xo7ANJENBqdNWtWNBpN9SDAVmF7xq5u\n/Yu9Dzw9CIIVK+J6HV40Gq2srKytrW3imGXLlgVBEIvF+6YXAG2Efewg3YQt7Boa6lauXBn/\n8XPnzh0zZkw8R7799tuJDgUQTvaxg3QTtrDL63DEvHnz4j++tLR0xowZTT9jd++99z733HN9\n+vTZ7ekAAFpR2MIuK7LHkCFD4j8+EomUl5c3fUxlZWUQBNnZXo8IAKS1TA27NSveXrJk6arV\n6zduqs0paF/UpVv/AQP7du+Y6rkA2hD72EG6ybCwa4ium17x7TsfmvbS4lU73tptwNCzJ066\nYdK4jjlZyZ8NoK2xjx2km0wKu2jd8vOPPGTqgo8juZ2HnDhm8MB+3ffqmJ+fs2Xz5rUfrXxn\n6aKXnn/59qvPenTab1+rfrRHnjOnAK3LPnaQbjIp7KqvGjl1wcfHXHrHY7dc3LP9TiaP1X38\n2K2XTLhx2kmXTVx0/wlJHxCSYfXq1cXFxameolVEIt4DOsNEo9E5c+aUlZX5bwdpIpOe1rp2\n6tIO3S96/q7Ld1p1QRBk53U554bH7xuyz5uPX5/k2QDaoKqqqhEjRlRVVaV6EGCrTAq7hRvr\nO+y3iytYgyA4/Liu9ZsWJWEegDbOPnaQbjIp7E7pUrhm8S0r65p8B4hYzc+mLyvoNCJZQwEA\npItMeo3ddbeOeOS8pwYNPePH3//Gl4Yd1j7y35e+Nmx+44XK2781+aFl60fffWOKZgTISPG8\nv+KOqqurt31sloKCgtGjR3tlHrS4TAq7/uc+8cBfhl9471MTRj4ZySvq279fj7075ufnRus2\nr/toxVtL31xduyUrK6v04ntmXDIw1cNCa+ncufMf//jHVE/R8iZNmvTCCy+keoq2K/73V9xR\nRUVFRUVFc+81e/bsYcOGJbYi8FkyKeyCIHvi3c+OmvDrex5+rHLuvMV/f2XpoobGG7Ky83v2\nO/ik0hFnTbz8lCP3Te2UABknnvdX3FEsFlu4cGFxcXFz35unoKCgtLS0WXcB4pFZYRcEQbDv\nkFO/N+TU7wVBw5aatWs/2VhTl1fYbo+OnQptSgyQqHjeX3Gnxo0b1+LDAAnLvLDbJiunsNNe\nhZ1SPQYAQJrIpKtiAQBogrADAAgJYQcAEBLCDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgB\nAISEsAMACAlhBwAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABAS\nwg4AICSEHQBASAg7AICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLCDgAgJIQdAEBICDsA\ngJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEjkpHoAoHk2bNgwa9asJCwU\ni8WWLl3av3//7Oxk/Avwgw8+SMIqAOEm7CCTFBYWrly58qqrrkr1IK2iqKgo1SMAZDZhB5lk\n2rRpr776anLWisViCxcuLC4uTs4zdkEQHHDAAclZCCCshB1kkv79+/fv3z9py40bNy5pawGw\n+1w8AexcNBqdNWtWNBpN9SAAxEvYATtXVVU1YsSIqqqqVA8CQLyEHbBzNTU12z4CkBGEHQBA\nSLh4AsIvGo1WVlbW1tY2617V1dXbPjZLQUHB6NGjI5FIc+8IwG4SdhB+c+fOHTNmTGL3raio\nqKioaO69Zs+ePWzYsMRWBCBhwg7Cr7S0dMaMGc19xi7hfewKCgpKS0ubdRcAWoSwg/CLRCLl\n5eUJ3NE+dgCZxcUTAAAhIewAAEJC2AEAhISwAwAICWEHABASwg4AICSEHQBASNjHLuMl9m5R\nu7P3rHeLAoD0JOwy3u68W1RivFsUAKQnYZfxEnu3qOrq6oqKismTJ5eUlDTrjt4tCgDSlrDL\neAm/W1RFRUVJScnYsWNbfCQAICVcPAEAEBLCro0qLCzc9hEACAenYtuoUaNGzZw5s6ysLNWD\nAAAtRti1UZFIZPjw4ameAgBoSU7FtlHRaHTWrFnRaDTVgwAALUbYtVFVVVUjRoyoqqpK9SAA\nQIsRdm1UTU3Nto8AQDgIOwCAkBB2AAAhIezaKPvYAUD42O6kjbKPHQCEj7Bro+xjBwDh41Rs\nG2UfOwAIH2HXRtnHDgDCR9i1UfaxA4DwEXYAACHh4ok08pOf/OTBBx9Mzlrr1q0LguDaa6+9\n9dZbk7Nily5dZsyYkZ+fn5zlAKANEnZpZObMmX/7298GDhyYhLXy8vJ69+6dl5eXnLOxq1at\nmj9//qpVq/bbb7/WXisajVZWVtbW1jb3jrFYbOHChcXFxdnZzXsmu6CgYPTo0ZFIpLkrAkDL\nEnbpJScn55e//GWqp2h5P/rRj6ZMmZKctebOnTtmzJjkrLXN7Nmzhw0bluRFAWA7wo6wKS0t\nnTFjRgLP2FVXV1dUVEyePLmkpKRZdywoKCgtLW3ucgDQ4oQdYROJRMrLyxO7b0VFRUlJydix\nY1t2JABIDlfFAgCEhGfsYKvCwsJtHwH4NNelZQphB1uNGjVq5syZZWVlqR4EIO24Li1TCDvY\nKhKJDB8+PNVTAKQj16VlCmEHW0Wj0Tlz5pSVlXnmH2A7rkvLFC6egK2qqqpGjBhRVVWV6kEA\nIEHCDrZqfBOO5LwVB0Ab4bq0JHMqFgBoLa5LSzJhBwBtS21tbTLPThx55JHr169Pzlq5ubkd\nOnRIzlrpSdjBVs4XAG3B+vXr999//zVr1qR6kNYyefLk22+/PdVTpIywg62cLwDagrVr165Z\ns+Zzn/vc4MGDUz1Ly3vqqafefvvtVE+RSsIOtrKPHdB2HH300VdddVWqp2h5v/nNb1I9Qoq5\nKha2ikajs2bNikajqR4EABLkGTvS17vvvltaWpq0F4LU19dv2LChQ4cOubm5SVguNzf3rrvu\nOuOMM5KwFgBthLAjfS1duvStt97q27dv165dk7PiunXrioqKkrBQXV3d3/72t7/85S/CDoAW\nJOxId+PHjw/fG9GsWLHC6/kAaHFeYwcAEBKesUsvsVisuLg41VMAABnJM3YAACEh7AAAQkLY\nAQCEhLADAAgJF0+kl+zs7FdffTXVU7S8H/3oR1OmTEn1FADxikajlZWVtbW1zbpXLBZbuHBh\ncXFxdnaznzcpKCgYPXp0JBJp7h3h04QdAGxv7ty5Y8aMSfKis2fPHjZsWJIXJWSEHQBsr7S0\ndMaMGc19xq66urqiomLy5MklJSXNXbGgoKC0tLS594LtCDsA2F4kEikvL0/gjhUVFSUlJeF7\nvxwyhbBLLw0NDW+88UYSFtqdF4Ik4KOPPkrCKgDQxgm7NJKTk7Nly5Zx48alepDWkpPj7xsQ\nZoWFhds+Qkr4P9o08q1vfeuII45IzloLFy78xS9+cc455yTtHcy6d+/eo0eP5KwFkBKjRo2a\nOXNmWVlZqgeh7RJ2aeTggw8++OCDk7PWE0888Ytf/OKUU07xQhCAlhKJRIYPH57qKWjThF3G\nS2yzperq6m0fm8VOSwCfJRqNzpkzp6ysLP1/SU6ZMiWU24um/yPf2oRdxtudzZYqKioqKiqa\ney87LQHsVFVVVXl5+TPPPHPyySenehbaKGGX8RLbbCnhq2LttATwWWpqarZ9hJQQdhkv4c2W\nQnz5LQC0TcKOdHfTTTfddNNNqZ4CyFRbtmxZsGBBQ0NDEtZ66623Gj/Onz8/CcsFQdChQ4cD\nDzwwOWuREYQdAGF22223XXfddclc8etf/3oyl5s/f/5hhx3W3Hudd955V111VWvMk1oJPBQh\nI+wACLM1a9YEQXDBBRfsueeerb1WLBZ77733evXqlZw39VmwYMGcOXMaf0BoJOwACL9x48Z1\n79491VO0sCeeeGLOnDmpnoL0kox/UgAAkASesSPdffOb3wzf22OsWLHC9vQAtDjP2AEAhISw\nAwAIiUw9FbtmxdtLlixdtXr9xk21OQXti7p06z9gYN/uHVM9FwBAymRY2DVE102v+PadD017\nafGqHW/tNmDo2RMn3TBpXMecrOTPBgCQWpkUdtG65ecfecjUBR9HcjsPOXHM4IH9uu/VMT8/\nZ8vmzWs/WvnO0kUvPf/y7Vef9ei0375W/WiPPGeZAYC2JZPCrvqqkVMXfHzMpXc8dsvFPdvv\nZPJY3ceP3XrJhBunnXTZxEX3n5D0AWkVP//5z2fNmpWctdatW1dUVJSEherq6pKwCgBtTSaF\n3bVTl3boftHzd13+WQdk53U554bHN1T+cdLj1wf3v5DM2WgNffr06dGjx5o1a5Kzr3p9ff2G\nDRs6dOiQm5ubhOU6deo0ePDgJCwEQNuRSWG3cGN9hwHluzzs8OO61v91URLmobX17dt3+fLl\nSVvuiSeeOOOMM372s5+Fb9s8ANqITAq7U7oUPr74lpV1I7s18fq5WM3Ppi8r6DQqiXORXqLR\naGVlZW1tbXPvWF1dve1jsxQUFIwePToSiTT3jgDQsjIp7K67dcQj5z01aOgZP/7+N7407LD2\nkf++9LVh8xsvVN7+rckPLVs/+u4bUzQjqTd37twxY8YkfPeKioqKiorm3mv27NnDhg1LeFGg\ntXmvF9qITAq7/uc+8cBfhl9471MTRj4ZySvq279fj7075ufnRus2r/toxVtL31xduyUrK6v0\n4ntmXDIw1cOSMqWlpTNmzEjgGbtYLLZw4cLi4uLs7OZdUl1QUFBaWtrc5QDaglgsNm/evKFD\nhzb3VyuJyaSwC4LsiXc/O2rCr+95+LHKufMW//2VpYsaGm/Iys7v2e/gk0pHnDXx8lOO3De1\nU5JakUikvHzXr8XcqXHjxrXsMADpJisrKwiCKVOmTJkyJdWz0PIyK+yCIAj2HXLq94ac+r0g\naNhSs3btJxtr6vIK2+3RsVOhTYkBYFd69OhxySWXfPDBB8lZ7v3336+uri4pKenZs2dyVhw/\nfnxyFkpPmRd222TlFHbaq7BTqscAgAwSiUTuvvvupC33xBNPVFdXT5482YYDyZHBYdci4rmC\nctmyZUEQxGKxJM0EQEt75JFHunbtGv/xsVjsz3/+c319fbNWicViy5Yt23///RN4PVlubu5R\nRx3VrDtWVlbeddddzV0oMTYcyBRhC7u69S/2PvD0IAhWrFgRz/HxX0H5/vvv79ZkAKRCYxmc\ne+65qR6ktSQnfWw4kCnCFnYNDXUrV66M//h4rqD83e9+98gjj5x99tm7PR0AyXbhhRfm5uY2\n97m3hoaGN998c8uWLc2914cffrj33ns3XqDQLDk5Of369WvuHTt37lxSUtLctRJgw4FMEbaw\ny+twxLx58+I/Pp4rKP/1r3898sgjyXmbKQBaVp8+fW6++eZUT5HxbDiQKcIWdlmRPYYMGZLq\nKQAAUiBTw27NireXLFm6avX6jZtqcwraF3Xp1n/AwL7dO6Z6LgCAlMmwsGuIrpte8e07H5r2\n0uJVO97abcDQsydOumHSuI72tAMA2p5MCrto3fLzjzxk6oKPI7mdh5w4ZvDAft336pifn7Nl\n8+a1H618Z+mil55/+farz3p02m9fq360R563LgEA2pZMCrvqq0ZOXfDxMZfe8dgtF/dsv5PJ\nY3UfP3brJRNunHbSZRMX3X9C0gcEAEilTHpa69qpSzt0v+j5uy7fadUFQZCd1+WcGx6/b8g+\nbz5+fZJnAwBIuUwKu4Ub6zvst+trrQ8/rmv9pkVJmAcAIK1kUtid0qVwzeJbVtY1+dZesZqf\nTV9W0GlEsoYCAEgXmRR21906YvO65wcNPePnM+dvjDZsf3PD5jeef3riSQPvW7b+hBtvTMWA\nAACplEkXT/Q/94kH/jL8wnufmjDyyUheUd/+/Xrs3TE/Pzdat3ndRyveWvrm6totWVlZpRff\nM+OSgakeFgAg2TIp7IIge+Ldz46a8Ot7Hn6scu68xX9/Zemirc/bZWXn9+x38EmlI86aePkp\nR+6b2ikBAFIis8IuCIJg3yGnfm/Iqd8LgoYtNWvXfrKxpi6vsN0eHTsV2pQYAGjbMi/stsnK\nKey0V2GnVI8BAJAmMuniCQAAmiDsAABCQtgBAISEsAMACAlhBwAQEsIOACAkMni7kyRbsmRJ\nQUFBqqdoMfX19VOmTOndu3d2trjfKhaL/fOf/zzggAM8Jo08IDvymGzHA7IdD8iOYrHYO++8\nc9555+Xm5qZ6lhazZMmSVI/wmYTdrjX+XbzgggtSPQgAZKT7778/1SO0vPRMVWG3a+ecc86W\nLVtqampSPUhLWrBgwbRp04455pjevXunepZ08c4777zwwgsek208IDvymGzHA7IdD8iOGh+T\ns88+e/DgwamepSUVFhaec845qZ5iZxpok6ZPnx4EwfTp01M9SBrxmGzHA7Ijj8l2PCDb8YDs\nyGOSZF4EAAAQEsIOACAkhB0AQEgIOwCAkBB2AAAhIewAAEJC2AEAhISwAwAICWEHABASwq6N\nKiws3PaRRh6T7XhAduQx2Y4HZDsekB15TJIsq6GhIdUzkALRaHTOnDllZWWRSCTVs6QLj8l2\nPCA78phsxwOyHQ/IjjwmSSbsAABCwqlYAICQEHYAACEh7AAAQkLYAQCEhLADAAgJYQcAEBLC\nDgAgJIQdAEBICDsAgJAQdgAAISHsAABCQtgBAISEsAMACAlhBwAQEsIOACAkhB0AQEgIu8yz\n/M9PX/Y/pw7s3WOPwtx2e3QacPjxl3/npyvqYrtzcI/8nEhu59afPUHN+pE/y7OjemdlZVV/\nUrfTPzZrrT+d1T8rK2vuus2780ORKr/9/D5ZWVnLNkeDf/+nbFrlmtp4vu12f6O+0n2Pwk7D\nWvHHaI7F938hKyuroOMxy+uiO97a+CB8971PdrzplRsPa3wQLn31o53eK56HbtEdQxv/OOGF\nFTsdb/07tzYe0PdLc7d9MX1+KSUw/7/+OCorK6t48p+TOCZslZPqAWie6dePOfN7vw2CoNeg\nI48edMi65cv+8fqLd/3tTw/f/8vZCyuHdsxP+OC0lcyfImlrrX/n+t6H3lNy718rz+rXUt+T\nBHQ54qRTawdt++PKuVXz1m0+cPjJA9v953fjPrmRVIzW8jave3H45ZWLflIe/11uuHtx4ye/\nvnLO3X8Y9+mbEnjonr3698G883dc5bWbH45/pBRKq/n9DuEzNZA5Xr3ntCAIig445ZkFH2z7\nYv3G93980WFBEOz1+a8nfHD3vEh2TqfWnj8BzfopmjZ75H5BELy0fvNO/9istf545gFBEPxh\nbW3CP9fat64KguDYKf9I+DuQsA3L/rl48eL62E5ueubQrkEQXLdsXQLfdru/URO7dSjoWLY7\nc7agv//k6CAIsnOys7Jy71m8ZrtbG/8+f+fd9dt9ff17FUEQdOz/te55kUhet1V10SaWaOKh\ne/3HQ4IgaN+jMDu38/ubd/wm0WOK8nPadQ+CoM+pf9j21fT5pZTA/MufGxkEwaArXm69qfwO\n4bM4FZsx6jf87cQrns7rcOiLr/7q5OK9t309p92+k+6tPr97h49eueWO9zckcHDTYpubecqz\n5bTgT5FWa5Fa7Xv3O/DAA3OyUj1H0g25/5pIsOWaYRdtjDXEc/z86+8LguCEH1/yo6H7ROtW\nTnpx5e6sfuj3R8XqV1/54vZnMz9590cvrNt86A1jduebJ0HrzZ/C37GEkrDLGK//YOLq+tjR\ntz9+cPsdTqBn5V1367kjR45cNv/jBA7eUeOrZ7bU/GNy+VHt2hXkRAp69S+e8LX71kcbgiDY\ntGpGl7xIUd/zaz/12+jeL/bOzs69/bXP/J4JaO5P0RBd94vvX370Qb33LMzv2uuAk8ZfNWvx\nulZaKwiChlj9r2655LA+3dvltevVv3j8V+9pfHwabXjnuasnlB+4794Fubkdiroedvypdzz9\neuNN9/Xv3LHvj4IgeP68z2VlZd2zYmOcQ9Iiqkp6bHuNXbM08d90lzb+a9YRHQtyC/tNe2Nt\nc9dtKZ0PnjT9/AM3vP/LUbf8dddHN2y+4sll2bmd7yzb98RbhwdB8IerfrM7q+/9+dv2y895\n7qu/2+7rr938s6ys3NtO7b073zwez0/93qihgzrtUZhX2OGAQ479xt2/+3Te1q1746YLx35u\n373z89r3PODwC6//6Yf1/5VbLTt/079jGzXxC83vEJqS6qcMiddVPfcIguCPazfv+tBmHtyw\nw1mPxlMzVx3VNbdD/7HnX/rVyyYe3Ck/CIKDLqhsPGD+D4cFQVD2g781/nH5nKuDIDjkst81\n5wdq4Z8iFt1w6THdgiDoPLDkzPMuOOWko/OzsyJ5+/zwuRWNBzR9KrZZazU+PheMHZDb4XNn\n/O9lX73sgoO6FARBcNDE3zcesOmDGfsX5GRl5R4x8ssXXPh/Z36prFNOdlZW9tdfWtnQ0LDo\nsYcrvjMsCIIDzr3pJz/5yesb65v1sLCbKod2D4Lg7dotO97UxPnEpv+bNjR5KnbTyj8M6VSQ\nU7DflNc+bp2faRcaT8V+cd7KLbXvDNkzPzun4zOrNm27daenYj9aeGUQBN2PebShoSFa/2G3\nvEh2pMOSTZ/5d3WXp2K//MZHT57UKzunaNl/PfLRY4vyO/b95ifv/zBozVOxL393RBAEhV0P\nHjfhgokTxh3YOT8IgmHf3/obbPP6l4/ZuzArK3tQyUnnXzChdHDXIAj2PvKi2liC8+/yVOwu\nf8c2/QvN7xCaIOwyRu+CnJz8Xq1xcMNnhF1hl7KXP6hp/Ert2hf3yYvkti/eekRs84UHdszO\n7Vz54aYttcuG7pnfvtuYj+ubeglOApr1U7x2yzFBEBw++dHN/37t1MqXf94jP5LX4fMf18ca\ndhV2zVqr8fEp6HzivH8/PnWfvNYrPydvj//f3p3HN1Hnfxz/TNKkNpo2igAADWtJREFU90E5\nWlqw0sppESrIJasUbKXCcoOcIgvIb5dfEVjlsLgenJZTYXFXC4u4oLKUBQUWEeRaRBAWxEVQ\nShEqFShHoSVt2qSzfxRCqQkJ6QXD6/lXZ+aTme9k0m/ej5nJd1oWT+59MVpE+q/4wfaSC4fm\niEj4k9eTH/fHVCH3gp3TY+oo2OVl7Wpfw1vvGZ5y4EL574xrbMFOVdWfv3hRRGo0n2D7d7Ub\n7FbH1xWRUd+cK55c8ZswEUlITXe0CVeCXdbB0SLS6/MM26Krp+eISMe/H6/gYFcU6eVh9G9p\nO+jmq/uDDTqvak8VT77XMVxExnxy5Ea95b0BUSIy4IsM99rvYrC7TR/rtEOjD4EjXIq9R6iF\np81WvWfd8i92rOPSJa1qehX/7RnYbmSor9X88/VlinHul0v81ezn4l5PHRW/N1fmbf8g2KNc\nP053uBdjZn3jGfD4ttmDjTfunQppNWjViIYFuQdnnXJ2Qdatd6xDypLWN94fg98jz4f4WM1n\niifD415dtmzZon4P2YqDGvUVEXNW3h1tAncP946p+fLeZ6Kf3n1JFu44OPzR6hXeSheEP7Vg\nXqfwC4eSB65Ic1RTVJg1Zkemwbv+nJjr95t2So4Xka8nfViWTVdvOivS22PnhLW2OYenLVF0\nhuTuFXsdVi0ynTZb9YYQWx9l9G+x75v9u7fMFRGL6fvE7ZlBUa+83a/JjVfohyyc3aZNG8vu\nW+6+KPf236aPLVOHhvsbw53cIxRDqEGXVXCm/Isde7ZNzZKTpXKbb1ivLdOefGxS8rOHpPXk\nbS80DCrj5kq7k70ozD2wI9vsV7vxqmVLS87P9tWJyL79FyXqts1z6x0b9HitkpNeupt344d3\n6TdURLWaTh79Mf2nn35KP7Hrs8V3tHLcbdw4ptaCzG7RnbadM4lIWp6lUprpktH/TF1Qq33q\nyM4HexyN8TX8uuDsVy9mmq2Rvd/x01//VNdsMTvU+OG5tGlf50xq4290b7uK3m9+x/Cemyan\n5f/+IS+9SFHSqvTAekkt/Ay5FZlVFJ3vrNiwl77cULfhb54f2P3Jx9u1adsqqllM8dLczD+b\ni9Qmg/uUfIlX9Z579vSs6PY76mPL2qHh/sYZu3tGQrCXJf/UrqulB9QtZs7e2rNnz4HDU9wo\ndqS6wcnHo/mYv/jrdYqiTBzb2lnz3eH6XljyfhSR3F9SRtzqpeT/ikhepvPzZG68Y2FGh8Ob\nWUzHpgyNr+4TENkkJq5rv1fnLjkf2MFpG3A3c+OYFpqObrsSsWzrQk+dsrjnwPOFd8tvH43+\nrTcv6mrJO9Gjn/1O4J/jtohIemqCbahhvbHm2QKrqha+vDK9LJtu/1avImvu+C1nRCTn5wU7\nss0t3hxclhW6aPznh5dOS2ykS3tn6oTendvXqebXrFP/Tw5kiYj58ikRCWgc4Mp6yrf9jvrY\nsndouJ8R7O4ZI3tFiMiUFSfsLj27a87atWt3ngx1o9htq17omqvqvRQZ1WVaRXxlub4XemO4\niIS2+tTuDQd7x0XbXYN727JRHI+XkdS2/fTlX8SOnfPvb9NyzebM9O83rJzntA24m7lxTPXG\nWisPfjW04/+vGdUkP3t7wpRdldNUVzQcvjqxSfDpjX+YvLP0ICaWvGMTD1/UG2uXShVDBzwm\nIofeXFiW7VZrPL2Rj+GrSatF5LvpKYrOMLvbA2VZoYsUj+BhSe/s/fFsdsbR9R+9P/a5+BM7\n/jGoXfSuqwWGgGARMZ02ubKeyml/2Ts03M8IdveMmOkLfPS6PS/13n/lV2eVVPO00btFJGHm\nY24Uu+fsrlcHr0iLHv3p+vExWftnDFh+vCxrs8v1vTAGtm/iY7iavqxUvkz7cPq4ceN2OzgP\n5962nLKYjiQfvhgUNTv1rbGPPxLl46GISFFhliuvxd3JvWNq8Gnap36giDw9f2NLf+OhuV3X\nnHUpPVQK/cwv/uqn183v3q/Uc8ZOrRtzzVoUFrvo/VstXb6pjqdHbubidRddesaaXYrOe158\nnUvHphw1FU75+GRg5KsxfnauBZev/IvrJk+ePC/1lIgE1mnUpf+IeX/7bOcbMdaC87OOXPIL\nHaEoSvoHm0q+pCBnj16nq9VsRZW0v+wdGu5nBLt7hmdQ3MY/xRaajsY27Zb6zc27wSymM28N\na52SkRPw4JDFrULcKHaDNT+te5dkr+Cnts6N7zBzc+ca3mtGxe0p7+7mTvZC9+7vGpourOn8\nxqe2rjDn5PqEUa+/u3Rvcxd63vJ8xxQPnaJYTMctN0akKirMWjS6l4iI3PINWmS5W67NwQmX\nj6ldes8HUlcMKbLmjuo81aWhgSuFb1ifTa+0Mmfven5TRsn5Ka/sE5EB8zqUqtd5BL/TIUxE\n3rzxnDH3tJnZT7XmvZDy8rbs/JbTBpZlVS5TZ82a9afEKRdv/sep+w5eEpGmId7GwCdeiw6+\n9P3EpM9sJ+zV1eOGF6lq6yltq6j9rnZo9CGwoyJ/cotyZ30vsWPxgQt/uPUz3XrEd2hXw6gX\nEd/wJ748b3K72O5wJxsu5ZWsmRcZZKtJ6V1PUXQz9mcVT2YdmKFTlNpPzKzCXbaaz/RuGCQi\nNRu06Dvs/4b07RzkodPp/aZuyywucPpIMde3ZfeRYtMfDNQbaxf/PbN9qIjUa99n4pTXxowc\n/GiIT2ir/nU9PQy+D89Y8FdVVa9mJItIUINer7/x2u4rrg43iHLh3nAnTo+p00eKJT1aU0SG\nrjlZ/rvkTMnhTkoqsmR3D/Ut/swXD3eSf3mLXlGM/i3N9h65dvG/fxQR7xo9f73IleFObmw1\nv6mvQVEURWc8mFtQPK+ix7GbERsmIr7hzXsP/N0fRj4fGx0iIiHtxhc/Wc50buPD/kZF0bfo\n0GXk74d3bl1HRIKjh+VYitxrf/FwJ4H1O/b5lf6Dxqou9LFOOzT6EDhCsLv3HNu6fFjvuAdD\nq3t56H38qzVqGTtmWsppe99SrhffUbD7+fPxIlJ/yMcll64cGCUiwyvmS8vFvbCYMxZOHBYT\nWdvbYKj1QIPY7iNSD9x86qsLwc7VbTkNdpb8U1NHda9XK8DoXe2RNp3GJK82F6lbk3oFeRv8\nQx9VVVW15iX1bRvkYzD6VPvg3LVyeI/gMveCndNj6jTYXTu31l+vM/o1T8+z/99acRwFO1VV\nL347R68otmB3OLmViDR+YZf9FRXlN/Mzisi7mbmlltxBsFPVzf2iRKRa/am2ORUd7KwFWX+e\nPDymQR0fo97DyzeyadvEqX+7WOKBwdcyv54wpGu9kGoGD8+aEdFDXp7/y43HwrrR/uJgZ5fB\np4nqQrBTnXVo9CFwRFHVu+fKAAAAANzHPXYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAA\noBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAAoBEE\nOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAAoBEEOwAA\nAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAAoBEEOwAAAI0g\n2AEAAGgEwQ4AAEAjCHYA7js7B9RXnNl4Ob9ctrUlIUJRlD05BeWyNgC4PY+qbgAAVLbqLeN6\n5EfbJs9u+9fXV8wN47s29rnZJYYY9FXRNAAoE0VV1apuAwBUpfUxIb89dD7ppyvTIgLKfeVb\nEiLiNp3+6qq5rb+x3FcOAKVwKRYAAEAjCHYAcDuFuT+8lTgwOiLU2+BZPbTeM4PGbU/PuaMC\nAKg0BDsAcMhi+i6+QYtJiz66EhTVc8iAFlEBmz96O+7h6OUnrrpYAACViWAHAA6tG9Rt+y/X\n4qdvyvh298qlyzbv/vbYuqQic0Zip5dcLACAykSwAwD7VOuVketPewV33jD5advMh3479e2Y\nmldPvf9xVp7TgqpoNYD7GsEOAOwzZa26bCkKaftHD+WW+fGJDURkRdoVpwWV11YAEBGCHQA4\nYjWfEhH/+qXHQAloHCAiuRkmpwWV0UoAKIFgBwD26T0jRCTneOmfuOam5YqIT5i304LKaCUA\nlECwAwD7fGr0DfLQnd8z33rr/K0LfxCRZxsEOi2orJYCwHUEOwCwT/EIei+hbt6lDd1nb7PN\nTN/4+uh95wMeGPFcLR+nBVXRagD3NZ4VCwAO9fho3RNR7TZM6FhvVYcnW9S/8MN/Nu34j+IZ\nsXjbXBcLAKAyccYOABwy+Dbbcnz/9NHP+pw78smSZbuPXIjr/+KW778bFBngYgEAVCZFVdWq\nbgMAAADKAWfsAAAANIJgBwAAoBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIId\nAACARhDsAAAANIJgBwAAoBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACA\nRhDsAAAANIJgBwAAoBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDs\nAAAANIJgBwAAoBEEOwAAAI0g2AEAAGjE/wAUOnNe9CbStgAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzdeVxVZeLH8edygXsBkc0NcV9SUFJTQ61UREVNUHPf2saywiXTstTcSssaJVNx\ncmkqzXXUERsYt7BQocWVXAhFzBQ0RSDWC/fe3x841I8MLgT33PPwef/BC849h/N1nNfM1+ec\n53k0ZrNZAAAAQP3slA4AAACAqkGxAwAAkATFDgAAQBIUOwAAAElQ7AAAACRBsQMAAJAExQ4A\nAEASFDsAAABJUOwAAAAkQbEDAACQBMUOAABAEhQ7AAAASVDsAAAAJEGxAwAAkATFDgAAQBIU\nOwAAAElQ7AAAACRBsQMAAJAExQ4AAEASFDsAAABJUOwAAAAkQbEDAACQBMUOAABAEhQ7AAAA\nSVDsAAAAJEGxAwAAkATFDgAAQBIUOwAAAElQ7AAAACRBsQMAAJAExQ4AAEASFDsAAABJUOwA\nAAAkQbEDAACQBMUOAABAEhQ7AAAASVDsAAAAJEGxAwAAkATFDgAAQBIUOwAAAElQ7AAAACRB\nsQMAAJAExQ4AAEASFDsAAABJUOwAAAAkQbEDAACQBMUOAABAEhQ7AAAASVDsAAAAJEGxAwAA\nkATFDgAAQBIUOwAAAElQ7AAAACRBsQMAAJAExQ4AAEASFDsAAABJUOwAAAAkQbEDAACQBMUO\nAABAEhQ7AAAASVDsAAAAJEGxAwAAkATFDgAAQBIUOwAAAElQ7AAAACRBsQMAAJAExQ4AAEAS\nFDsAAABJUOwAAAAkQbEDAACQBMUOAABAEhQ7AAAASVDsAAAAJEGxAwAAkATFDgAAQBIUOwAA\nAElQ7AAAACRBsQMAAJAExQ4AAEASFDsAAABJUOwAAAAkQbEDAACQBMUOAABAEhQ7AAAASVDs\nAAAAJEGxAwAAkATFDgAAQBIUOwAAAElQ7AAAACRBsQMAAJAExQ4AAEASFDsAAABJUOwAAAAk\nQbEDAACQBMUOAABAEhQ7AAAASdgrHUAFMjMzP/3007y8PKWDAAAAm+Dk5PTUU0+5ubkpHaQ0\nil35Pv/88+nTpyudAgAA2BB7e/uXXnpJ6RSlUezKV1hYKITYuHFjhw4dlM4CAAAUdubMmb/9\n7W/F9cDWUOws1aZNm86dOyudAgAAKCw/P1/pCH+KyRMAAACSUOuI3d3UK4mJSTfTs3Jy8+31\nLm5eDVq39W3h7a50LgAAAMWorNiZjZk7whd9uHHL8Ys3//hpg7bdxk2a/ub00e72GutnAwAA\nUJaaip3RcP2Zrh02nb2jdfAM6BP6oG9L7zruOp19UUFBxu20q0nnjsd+s2LW2M+2fHEm7rOG\njjxlBgAANYuail3czAGbzt55dMrKre++1MjlPslNhjtbl4VNXLCl39RJ5z7qbfWAAAAASlLT\nsNacTUm1vF+IXTXtvq1OCGHn6DX+zW1rA+pf3jbPytkAAAAUp6Zil5BTWKtJSLmnde5ZrzD3\nnBXyAAAA2BQ1FbshXk53L76bZjCVdZIp7+MdKXqPYGuFAgAAsBVqKnZzlwUXZMa27zZq8/4T\nOUZz6Y/NBedj90zq57s2Jav3ggVKBAQAAFCSmiZPtH5q5/rv+k+O2D1xwC6to1uL1i0b1nXX\n6RyMhoLM26nJSZfT84s0Gk3gS2siw3yVDgsAAGBtaip2QthNWn1o4MR/r/nn1qiY+IsXTiWd\nuzdup7HTNWrZrl9g8NhJ04Z09VE2JQAAgCLUVeyEEMInYOjSgKFLhTAX5WVk/JqTZ3B0cnZ1\n93BiUWIAAFCzqa/YFWNLMQAAgFJUVuzYUgwAAODPqKnYsaUYAABAGdRU7NhSDAAAoAxqGtZi\nSzEAAIAyqKnYsaUYAABAGdRU7NhSDAAAoAxqesdu7rLgT5/e3b7bqA/eeWNY34dctP9/6qu5\n4PzRqBULZ2xMyRq02tItxYxGY1RUVH5+fhnnnDp1SghRWFhY2eCwKkv+Tu/LZDIlJCT4+/vb\n2VXsHzx6vX7QoEFarbaidwQAoGqpqdhVx5ZiMTExoaGhlpy5ZcuW3r17Vz49rMXyv9MqdPDg\nwb59+1r5pgAAlKKmYlcdW4oFBgZGRkaWPboTERFx5MiRRo0a/aXssBZL/k7vKy4uLjw8fMaM\nGd27d6/QhXq9PjAwsKK3AwCgyqmr2AlR1VuKabXakJByJmRERUUJISr6eA5KseTv9M+Eh4d3\n79595MiRVRsJAADrUF+xK6Gxd/Ko4+ShdAwAgI3LyMgwm80VusRoNB48eLCgoKBCV5lMpvPn\nz/v5+VViLECn0/Xr16+ib+s6Ojq6uLhU9F7Xrl27detWhS4xmUxHjx41GAwVvZfJZLp06VKr\nVq0q+p+Jo6Pjo48+WtGr7O3t27Rpo9frK3SVTFRc7ICq5eTkVPIVgDRWrVo1bdo0pVNUFwcH\nhzNnzvj6WvpmuRAiPT29efPmRqOx+lIp69lnn924caPSKRQjW7EzZB1r2maEECI1NVXpLFCZ\ngQMH7t+/PygoSOkgAKrSTz/9JIQYNGiQs7NzhS5MTU2tRPvJzMx0c3Or6FVCCK1W6+3tXaFL\nkpOTT548eePGjQoVu+zsbIlbnUajSU9PVzqFkmQrdmazIS0tTekUUCWtVtu/f3+lUwCoFi+/\n/HJFa5Pt27lz58mTJ5VOYVt4IV62P79jrS7x8fHx8fFKB4H6GI3GAwcOSPwPWQCA9GQrdhqt\na0BAQEBAgNJBoD7R0dHBwcHR0dFKBwEAoJJkexQLmdy4cSMkJCQjI8M6t8vNzRVCPPfccxV9\nEadyHB0d33vvvUqvzAKgQnjRopSnn3565syZSqeoeg899JDSERRGsYPtunDhwsmTJ729vT08\nrLGsjV6v1+l0lVg4oBIKCwsvXrz49ddfU+wAAFWIYgdb99xzz8m3YnBqairjBwCAKqemYpeR\nlppjNFl4so9PBTYWAwAAkICait2rnR7YkJZt4ckVXWQcAABA7dRU7N4+FN3mkzXzw7fnGc0e\n/r0faVpL6UQAAHU4cOCAlOvYLV68uHLXpqSkHDhwoGrz2AKTydIne7JSU7Gr3+7RWe8/GuiZ\n3GXOt75ha/dNbqt0IgAAVMbJycnOzu7IkSNHjhxROku1sM7KBjZLTcWumH/YcjHnMaVTAACg\nSnXr1j127Ni1a9esczuTyZSQkODv72+dPSG0Wu1jj9XokqC+YudY+9GHGjVw02uVDgIAgCp1\n69atW7du1rnXF198sWTJkn379g0ePNg6d6zh1FfshBAnrqUqHQEAAJQvLy+v5CusQLYtxQAA\nAGosVY7YoUZZvHhxpad9AQCqhNFojIqKys/Pr+iFcXFxJV8rRK/XDxo0SKvlzauKodgBAIBy\nxMTEhIaGVvry8PDw8PDwil518ODBvn37VvqmNRPFDgAAlCMwMDAyMrISI3aVnhWr1+sDAwMr\nejtQ7AAAQDm0Wm1ISEjlrh09enTVhkEZmDwBAAAgCUbsYOvmz58/cuRIK9zIZDLFx8d369bN\nCqtopqam9u/fv7rvAsDKrPk/I8B9Uexg61atWvXxxx9b4Ub5+fm3b9+uU6eOXq+v7nsZjcbq\nvgWA3xs+fLhGo6nuuxQVFeXm5jo7O9vbW+P/Xg0GgxXuAnWh2MF2tW/fvlevXtnZ2da53d27\nd2/fvl27dm0PDw8r3M7Hx4fZXoAVPP7442fOnDGbzVa41+3bt0+fPv3AAw/UqVPHCrcTQjg7\nO3fo0ME694IqUOxgu+rXr1+JPar/ymJL4eHhQ4YM6d69e4UuZLElwJb17t27d+/e1rmX0Wg8\nfPhwUFAQ/4MApVDsIBsWWwKgFK1Wy+uzUBbFDrJhsSUASmHEDoqj2EE2LLYEQCnR0dEhISH7\n9u0bPHiw0llQQzEfGwCAqpGXl1fyFVAExQ4AAEASPIoFAKC0ys2vj4uLK/laUcyvR5Wg2AEA\nUNpfmV9fucn1gvn1qAoUOwAASqvc/PpKT64XzK9HFaHYAQBQWqXn1zO5Hspi8gQAAIAkKHYA\nAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg\n2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAA\nSIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIH\nAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJ\nih0AAIAkKHYAAACSoNgBAABIwl7pAJViNvyYcDrxx+Q7Wdl5BqOj3qWeT1Nfvw6tfGornQwA\nAEAxKit2xoJr/1g4N3z9zst38kt9pNFomnUKCntt0czRPRTJBgAAoCw1FTtj/pVh/p32Xcr0\n8usxNrSjU971Y9HRiZmGvlPmBOizLiZ8f/jwl7PGHNp+cPW3G8KUDgsAAGBtaip2sVMG7ruU\nOeK9vVtmhTpohBCiKDf5lV7d1m85+klqjI+jXUF64tIXhizeOGV8/0Gfj2qudF4AAACrUtPk\nifn/SnFtPH3nq/danRDC3rnF0r1L8tO/fjbyqhBC59lm4bYTAz2d9s38QMmgAAAASlBTsTud\nXej5YEipgy71xwohknddK/5RY+cyrb1n7s2t1g4HAACgNDUVu4ddHe8mRJU6mH83Wgihb6Av\nORJ3NdvOwdOqyQAAAGyAmordgnEtsn5aMer9fxvM946YDKkLhk8XQoRObi2EMJtyosKfXXw1\ns/4jbyqYEwAAQBFqmjzRffne4D0dd7427OD6zj0f8nPIvnHqaGxypqFR8JIlbT2EELu6thh5\n8pbOvfOW7SOUDgsAAGBtahqxs9e32pf47Wvjexcln4rcvmnXfw5fzXcdMX3lD/95o/gEndcD\no8IWfZd87DEPnbJRAQAArE9NI3ZCCAdXv2WbY5asu3UhMaXAzqW1n6+bw2/dNORAbOm5FQAA\nADWGyopdMXsnd51WpPyY8MOJOLYUAwAAKKayYseWYgAAAH9GTcWOLcUAAADKoKZix5ZiAAAA\nZVDTrFi2FAMAACiDmoodW4oBAACUQU3Fji3FAAAAyqCmd+wWjGvRM2LFqPcf2zxrqKNGiPtt\nKRa9curiq5mN+q2x8HcajcaoqKj8/NJzbH8vJSVFCGEymf5afAAAgOqlpmJXHVuKxcTEhIaG\nWnLmlStXKh8dAACg+qmp2BVvKTbvxbCIbV9HJp0QQmh1XiOmz9+wfGrxCTqvB0aFhc17a7a/\nxVuKBQYGRkZGlj1iFxERceTIkebNmWYLAABsmpqKnaiGLcW0Wm1ISDkXRUVFCSHs7NT0PiIA\nAKiBVFbsitk71/PvVE/pFAAAALaFUSgAAABJyFbsDFnHvL29vb29lQ4CAABgbap8FFsGs9mQ\nlpamdAoAAAAFyFbsHGt1iY+PVzoFAACAAmQrdhqta0BAgNIpAAAAFKDWYnc39UpiYtLN9Kyc\n3Hx7vYubV4PWbX1beLsrnQsAAEAxKit2ZmPmjvBFH27ccvzizT9+2qBtt3GTpr85fbS7vcb6\n2QAAAJSlpmJnNFx/pmuHTWfvaB08A/qEPujb0ruOu05nX1RQkHE77WrSueOx36yYNfazLV+c\nifusoaNsE34BAADKpqZiFzdzwKazdx6dsnLruy81crlPcpPhztZlYRMXbOk3ddK5j3pbPSAA\nAICS1DSsNWdTUi3vF2JXTbtvqxNC2Dl6jX9z29qA+pe3zbNyNgAAAMWpqdgl5BTWalL+ZrCd\ne9YrzD1nhTwAAAA2RU3FboiX092L76YZTGWdZMr7eEeK3iPYWqEAAABshZqK3dxlwQWZse27\njdq8/0SO0Vz6Y3PB+dg9k/r5rk3J6r1ggRIBAQAAlKSmyROtn9q5/rv+kyN2TxywS+vo1qJ1\ny4Z13XU6B6OhIPN2anLS5fT8Io1GE/jSmsgwX6XDAgAAWJuaip0QdpNWHxo48d9r/rk1Kib+\n4oVTSefujdtp7HSNWrbrFxg8dtK0IV19lE0JAACgCHUVOyGE8AkYujRg6FIhzEV5GRm/5uQZ\nHJ2cXd09nFiUGAAA1GzqK3YlNPZOHnWcPJSOAQAAYCPUNHkCAAAAZaDYAQAASIJiBwAAIAmK\nHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACA\nJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYA\nAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg\n2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAA\nSIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEjCXukAAADlGY3GqKio/Pz8Cl1lMpkSEhL8/f3t\n7Co2TKDX6wcNGqTVait0FYByUewAACImJiY0NNSadzx48GDfvn2teUegJqDYAQBEYGBgZGRk\nRUfs4uLiwsPDZ8yY0b179wpdqNfrAwMDK3QJAEtQ7AAAQqvVhoSEVOLC8PDw7t27jxw5ssoj\nAagEih0gP16fAoAagmIHyI/Xp1BNnJycSr4CsAUUO0B+vD6FajJw4MD9+/cHBQUpHQTAPRQ7\nQE3ef//9AwcOWOdeN2/eFEIcOnQoISGhotd++OGHlbijn5/fypUrK3EhlKLVavv37690CgC/\nodgBavLpp59evHjRxcXFCvcqKioSQly+fPnq1atWuF1ubu7XX39NsVMXo9F4+PDhoKAgXqkE\nbATFDlCZ5s2b79mzxwo3MplM8fHx3bp1q+jkicqZPn360aNHrXAjVKHo6OiQkJB9+/YNHjxY\n6SwAhKDYAfgzdnZ2PXr0UDoFbFpeXl7JVwC2gL1iAQAAJMGIHaAy6enp/v7+SqeoFrynBQB/\nESN2AIBKYh07wNYwYgcAqCTWsQNsjQzFzph3/d87oy5dv+vZ2Lf/sIFNXWT4QwGA7WMdO8DW\nqOxR7N1zeyc+3qupl7OH9wNhyw8LIW5//7FvvRYjnnr+9Tmzn58Y2rpeqwX/SlQ6JgDUCEaj\n8cCBA0ajUekgAO5R0+BW7s3/+Hcefr3A6OTlY3/ncsSsvnkNDpx+4cXkwrovvvFilzZ1f0o4\nvurDTW+PeajxpbRJzVyVzgtUC09Pz6+++krpFFWPdezUiHXsAFujpmIXOeGFGwbT61tPvjOm\nk8lwa/HQLosm9Nc61Nl7+eLjjWsJIYR4/uW/BXm3f+rNcbsnHX9K4bgAIDvWsQNsjZoexb4T\nd8u1yZvvjOkkhLBzrDd70wdCiHoPR/yv1QkhhLvvxPdae9w5u1yxlAAAAApRU7G7nF/kXL9r\nyY+62o8JIdz8fEqd1raxizH/ilWTAQAA2AA1FbtHajtmXdlU8o5u1pWPhRC3jsaXOm3fhQxH\n14etGw0AaiLWsQNsjZqK3ZvjW+b+siMwbOV35y59f2TXuP5L7J3c7l58bd6/zpac89VHz666\n/mvjwa8rmBMAaojidewGDhyodBAA96hp8kT3v0eFRvlHRrz8cMTLQgg7B8+Pzp47+njbJSM7\n7OnRr3ObetcSjh75/qpjrfabI3opHRYA5Mc6doCtUVOx0+qa7D5/4dNV677+5sSvDg3HzHh7\nZNu6T52OFUNGfRpz8PxxIYRo8cjoNZs3POzqqHRYAJCf0Wg8fPhwUFAQ+/wCNkJNxU4IodU1\nfHbWwmd/d8TB1f+TLy/8/Wpi0s8ZHo3atG3qrlg4AKhhWMcOsDUqK3b3mA0/JpxO/DH5TlZ2\nnsHoqHep59PU169DK5/aSicDgBqEdewAW6OyYmcsuPaPhXPD1++8fCe/1EcajaZZp6Cw1xbN\nHN1DkWwAAADKUlOxM+ZfGebfad+lTC+/HmNDOzrlXT8WHZ2Yaeg7ZU6APutiwveHD385a8yh\n7QdXf7shTOmwAAAA1qamYhc7ZeC+S5kj3tu7ZVaog0YIIYpyk1/p1W39lqOfpMb4ONoVpCcu\nfWHI4o1Txvcf9Pmo5krnBQDJsY4dYGvUtI7d/H+luDaevvPVe61OCGHv3GLp3iX56V8/G3lV\nCKHzbLNw24mBnk77Zn6gZFAAqBlYxw6wNWoqdqezCz0fDCl10KX+WCFE8q5rxT9q7FymtffM\nvbnV2uEAoOYpXseOtU4A26GmYvewq+PdhKhSB/PvRgsh9A30JUfirmbbOXhaNRkA1EhGo/HA\ngQNGo7H8UwFYhZresVswrkXPiBWj3n9s86yhjhohhDAZUhcMny6ECJ3cWghhNuVEr5y6+Gpm\no35rLPydRqMxKioqP7/0HNvfS0lJEUKYTKa/Fh8AZMM6doCtUVOx6758b/CejjtfG3Zwfeee\nD/k5ZN84dTQ2OdPQKHjJkrYeQohdXVuMPHlL5955y/YRFv7OmJiY0NBQS868cuVK5aMDVSc/\nP//8+fNWuJHJZEpISPD397ezs8bQ/q+//mqFu6BqsY4dYGvUVOzs9a32JX4778WwiG1fRyad\nEEJodV4jps/fsHxq8Qk6rwdGhYXNe2u2v4fOwt8ZGBgYGRlZ9ohdRETEkSNHmjdnmi2Up9Vq\nf/7559GjRysdpFowuRIA/iI1FTshhIOr37LNMUvW3bqQmFJg59Laz9fN4bexhJADsaXnVpRH\nq9WGhJRzUVRUlBDCOoMWQNlWrlz5zTffWOdeJpPp8uXLLVu2tNp/+Vu3bm2dGwGArFRW7IrZ\nO9fz71RP6RSAAnr37t27d2+lUwD3sI4dYGsYhQJwf0x4RLlYxw6wNbIVO0PWMW9vb29vb6WD\nAKoXHR0dHBwcHR2tdBDYLtaxA2yNbMXObDakpaWlpaUpHQRQPSY8olwM6wK2RrZi51irS3x8\nfHx8vNJBAEB+DOsCtkaVkyfKoNG6BgQEKJ0CAGoEhnUBW6PWYnc39UpiYtLN9Kyc3Hx7vYub\nV4PWbX1beLsrnQuQBxMeAUB1VFbszMbMHeGLPty45fjFm3/8tEHbbuMmTX9z+mh3e431swGS\nKZ7wGBQUpHQQAICl1FTsjIbrz3TtsOnsHa2DZ0Cf0Ad9W3rXcdfp7IsKCjJup11NOnc89psV\ns8Z+tuWLM3GfNXSU7fVBwMqKJzwqnQI2jWFdwNaoqdjFzRyw6eydR6es3PruS41c7pPcZLiz\ndVnYxAVb+k2ddO6j3lYPCEjFaDQePnw4KCiIxSzwZxjWBWyNmoa15mxKquX9QuyqafdtdUII\nO0ev8W9uWxtQ//K2eVbOBsiHCY8oF+vYAbZGTcUuIaewVpPyN4Pt3LNeYe45K+QB5MaER5SL\ndewAW6OmYjfEy+nuxXfTDKayTjLlfbwjRe8RbK1QAFBzMawL2Bo1vWM3d1nwp0/vbt9t1Afv\nvDGs70Mu2v8/9dVccP5o1IqFMzamZA1avUChjIAtMhqNUVFR+fn5FboqLi6u5GuF6PX6QYMG\n8XiuJmBYF7A1aip2rZ/auf67/pMjdk8csEvr6NaidcuGdd11OgejoSDzdmpy0uX0/CKNRhP4\n0prIMF+lwwI2JCYmJjQ0tHLXhoeHh4eHV/SqgwcP9u3bt3J3BABUmpqKnRB2k1YfGjjx32v+\nuTUqJv7ihVNJ58zFH2jsdI1atusXGDx20rQhXX2UTQnYmsDAwMjIyIqO2JlMpoSEBH9/fzu7\nir2zodfrAwMDK3QJAKBKqKvYCSGET8DQpQFDlwphLsrLyPg1J8/g6OTs6u7hxKLEwJ/QarUh\nIeVPPPqj0aNHV3kYyIR17ABbo75iV0Jj7+RRx8lD6RgAUGOxjh1ga1Rc7AAAymJ7EsDWqGm5\nEwCATWEdO8DWUOwAAJXEOnaAraHYAQAqiXXsAFtDsQMAAJAExQ4AAEASFDsAQCWxjh1ga1ju\nBABQSaxjB9gaih0AoJJYxw6wNTyKBQBUEuvYAbaGYgcAqCTWsQNsDY9iAUAqAQEB3377rTXv\nGBISYp0baTSa8ePHb9q0yTq3A9SIYgcAUjl9+nS9evU6duxondvduXPHy8vLOvc6fvz4qVOn\nrHMvQKUodgAgm/bt2y9fvlzpFFVv2LBhSkcAbB3v2AEAAEiCYgcAACAJih0AAIAkKHYAAACS\noNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEA\nAEjC0mI3YvKcPV+fM1ZrFgAAAPwFlha7XeveeaJXe/fGHZ6b/V7M2evVmgkAAACVYGmxO380\ncv6UcT6GpA3vze7ToZF3+16vvrPu1E9Z1RoOAAAAlrO02Pk+ErJo1ecX0zJOHtr52qTh+p/j\n/z5ncudmnn6PDlmydkdyhqFaUwIAAKBcFZw8oXHsFDRi2fp/Jaenx/1n07QJAzJORc17aXTr\nOh49Bj+1ZuuB2wZT9eQEAABAOSo/K9ZOa693cnF1cRBCmIy530RtmjIuuKFH08nv7a+6eAAA\nALBUxYqdseCXL3dtmDL+8cZuHgEDxi5btyO7Xuep88OPnLn26/Wz69+d9WDt9HWzBzx/mNkV\nAAAA1mZv4Xl7P/lgz549+/57LN1g1Gg0TTv2mTl8+IgRw7u1qfe/UxpNmv3+xMkheo9eXyw4\nI4J8qikxAAAA7svSYjf0mRkajfaBgP4vDB8+fPgTDzX3uP+vc2reoEGDZh29qi4hAAAALGJp\nsVu4ctPw4UPb+9Qq+zStrnFqaupfTgUAAIAKs/Qdu8GP+OpqOd73o5yr506dSaq6SAAAAKgM\nS4tdly5dXjhy474fJa6f0PXhXlUXCQAAAJVRzqPYT9asyiy6tzTdtX3/XJniWfoMc9GxbVeE\n0FVHOAAAAFiunGL31qxXkvOLir9P2rj45T85rdmgdVWaCgAAABVWTrHbFBfkgMMAACAASURB\nVLU/z2QWQvTt27fTos3vP9LgPr/C2SsgoGO1pAMAAIDFyil2PQL7FH8zYMCAjv36BnWvX/2R\nAAAAUBmWLncSHR1drTkAAADwF5VV7DIzM4UQLrXd7DX3vi+Dm5tbVeYCAABABZVV7Nzd3YUQ\nu27nPuHlVPx9Gcxmc1XmAgAAQAWVVezGjBkjhGjkaC+EmDBhgpUSAQAAoFLKKnZbt24t+X7T\npk3VHwYAAACVZ+nOE0IIYS44+NnK1974V8mBQcMnvLMuMsfEQ1gAAADlWVrsTIW3nu/WpP9T\nL/9j87clB/fv2TJn8pBmXZ9OKzRVTzwAAABYytJid3LhoPXf3ur65NsHDr1WcjDjWsLfn3/s\n9snP+s+Oq554AAAAsJSlxW7xR+dd6k2I+3RutzZ1Sg66+rSb+dGRF7xrJX2yuHriAQAAwFKW\nFruvMgs8Oz6pvd9vGNm1jiHreFWGAgAAQMVZWux8nR2yU07f96OzSb/aO7epukgAAACoDEuL\n3cJBjTOSXp+9I6HU8cTI+bMupnv3mlfVwQAAAFAxlu4VG7RhxyOHHnlv9INfRDwRGhTQqE6t\n/MybJ498sfXAKXvndp9uHlitKQEAAFAuS4udg0unQxePzX5uytq9e979anfJcb8+E1Z9vLaX\nm6564gEAAMBSlhY7IYTOq9MHu4+9ezsl/ruEtPQsR1fPth27+TXxqL5wpUycONGhVuvXlr3R\ntraD1W4KAACgFhUodsX0dZr1HtisGpKUb/PmzUKIz7dsXfDRZ3PGdFUkAwAAgM0qa/JEZmZm\nZmZmkfm378tgnbhOXiHTghznjn34waGvHLvyq3VuCgAAoAplFTt3d3d3d/fI9LyS78tgnbha\nXdP3d5/5bvs7mqNrH2tVf+CkBd/9nGOdWwMAANi4sh7FjhkzRgjRyNFeCDFhwgQrJbJAl1Gv\nnwydsHreq4s/fLvbJyv6jfnbU08+ObzfQ44apZMBAAAop6xit3Xr1pLvN23aVP1hKkCrbzT9\n71snv/b68oVvvrtu1f7PVzp7txs1ZkRQz0d79Ojeop6L0gEBAACsraxHsXGHD8YlZRV/v3//\n/u9/ybdKpArQ1+swNyLy5tWTa9+a3k5//ZPwRROH9WtZv5Z3605KRwMAALC2sord4IEDnlt9\nofj7AQMGzD6eZpVIFebs0+GFeR98m3znhyN7Fr/yt+7tm/5y+YzSoQAAAKytrEex3Vx1Bzc+\nN9vtCWetRghxZdvKRaf/dJLEggULqj5dxdi16zW0Xa+hbwpRcPcnpcMAAABYW1nFLmL91IfG\n/P29t+7tD3tl2wcL//xkGyh2v9F5NFE6AgAAgLWVVeyaPrEs7fYrFy9fN5jMXbp06fr3yLW9\nG1ot2R9lZGRo7Ni7DAAA4P7K2XnCoXZ9/071hRATJkxoF9i980N1rJLq/tzc3O59Zzb8mHA6\n8cfkO1nZeQajo96lnk9TX78OrXxqKxgPAABAWZZuKWYjy50YC679Y+Hc8PU7L98pPUVXo9E0\n6xQU9tqimaN7KJINAABAWWUVu+KNwlxqu9lrRLmbhv02nFZtjPlXhvl32ncp08uvx9jQjk55\n149FRydmGvpOmROgz7qY8P3hw1/OGnNo+8HV324Iq+4wAAAAtqasYle8Udiu27lPeDmVu2mY\n2Wyuylz3Eztl4L5LmSPe27tlVqiDRgghinKTX+nVbf2Wo5+kxvg42hWkJy59YcjijVPG9x/0\n+ajm1Z0HAADApqhpS7H5/0pxbTx956uhJUfsnVss3btklc/zz0Ze3T+iuc6zzcJtJ747XHff\nzA/EqJUKRgUAALA+NW0pdjq70PPRkFIHXeqPFeL55F3XxIjmQgiNncu09p4H4rYKQbEDAAA1\nS1k7T5RmLjj42crX3vhXyYFBwye8sy4yx1TtD2GLPezqeDchqtTB/LvRQgh9A33Jkbir2XYO\nntaJBAAAYDssnRVrKrz1wqP+67+95dro1ffeGVF8cP+eLdG7P1/x0ZMJ8f9s4FCRjlgpC8a1\n6BmxYtT7j22eNdRRI4QQJkPqguHThRChk1sLIcymnOiVUxdfzWzUb42Fv9NoNEZFReXnl7UN\nbkpKihDCZDL9tfgAAADVy9Jid3LhoPXf3ur65NsfzplccjDjWsK6xS/OWvdZ/9nPn13xSPUk\n/E335XuD93Tc+dqwg+s793zIzyH7xqmjscmZhkbBS5a09RBC7OraYuTJWzr3zlu2j7Dwd8bE\nxISGhpZ/nhBXrlypfHQAAIDqZ2mxW/zReZd6E+I+nav93UFXn3YzPzpyaZ/bJ58sFiv2V0e+\n37PXt9qX+O28F8Mitn0dmXRCCKHVeY2YPn/D8qnFJ+i8HhgVFjbvrdn+HpZuUBEYGBgZGVn2\niF1ERMSRI0eaN2eaLQAAsGmWFruvMgs8+zypvc8ndiO71ln3n+NVGerPObj6Ldscs2TdrQuJ\nKQV2Lq39fN1+9wg45EBs6bkV5dFqtSEh5VwUFRUlhLCzq/ZnzQAAAH+FpcXO19nhx5TTQvT7\n40dnk361d25TpanKYe9cz79TPWveEQAAwPZZOgq1cFDjjKTXZ+9IKHU8MXL+rIvp3r3mVXUw\nAAAAVIylI3ZBG3Y8cuiR90Y/+EXEE6FBAY3q1MrPvHnyyBdbD5yyd2736eaB1ZrScoasY03b\njBBCpKamKp0FAADAqiwtdg4unQ5dPDb7uSlr9+5596vdJcf9+kxY9fHaXm6WTlaobmazIS0t\nTekUAAAACrC02AkhdF6dPth97N3bKfHfJaSlZzm6erbt2M2viUf1hasEx1pd4uPjlU4BAACg\ngAoUOyGEqSj9++9OnP8xJTM7742543NSrpqEh01NFtVoXQMCApROAQAAoIAKFLvUmIgh4179\nLi23+Mc35s47vSj48SOuiz/aNa1/k+qJ96fupl5JTEy6mZ6Vk5tvr3dx82rQuq1vC293K8cA\nAACwHZYWu+yft3caMO0Xk+u4l19pZ7d77orzQgifQcM9d7w/43F/1x+vPdO8dnXmvMdszNwR\nvujDjVuOX7z5x08btO02btL0N6ePdrfXWCEMAACATbG02O0Y/fIvRv2nCVcm+Lpf2/9NcbFr\nNnLJmS4BTR4YNmfcjmfiJlVnTiGEMBquP9O1w6azd7QOngF9Qh/0beldx12nsy8qKMi4nXY1\n6dzx2G9WzBr72ZYvzsR91tDRph4RAwAAVDtLi92yU3c8262d4Fv6Wadr89DV7es8c3a5ENVe\n7OJmDth09s6jU1ZuffelRi73SW4y3Nm6LGzigi39pk4691Hv6s4DAABgUywd1rpZaHRp1Oy+\nH3k3cTYablRZoj83Z1NSLe8XYldNu2+rE0LYOXqNf3Pb2oD6l7exYDIAAKhxLC12Azz0t098\nar7PJ6ZPvvlF59arKkP9iYScwlpNyt8MtnPPeoW556yQBwAAwKZYWuzmvNIp5+amvrM/zjH9\nrt2ZC/csHLjpZs4Dz86tlnT/3xAvp7sX300zmMo6yZT38Y4UvUewFfIAAADYFEuLnf+r/5nS\nrf6X7/2tXqO2ExadFkI898z47g/UfWLRAbfWI794u0t1hrxn7rLggszY9t1Gbd5/Isf4h9FD\nc8H52D2T+vmuTcnqvWCBFfIAAADYFEsnT2i0bh8evdTlndeXr/v867gMIcSGT7bovZqNe2X+\n+++8bJ0pqK2f2rn+u/6TI3ZPHLBL6+jWonXLhnXddToHo6Eg83ZqctLl9PwijUYT+NKayDBf\nK+QBAACwKRVYoFijrfXUvNVPzVudfuPqzfRsXW3PZk28rbumiN2k1YcGTvz3mn9ujYqJv3jh\nVNK5e+N2Gjtdo5bt+gUGj500bUhXH6uGAgAAsA0V21KsIP3Szm27j3//w6272braXm07dXti\n3Ph29fTVFO6+fAKGLg0YulQIc1FeRsavOXkGRydnV3cPJxYlBgAANVsFit03/5gROn3VLYPx\nt0OfbVj46qwXV3yxeuojVR+tPBp7J486Th7WvzEAAIBNsvRRaupXr/V4aWW6nc/0peuPn/nx\nRtrPZ7756p/vvdLEITti+mOzjqRWa0oAAACUy9IRu1VPrxN2Lp+ePjuujVvxEe/6Pg8+3HPY\n0K6Nfcevf3rl31PerbaQAAAAKJ+lI3Ybb2S7t36vpNWVcGs9ZkVbj5wbH1d1MAAAAFSMRcXO\nZLhxy2B0rN3ovp829NBptLWqNBUAAAAqzKJiZ+fYsI+7Pv38ght/2PXBVHhz0ZnbdTrNqYZs\nAAAAqABLH8V+tu1lbd7phwKf+/KHtJKDN899OblPx1Omlh/vHV898QAAAGApSydPvLzhUhcf\nl9jjHwf5f+zm3bxxXZec2z9fuZEhhHBq4Danf4/fD9mdOnWqGqICAACgLJYWu6NHjwpRq0GD\nWkIIYc67fStPCH2DBg2EEEJkpqVlVldAAAAAWMbSYpeaykp1AAAANs26e70CAACg2lg0Ync3\nJfGWU5M29Z2KfyzKTVn7zqrj56417tSjX8j4fh3rVmdCAAAAWKScEbv8X76b3NfXs3nbCZ8k\nFR8pyrvYt5nftLdXbNuz8/35M4I7N37qw2PVnxMAAADlKKvYmQzXh7bvve7wxWYdH+vv71F8\n8PDfQr76Ja9h75f+c+T4jg1/D/Cy2/Ry7/VXf7VKWgAAAPypsh7Fnlv5xP5buaHhX+19uWfx\nEbMp5/k9KfZOrb6KXtlKby96dR80oIlb49FLpnz13L7BVgkMAACA+ytrxG7dB+d1bj13/6/V\nCSGyr3/4U35Ro74rW+nvNUIXn5Hj6jnfiltfvTEBAABQnrKK3cGMgtrNpmh/dyRl2x4hxCNv\ndvn9ad1dHQ1ZcdWSDgAAABYrq9j9XGB09Kj9+yMH1l/SaLSvt/P8/UG9ncZsLqqWdAAAALBY\nWcXO38UhO+W7kh9Nhb8sS8lyqju6vfP/ezPvUEa+Y62O1RUQAAAAlimr2L3U3jPr6qId17KL\nf/xpX9gvhUaffpN/f0729c3bf8nz6hBWjRkBAABggbJmxQ5eP13T/o0nH+x1cf6LjeyvLZm9\nRwjx3Nu/Dc6l/xA1YcCLRrN56urAak8KAACAMpVV7Dz8Zv/3re8ff3PXgleeKz7SbvyGV5vV\nFkKYjdkdfP0SL/9sMJl7ztgxu71nGb8HAAAAVlDOlmL95u78+fFDn+6NScu1b9t14KQR3f73\nifFc8q1m/n1efH3BrDGPVXdKAAAAlKv8vWLrdez7ase+pQ5qtG6FRfnl7EcGAAAAK6p8N6PV\nAQAA2BTqGQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAg\nCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0A\nAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQo\ndgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCXulA1SC+Zdr2XUb\nu/7vR9OZr/7z9Ynz2SZdc7+ug4J71NZqlEwHAIoyGo1ffvmlv7+/0kGqRd26dZWOANg0lRW7\nlAMRT06bf878/p3EZ4QQebe+mhA8evfpmyUnOHs/tGLrF5N7eSuXEQAAQBlqKna3Ty33Hfiq\nQePS72+NhRBm46+jOz2+70bOgwOfHhXUpVFt0w/f7V+9MSqsXwePlCujGroonRcAAMCq1FTs\nVo9eYtA4b4hPfqZLXSFE6tFJ+27kPPTaFyeWPX7vjOemvvq3NU16TH159O5RsROVzAoAAGB1\napo8sSYly+OBlcWtTgiRsuWsEGLj/P6/P6deQNjyNp63T76rQD4AAABFqWnEztPeLkNXMmdC\n2DnaCSGa6Er/EVrU1RuTUq2aDABshlar7dWr18qVK5UOUvWGDRum0+mUTgHYNDWN2L3cziP9\nwqvfZBqKf2z59GNCiMUnbv3+HHPR3SWnbzt5DVYgHwAAgKLUVOzGfb7EoehaH98+a3bFZhaZ\n6nZe8+ojDf4RPPifR5KLT8hN/W5GaKdjWQW95r+hbFQAAADrU9OjWLcHJp3aeaPP2MVTRvSc\nrnNv1faBBm4NCzK/fzaw5bS6TRq5FCRdvWU0mx957oO9L/oqHRYAAMDa1DRiJ4RoM2x+curZ\nFfPCerStd+PCia++Pll8PPuXn1Lz9EGjJm86cunouun2LFEMAABqHjWN2BXTefjNeGv1jLeE\nMBem376dk1eoddS71PJwq+WgdDQAAAAlqa/YCSGE2fBjwunEH5PvZGXnGYyOepd6Pk19/Tq0\n8qmtdDIAAADFqKzYGQuu/WPh3PD1Oy/fyS/1kUajadYpKOy1RTNH91AkGwAAgLLUVOyM+VeG\n+XfadynTy6/H2NCOTnnXj0VHJ2Ya+k6ZE6DPupjw/eHDX84ac2j7wdXfbghTOiwAAIC1qanY\nxU4ZuO9S5oj39m6ZFeqgEUKIotzkV3p1W7/l6CepMT6OdgXpiUtfGLJ445Tx/Qd9Pqq50nkB\nAACsSk2zYuf/K8W18fSdr95rdUIIe+cWS/cuyU//+tnIq0IInWebhdtODPR02jfzAyWDAgAA\nKEFNxe50dqHngyGlDrrUHyuESN51rfhHjZ3LtPaeuTe3WjscAACA0tRU7B52dbybEFXqYP7d\naCGEvoG+5Ejc1Ww7B0+rJgMAALABanrHbsG4Fj0jVox6/7HNs4Y6aoQQwmRIXTB8uhAidHJr\nIYTZlBO9curiq5mN+q2x8HcajcaoqKj8/NJzbH8vJSVFCGEymf5afAAAgOqlpmLXffne4D0d\nd7427OD6zj0f8nPIvnHqaGxypqFR8JIlbT2EELu6thh58pbOvfOW7SMs/J0xMTGhoaGWnHnl\nypXKRwcAAKh+aip29vpW+xK/nfdiWMS2ryOTTgghtDqvEdPnb1g+tfgEndcDo8LC5r01299D\nZ+HvDAwMjIyMLHvELiIi4siRI82bM80WAADYNDUVOyGEg6vfss0xS9bdupCYUmDn0trP183h\nt9cEQw7Elp5bUR6tVhsSUs5FUVFRQgg7OzW9jwgAAGoglRW7YvbO9fw71VM6BQAAgG1hFAoA\nAEASshU7Q9Yxb29vb29vpYMAAABYmyofxZbBbDakpaUpnQIAAEABshU7x1pd4uPjlU4BAACg\nANmKnUbrGhAQoHQKAAAABai12N1NvZKYmHQzPSsnN99e7+Lm1aB1W98W3u5K5wIAAFCMyoqd\n2Zi5I3zRhxu3HL9484+fNmjbbdyk6W9OH+1ur7F+NgAAAGWpqdgZDdef6dph09k7WgfPgD6h\nD/q29K7jrtPZFxUUZNxOu5p07njsNytmjf1syxdn4j5r6CjbhF8AAICyqanYxc0csOnsnUen\nrNz67kuNXO6T3GS4s3VZ2MQFW/pNnXTuo95WDwgAAKAkNQ1rzdmUVMv7hdhV0+7b6oQQdo5e\n49/ctjag/uVt86ycDQAAQHFqKnYJOYW1mpS/GWznnvUKc89ZIQ8AAIBNUVOxG+LldPfiu2kG\nU1knmfI+3pGi9wi2VigAAABboaZiN3dZcEFmbPtuozbvP5FjNJf+2FxwPnbPpH6+a1Oyei9Y\noERAAAAAJalp8kTrp3au/67/5IjdEwfs0jq6tWjdsmFdd53OwWgoyLydmpx0OT2/SKPRBL60\nJjLMV+mwAAAA1qamYieE3aTVhwZO/Peaf26Niom/eOFU0rl743YaO12jlu36BQaPnTRtSFcf\nZVMCAAAoQl3FTgghfAKGLg0YulQIc1FeRsavOXkGRydnV3cPJxYlBgAANZv6il0Jjb2TRx0n\nD6VjAAAA2Ag1TZ4AAABAGSh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiC\nYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAA\nIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYod\nAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJKg2AEAAEiCYgcAACAJih0AAIAk\nKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAA\nAJKg2AEAAEiCYgcAACAJih0AAIAkKHYAAACSoNgBAABIgmIHAAAgCYodAACAJCh2AAAAkqDY\nAQAASMJe6QAAYG1GozEqKio/P7+iF5pMpoSEBH9/fzu7iv2rWK/XDxo0SKvVVvSOAFAhFDsA\nNU5MTExoaKiVb3rw4MG+ffta+aYAahqKHYAaJzAwMDIyshIjdnFxceHh4TNmzOjevXuFLtTr\n9YGBgRW9HQBUFMUOQI2j1WpDQkIqd214eHj37t1HjhxZtZEAoEoweQIAAEASFDsAsJSTk1PJ\nVwCwQap/FGsy3Pzyi/+evXTdpHNv2+mRAY91sNconQmApAYOHLh///6goCClgwDA/amp2AUG\nBjrVGRa1c1rJkaQ9Swc9s+hSpqHkiJdf/w07tw/1c1ciIADJabXa/v37K50CAP6UmordkSNH\najVsX/JjZlJEx5Fv5pkdgp+c2qdz+4au4odvoiM2RI7q2v3QjdM93XQKRgUgJaPRePjw4aCg\nIFakA2Cb1FTsSvl81MI8k3nBf39c0L/JvUPPPD/jyeU+j7769LOHknc9rmg6ABKKjo4OCQnZ\nt2/f4MGDlc4CAPeh4skTyxPvujae+1urE0IIUb/HzMXN3W4cWqJUKgASy8vLK/kKADZIxSN2\ntwtNtZo+8sfjXZrVKrz6g/XzAICN+OGHH2bOnKl0iqqXlpbWtGlTpVMANk3Fxe7JBi6f/RQn\nxIBSx2MvZTnU6qBIJABQXNu2bc+ePXvgwAGlg1SLtm3bKh0BsGkqK3b56VFPT7Zr1apVq1at\n+kzttub1t+cfmLS4f+OSExJ3z3n7p6ymg2crGBKArFSxjt2ZM2eUjgBAMWoqdg8/2OZScvKn\n6z78/cF3hvZenHtZCCHMRU8+3n3Lf09odT4ffcJO2wCqHuvYlcI0YcDWqKnYfXPmohAi4+ZP\nly9dunz58qXLly9fvnzlp6z/fW7cFP29R+ueK7fvDvbSK5gTgNXk5uZu3769sLDQmjfduHGj\ndW7k4OAwevRoZ2dn69yuEpgmDNgaNRW7Yu71m3Su36TzI31Kf6BxOJ5wpVv7Zmw8AdQc27dv\nf/bZZ5VOUb2eeeYZpSP8KaYJA7ZGfcVOCCHMhh8TTif+mHwnKzvPYHTUu9Tzaerr16F7+2ZK\nJwNgVQaDQQjxyiuv+Pr6Kp2lil24cGHFihXFf0AAsJDKip2x4No/Fs4NX7/z8p38Uh9pNJpm\nnYLCXls0c3QPRbIBUIqfn19AQIDSKaqYRsPjBwAVpqZiZ8y/Msy/075LmV5+PcaGdnTKu34s\nOjox09B3ypwAfdbFhO8PH/5y1phD2w+u/nZDmNJhAUB+qpgmDNQoaip2sVMG7ruUOeK9vVtm\nhTpohBCiKDf5lV7d1m85+klqjI+jXUF64tIXhizeOGV8/0Gfj2qudF4AkBzThAFbo6Ytxeb/\nK8W18fSdr95rdUIIe+cWS/cuyU//+tnIq0IInWebhdtODPR02jfzAyWDAkDNoNVq+/fvz1on\ngO1QU7E7nV3o+WBIqYMu9ccKIZJ3XSv+UWPnMq29Z+7NrdYOBwA1j9FoPHDggNFoVDoIgHvU\nVOwednW8mxBV6mD+3WghhL7BbwvXxV3NtnPwtGoyAKiRoqOjg4ODo6OjlQ4C4B41vWO3YFyL\nnhErRr3/2OZZQx01QghhMqQuGD5dCBE6ubUQwmzKiV45dfHVzEb91lj4O41GY1RUVH5+6Tm2\nv5eSkiKEMJlMfy0+AMiGdewAW6OmYtd9+d7gPR13vjbs4PrOPR/yc8i+cepobHKmoVHwkiVt\nPYQQu7q2GHnyls6985btIyz8nTExMaGhoZaceeXKlcpHBwAAqH5qKnb2+lb7Er+d92JYxLav\nI5NOCCG0Oq8R0+dvWD61+ASd1wOjwsLmvTXb30Nn4e8MDAyMjIwse8QuIiLiyJEjzZszzRYA\nANg0NRU7IYSDq9+yzTFL1t26kJhSYOfS2s/XzeG31wRDDsSWnltRHq1WGxJSzkVRUVFCCDs7\nNb2PCABWwDp2gK1RWbErZu9cz79TPaVTAEBNxzp2gK1RZbEDANiC4nXslE4B4DeyPV40ZB3z\n9vb29vZWOggAyI917ABbI1uxM5sNaWlpaWlpSgcBAPmxjh1ga2R7FOtYq0t8fLzSKQCgRmAd\nO8DWyFbsNFrXgIAApVMAAAAoQK3F7m7qlcTEpJvpWTm5+fZ6FzevBq3b+rbwdlc6FwCokiXb\n8PxRXFxcydcK0ev1gwYN0mq1Fb0QQNlUVuzMxswd4Ys+3Ljl+MWbf/y0Qdtu4yZNf3P6aHd7\njfWzAYB6Wb4Nzx+Fh4eHh4dX9KqDBw/27du3cncE8GfUVOyMhuvPdO2w6ewdrYNnQJ/QB31b\netdx1+nsiwoKMm6nXU06dzz2mxWzxn625YszcZ81dJRtXggAVB9LtuH5I5PJlJCQ4O/vX9El\n3PV6fWBgYIUuAWAJNRW7uJkDNp298+iUlVvffamRy32Smwx3ti4Lm7hgS7+pk8591NvqAQFA\nrSzZhue+Ro8eXeVhAFSamoa15mxKquX9QuyqafdtdUIIO0ev8W9uWxtQ//K2eVbOBgAAoDg1\nFbuEnMJaTcr/B2XnnvUKc89ZIQ8AAIBNUVOxG+LldPfiu2kGU1knmfI+3pGi9wi2VigAAABb\noaZiN3dZcEFmbPtuozbvP5FjNJf+2FxwPnbPpH6+a1Oyei9YoERAC8iSLQAAHwRJREFUAAAA\nJalp8kTrp3au/67/5IjdEwfs0jq6tWjdsmFdd53OwWgoyLydmpx0OT2/SKPRBL60JjLMV+mw\nAAAA1qamYieE3aTVhwZO/Peaf26Niom/eOFU0rl743YaO12jlu36BQaPnTRtSFcfZVMCAAAo\nQl3FTgghfAKGLg0YulQIc1FeRsavOXkGRydnV3cPJxYlBgAANZv6il0Jjb2TRx0nD6VjAAAA\n2Ag1TZ4AAABAGSh2AAAAkqDYAQAASIJiBwAAIAmKHQAAgCQodgAAAJJQ8XInAFBs0qRJSkcA\nAJvAiB0AAIAkKHYAAACSoNgBAABIgmIHAAD+r707jYuqbOM4fp3ZGDZZTAVRyX0JStREzUpU\nVMw1cy0f9dGyMjXLVu3RckktxTS1RUsrrSwrN1pcqMzUVpfcyhDcclcQgQFm5nkxSso6MwwM\nc/x9X/hhzrnPua9zD978P3OWgUpw8wQAj7d48eLo6Gh3V+FiO3bs4KYQAI7iEzsAAACVINgB\nAACoBMEOAABAJQh2AAAAKkGwAwAAUAmCHQAAgEoQ7AAAAFSCYAcAAKASBDsAAACVINgBAACo\nBMEOAABAJQh2AAAAKkGwAwAAUAmCHQAAgEoQ7AAAAFSCYAcAAKASBDsAAACVINgBAACoBMEO\nAABAJQh2AAAAKkGwAwAAUAmCHQAAgEoQ7AAAAFRC5+4CAKC09u3bZ7Va3V2Fi+3fv9/dJQDw\nPAQ7AB7MYDCIyJw5c9xdSFmxHSAA2IlgB8CD9evXT0RycnLKpzuLxXLgwIFGjRppNOVxHYte\nr7cdIADYiWAHwIP5+voOGzas3Lpbt27dI488snbt2m7dupVbpwBgP26eAAB7ZWZm5v0LABUQ\nwQ4AAEAlOBUL4IZjNpsTEhKysrIc3XDbtm15/zrEaDR27dpVq9U6uiEAOIRgB+CGk5iY2KNH\nD6c3j4+Pj4+Pd3SrDRs2dOzY0elOAcAeBDsAN5yYmJg1a9Y48YmdxWLZs2dPZGSko3fFGo3G\nmJgYR7sDAEcR7ADccLRabffu3Z3btn///q4tBgBciJsnAAAAVIJgBwAAoBIEOwAAAJUg2AEA\nAKgEN094POeeyFWam/t4HBcAABUTwc7jlfKJXE7gcVwAAFRMBDuP59wTubZt2xYfHz9u3LjW\nrVs7tCGP4wIAoMIi2Hk8p5/IFR8f37p16759+7q8JAAA4BbcPAEAAKASBLsblLe3d96/AABA\nHTgVe4OKi4v7+uuvO3To4O5CAACAy6jnE7vBgwePnb7H3VV4DK1W26lTJ55aAgCAmqjnE7sP\nPvggrN0Drz0f6e5CnDd//vy5c+e6u4qyEhoaunHjRqPR6O5CAABQLU8KdknL575/KLWYBpeS\nl7/44nbbz5MmTSqXolxp8+bNhw8fbty4sbsLcb1Tp04lJSWdPn26Vq1a7q4FAADV8qRgd+Sz\n+ZM/SyqmQVry+5MnX/nZE4OdiOh0uo8//tjdVbje7Nmzly5d6u4qAABQOU8Kdnd9uHXGo/2f\nXfK9Mbjp1PkT6/leV3yvXr0qR0xaMjXKXeUBAAC4lycFO40h5JnF33XtOrPPkBcmjp0+Z8Un\nj8TWubaB8abWPXt2dld5AAAA7uV5d8VG3vvMnuQdQ287P6pzg7gx887lWtxdEQAAQIXgecFO\nRLwqRy3a+PfqVx/8cdG4uo3jPt151t0VAQAAuJ8nnYq9nqb7E4tSOscN6jOkf4vwwS++4+56\nXMNisURGevATWwAAgBt55Cd2eQJv6bH+j7/nPnrX+y8MdHctAAAAbua5n9hdoeiCR8/7smv3\n99btu+BXQ4VPgAMAALCTZwY7a/afe3Ye/DPpXFp6ZrbZYPStGlb7nvt61Qur5O7KAAAA3MbD\ngp3ZdPSNyRPi3/7k73NZ+VYpinJzVIdRT7/4ZP82bqnNJSwWbvIFAABO8qRgZ8463Dsyau2h\n1MpN2gzs0dQ78/jWL788mJrd8bHno41pB/b8smnT5vEDNn684fWfFo9yd7HOCAsLs1qt7q6i\nrPj5+QUEBLi7CgAA1MyTgt2Wx+LWHkq9b9bqFeN76BURkdyMpCfubvX2ih+W/pMYZtCYzh+c\n/nDPl5Y8dn+nrsv71XZ3vQ6bN2/elClTHN3KbDZv2LDBZDI5tJXFYtm3b1+TJk00GsduoPHy\n8oqNjdVqtQ5tJSJGo9Hb29vRrQAAgP08Kdj979Nk/5pjP3mqR94SnU+d6aunzQ976L9rUr6+\nr7ZXcMPJH/3686Yqa5+cK/1ec2OpztFoNEFBQU5sOHAgNwUDAACPetzJzvSc4Fu751voW22g\niCStOmp7qWh8x0QEZ5z6sLyLAwAAcDdPCnYt/Q0X9iTkW5h14UsRMYYY85ZsS0nX6IPLtTIA\nAIAKwJNOxU4aVOeuhXP6vXLnB+N7GRQREUv2P5P6jBWRHiPri4jVcvnL10a/lJJaI3aBnfs0\nm80JCQlZWfnvsb1WcnKycL8qAACo8Dwp2LWevbrz500/ebr3hreb39WsiT79xO8/bElKza7R\nedq0RkEisur2On1/O+0V2HzFx/fZuc/ExMQePXqU3E7k2LFjzpcOAABQ9jwp2OmM9dYe/Gni\nI6MWfvT9mr9+FRGtV+X7xv5v8ezRtgZelRv0GzVq4pRnIoO87NxnTEzMmjVriv/Ebv369cuW\nLRs0aFAp6wcAAChTnhTsRETv32TmB4nT3jq9/2CySeNbv0njAP2/lwl2/2ZL/nsrSqLVart3\nL2GjEydOLFu2TK/XO14vAABA+fGwYGej86kaGVXV3VUAAABULJ50VywAAACKobZgl522NTQ0\nNDQ01N2FAAAAlDePPBVbDKs1++TJk+6uAgAAwA3UFuwMfi22b9/u7ioAAADcQG3BTtH6R0dH\nu7sKAAAAN/DUYHfhn8MHD/516nza5YwsndE3oHJI/UaN64QGursuAAAAt/GwYGc1p66Mf3He\nkhU/HjhVcG1Io1aDRox9YWz/QJ1S/rUBAAC4lycFO3P28WG33/b+7nNafXB0+x63Nq4belOg\nl5cu12S6ePZkyl97f9yyY874ge+tWLdr23vVDWq74RcAAKB4nhTstj3Z5f3d59o+9tqHMx6t\n4VtI5Zbscx/OHDV40orY0SP2vtmu3AsEAABwJ0/6WOv59//yC314y/wxhaY6EdEYKt//wkeL\noqv9/dHEcq4NAADA7Twp2O25nONXq+Qvg21+V9WcjL3lUA8AAECF4knBrmdl7wsHZpzMthTX\nyJL5zspkY1Dn8ioKAACgovCka+wmzOy8bOhnEa36zX35ud4dm/lqr7/11Wra90PCnMnjliSn\ndX19kst7P3jwoNFodPlu3SUnJ2fp0qXh4eEajSeF+zJlsVgOHTpUr149xsSGASmIMcmHAcmH\nASnIYrGkpKQMHTpUr9e7uxaXOXjwoLtLKJrVk5jfHtVBoygiojUE1L+l2d3t2nfq3LlDTLsW\nkQ2DjToRURQl5tEFuS7tdcGCBe5+lwAAQMWyYMECl8YN11CsVqu7R8Yxx3d8seDdDxMStx84\ndNRkuVK8ovGqUbdRm5jOA0eM6Xl7mGt7TE1NXbZsWWZmpmt36167d+9esWJF27Ztw8PD3V1L\nRZGSkvLDDz8wJnkYkIIYk3wYkHwYkIJsYzJo0KBbb73V3bW4kre395AhQwICAtxdSAHuTpbO\ns+RknD9z6uiRo6fOnMvIsbi7HA+zcuVKEVm5cqW7C6lAGJN8GJCCGJN8GJB8GJCCGJNy5knX\n2OWj6LyDbvIOcncZAAAAFQRXdwIAAKgEwQ4AAEAlCHYAAAAqQbADAABQCYIdAACAShDsAAAA\nVIJgBwAAoBIEOwAAAJUg2N2gvL298/6FDWOSDwNSEGOSDwOSDwNSEGNSzjzvu2LhEmazedOm\nTR06dNBqte6upaJgTPJhQApiTPJhQPJhQApiTMoZwQ4AAEAlOBULAACgEgQ7AAAAlSDYAQAA\nqATBDgAAQCUIdgAAACpBsAMAAFAJgh0AAIBKEOwAAABUgmAHAACgEgQ7AAAAlSDYAQAAqATB\nDgAAQCUIdgAAACpBsAMAAFAJgh0AAIBKEOwAAABUgmDneY7/9Pno//RqHF7d31vv4x/UqPnd\nY6a+9U+2pTSNq3vptPrgsq/dSQ4dclE2xoUrirLtUnahLx3q6/uB9RVFSUw1leag4C7roqop\nipJsMsvVt7J4CRey7Nltvt+oB0P9vYM6luFhOOLAm3coimIMbHs821xwrW0Qph29VHDV75Oa\n2QbhsZ1nC93KnqHb+1or28vBP/xTaHlpKTNtDer0TsxbWHEmJSfqP/FdnKIokeN+KscygSt0\n7i4Ajlk5sceA6etEpGbE7W0ibks9nvznH1vn//b9u29+vGFPQqtAL6cbV1jleRTl1ldaysTw\npgtaL/wlYWBdV+0TTqjcIrZXVkTey5OJX25PNTXs1K2xz79zYzW91h2luZ4pdWunMQl73+hu\n/yYvvH7A9sMXT2x6fXP/a1c5MXQbx38l24cV7GXXlHftL8mNKlT9zCEokhWeY+eCPiISUK/n\n2t2n8xbmXD429+FmInJT1LNONw41aDW6oLKu3wkOHUXxNnSpJSI/ppkKfelQX98NqCcimy9m\nOX1cF5OeFJE7l/7p9B7gtPTkQwcOHMixFLJqbdOqIjIhOdWJ3eb7jRoR4mcM7FCaOl1o/xtt\nRESj0yiKfsGBC/nW2n6fpx5Jy7c87Wi8iATWfzrUoNUaQk5lm4vpopih+2NutIj4VvfW6IOP\nmQruxNw2wEvnEyoitXttzltacSYlJ+o//m0XEYl4fEfZVcUcgqJwKtZj5KT/1v7xzw1+Tbfu\n/LRbZJW85TqfsLELtw0L9Tv7+4zXjqU70bh4FpODpzxdx4VHUaH6gnv5htdt2LChTnF3HeUu\n+s1ntJL7TMeHL1us9rT/deIiEWk3d9TsVtXM2SfHbj1Zmt6bvhxnyTn/xNb8ZzMvHZn9Q6qp\n6Qs9SrPzclB29btxjoUqEew8xh+vjDifY2kz56NbfAucQFcME2YO6dKlS/Kv55xoXJDt6pnc\nzD/HdW/p42PUaY0160cOfnpRmtkqIhmn1lQ2aAPqDMu6ZjZaeE+4RqOfs6vIfTrB0aOwmlOX\nvzymTZPwSt5eVWvWi33gyW8OpJZRXyJiteR8OmNUs9qhPgafmvUjH3hqgW18bNJTvh0/uHvD\nsCpGvd4voGqzu3u99vkftlWL6gcH1pktIluGNlAUZcE/l+0sEi7xZevqedfYOaSY97REl098\n0yLQqPeuu2LfRUf7dZXgW8auHNYw/djHcTN+Kbm11fT4qmSNPnheh7D2MzuJyOYnV5em9ypR\ns2p56b59an2+5bumvKMo+lm9wkuzc3tseX96XKuIIH9vg7dfvdvufO719dfG2+zUfS+N7Nsg\nrIqXwbdGveYjJ751Jue6uOXa+oufY22KmdCYQ1Acd39kCHs9WcNfRL67aCq5qYONrQXOethO\nzTzZsqrer37fYY89NXrELUFeItJkeIKtwa+vdhSRDq/8Znt5fNN4Eblt9HpHDsjFR2Expz/W\nNkREghu3HjB0eM/YNl4aRWuo9uq3/9gaFH8q1qG+bOMzvG8jvV+Dfv8d/dTo4U0qG0WkyYiv\nbA0yTq+52ahTFH2LLvcOH/nQgN4dgnQaRdE8++NJq9W698N346d2FJF6Q1564403/ric49Cw\noJQSWoWKyOGs3IKrijmfWPx7ai32VGzGyc3RQUadsdbSXefK5phKYDsVe8/2k7lZKdGVvDS6\nwLWnMvLWFnoq9uyeJ0QktO17VqvVnHMmxKDVaP0OZhT5u1riqdh7951dFVtTowtIvm7kzXcG\neAXW+d+lY69KWZ6K3TGts4h4V72l/+DhIwb3bxjsJSIdX74yg5nSdrSt4q0omojWscOGD465\ntaqIVLn94SyLk/WXeCq2xDm2+AmNOQTFINh5jHCjTudVsywaW4sIdt6VO+w4nWlbknVxazWD\nVu8beaWFxTSyYaBGH5xwJiM3K7lVJS/fkB7ncoq7BMcJDh3FrhltRaT5uPdMV6+dOrnjg+pe\nWoNf1Lkci7WkYOdQX7bxMQa33351fLIv7arppTP4t7C93DE2QkQGLD+Yt8nZna+KSNjdV5If\n18e4kXPBrsT3tKhgl3lmS9ubvLVeYYt/Pev6g7FPXrCzWq3HNowVkZuaPp3337XQYPdpp5oi\nMvLnU7aXy++sLiJxq5KK6sKeYHfm91Eicu/XR/NWpR15VUTaf/BXGQc7Sx2jzuDfIu9NN6X9\nEqzXGIM62l6+1T5MRMZ8vPdq+9y3BtYVkYEbjjpXv53Brpg5tsQJjTkEReFUrIew5hwxmbVe\nNV3fuGjt31nSsorR9rNXQJsHQ3zNpmNX1imG2ZuX+Fsv/id28qqRnXaky5xvlwXrXPrr5OBR\njJnxs1elOxJfecBw9dqpai3vXzmiYXb67zNSSjoh69SItVu8JPrq+Oj9bh1azcdsOm57GRb7\nwtKlS1/vVy+vcWCjviJiOpPpUBeoOJx7T00XdnSN6Lz1vMz/7vfhzSqXeZV2COs4d06HsLM7\nZw1afqioNpacM2O+O6H3rv9q1JXrTTvM6iQi2599vzRdV46cUcdb9/3TX+Qt2T11iaLRz+pZ\ntudhrZaMIyazVl8tb44y+Df/6edftm6cLSK5GftGf3sisO7zr/VrcnUL7eD5r7Rq1Sp363VX\nX7i8/mLm2FJNaLix8bgTD6HoQ/SaM9nHXd+4aP1bVbn2Zb7c5lv93o1T77792Vn9d0r0c4kP\nNQwsZXf5OXIUOem/fnfR5BfaeOXSd65dftFXIyI//XJO6hZbnlMjdv8dVa99adT8ezV+2D39\nhohYzRmH9/+ZlJycnPT3lrULHdo5Khon3lNz9okeER0ST2WIyKHM3HIp0y6jPl81t2rbVQ92\n+b3X/ihffcEGJ38ce8JkrtNnnp/2ym91leavhBjeP3Vo6vZLz7byNzjXr6L1i28f1vur5w5l\nPVLPqBWxTFiZFFB7QnM/fXpZZhVF4zsjpvr4zetrNrxz6KCed9/RplXrlnVvi7KtTT+xwGSx\nNnngvms3MVbuvW1b77Kuv6g5trQTGm5sfGLnMeKCjblZKVvS8j9Q18Z0cVPv3r0HDV/sROOi\nVNaX8OvRdMwb/lqNoijPPB5dUvnOsP8ocjP/FJH0fxaPuN74WX+ISOaJkj8nc2LEqhuKfLxZ\nbsaBiUM6VfapVKdJVGy3fi/MXnI6oF2JNaAic+I9zcnYn5gavnTTfC+NsrD3oNM5FeXeR4N/\n9Devd8vN/LtXv8Ingc/HbRSRpFVxeY8a1hqqnMw2W605T61IKk3XbWfeazGnP7HxuIhcOjb3\nu4um5i89UJod2umJr3e/M3V0I82heVOe7tOlbY0gv9s6DPj41zMiYrqQIiKVGleyZz+urb+o\nObb0ExpuZAQ7j/HgveEiMnH534WuPbnl1S+++OL7wyFONHbayoe6pVu1RkVG3jO1LP5k2X8U\nWkOYiIS0XFPoBQc7xkUUugfn+sqjFP28jAmt2057b0PM46/+sOtQusl0Imnf+hVzSqwBFZkT\n76nWUHXF7z8Oaf/YZyObZF38Nm7ilvIp1R4Nh386uknwkYRHn/s+/0NMcjMPPLP7nNYQmi9V\nDBl4u4jsfGl+afoNajytkY/+x2c/FZE90xYrGv0rPWqVZod2UnTBwybM2/HnyYtH96/78O3H\n/9Pp7+8+ub9NxJa0bH2lYBHJOJJhz37Kp/7ST2i4kRHsPEbUtLk+Ws228X1+SS3wqZLVNHXU\nVhGJe/l2Jxo75+SWFx5Yfihi1Jp1T0Sd+WX6wPf+Ks3eCmX/URgC2jbx0aclLc2XLw+9P23c\nuHFbi/gczrm+SpSbsXfW7nOBdV9ZNfPxO26t66NTRMSSc8aebVExOfee6n0i76sfICKd4xNa\n+Bt2zu722Um70kO50L684U0/rSa+Z7983zOWsnrMZbOleszrb1/vnfe+quGlSz+xcPU5u75j\nrVCKxntOpxrnD0zcn5Ez8aPDAXVeiPIr5Fywa2WdW/3cc8/NWZUiIgE1Gt0zYMScd9d+/2KU\nOfv0jL3n/UJGKIqStOyrazfJvrRNq9FUvW25W+ov/YSGGxnBzmN4BcYm/C8mJ2N/TGSPVT//\nezVYbsbxmcOiFx+9VOnmwQtbVnOisRPMWYd63jPLGNxx0+xO7V7+pstN3p+NjN3m6unGkaPQ\nLPpvw4yzn3V5cU3eVHjp8Lq4kZMXvbOjqR0zrytHTNFpFCU346/cq0+ksuSceX3UvSIict1f\nUEtuRTk3hxLY/Z4WSutVa9XywRZz+sguU+x6NHC58K1+31fPtzRd3DL0q6PXLl/8/E8iMnBO\nu3ztNbrgee2qi8hLV79nzDmtXu5nNWc+tPipxItZLaYOKs2u7GadMWPG/0ZPPPfv/zjrT7+f\nF5HIat6GgLsmRQSf3/fMhLV5H9hbPx033GK1Rk9s7ab67Z3QmENQiLK85RYuZ35rdHvbGxd2\nS3TXHr06tWtzk0ErIr5hd20+neF040Ifd7L+fOa1bebUCcxrs7hPbUXRTP/ljO3lmV+naxQl\n9K6X3XjIZtPxPg0DRaRKg+Z9hz08uG+XQJ1Go/WbknjC1qDErxSzv69Cv1Js2s0BWkOo7eeX\n24aISO229z0zcdKYBx9oVs0npOWAml46ve8t0+e+abVa047OEpHABvdOfnHS1lR7HzcIl3Du\ncSclvqclfqXYhGZVRGTIZ4ddf0glufZxJ9ey5F7sGeJr+523Pe4k68JGraIY/FuYCvvKtXN/\nPCki3jf1LrjKnsedXO01K9JXryiKojH8np5tW1bWz7GbHlNdRHzDmvYZ9N9HHxwaE1FNRKq1\necL2zXIZpxJu8TcoirZ5u3sefGR4l+gaIhIcMexSrsW5+m2POwmo3/6+Agbc/7jVjjm2xAmN\nOQRFIdh5ngOb3hvWJ/bmkMpGndbHP6hRi5gxUxcfKeyvlP2NHQp2x75+QkTqD/7o2rUrBtUV\nkeFl80fLzqPINR2d/8ywqDqh3np91VoNYnqOWPXrv9/6akews7evEoNdblbKlJE9a1etZPAO\nurVVhzGzPjVZrJsm3BvorfcPaWa1Wq3mzAl9Wwf66A0+QctOXXbBGMFuzgW7Et/TEoPd5VNf\n+Gs1Br+mSZmF/28tO0UFO6vVem7Xq1pFyQt2u2e1FJHGD20pfEeWrNv8DCKy6ER6vjUOBDur\n9Zt+dUUkqP6UvCVlHezM2WcWPDc8qkENH4NWZ/StE9l69JR3z13zhcGXT2x/enC32tWC9Dqv\nKuERg5+K/+fq18I6Ub8t2BVK79PEakews5Y0oTGHoCiK1VpxzgwAAADAeVxjBwAAoBIEOwAA\nAJUg2AEAAKgEwQ4AAEAlCHYAAAAqQbADAABQCYIdAACAShDsAAAAVIJgBwAAoBIEOwAAAJUg\n2AEAAKgEwQ4AAEAlCHYAAAAqQbADAABQCYIdAACAShDsAAAAVIJgBwAAoBIEOwAAAJUg2AEA\nAKgEwQ4AAEAlCHYAAAAqQbADAABQCYIdAACAShDsAAAAVIJgBwAAoBIEOwAAAJUg2AEAAKgE\nwQ4AAEAlCHYAAAAqQbADAABQCYIdAACAShDsAAAAVIJgB+CG8/3A+kpJEi5kuaSvjXHhiqJs\nu5Ttkr0BQPF07i4AAMpb5RaxvbIi8l6eTPxye6qpYadujX3+nRKr6bXuKA0ASkWxWq3urgEA\n3GldVLXuO09PSE6dGl7J5TvfGBce+9WRH9NMrf0NLt85AOTDqVgAAACVINgBQHFy0g/OHD0o\nIjzEW+9VOaR21/vHfZt0yaEGAFBuCHYAUKTcjD2dGjR/9vUPUwPr9h48sHndSt98+FrsLRHv\n/Z1mZwMAKE8EOwAo0ur7e3z7z+VO0746umvrineWfrN114HVEyymo6M7jLezAQCUJ4IdABTO\nak59cN0RY3CX9c91zltYr/uU16KqpKW8/dGZzBIbuKNqADc0gh0AFC7jzMoLuZZqrZ/UKdct\n7zS6gYgsP5RaYoPyqxUARIRgBwBFMZtSRMS/fv5noFRqXElE0o9mlNigPKoEgGsQ7ACgcFqv\ncBG59Ff+W1zTD6WLiE917xIblEeVAHANgh0AFM7npr6BOs3pbfHm65dvmn9QRPo3CCixQXlV\nCgBXEOwAoHCKLvCtuJqZ59f3fCUxb2FSwuRRP52uVGvEf6r6lNjAHVUDuKHxXbEAUKReH66+\nq26b9U+3r72y3d3N6589+NtX3/2meIUvTJxtZwMAKE98YgcARdL73rbxr1+mjervc2rvx0uW\nbt17NnbA2I379txfp5KdDQCgPClWq9XdNQAAAMAF+MQOAABAJQh2AAAAKkGwAwAAUAmCHQAA\ngEoQ7AAAAFSCYAcAAKASBDsAAACVINgBAACoBMEOAABAJQh2AAAAKkGwAwAAUAmCHQAAgEoQ\n7AAAAFSCYAcAAKASBDsAAACVINgBAACoBMEOAABAJQh2AAAAKkGwAwAAUAmCHQAAgEoQ7AAA\nAFSCYAcAAKASBDsAAACVINgBAACoBMEOAABAJQh2AAAAKkGwAwAAUAmCHQAAgEr8H69Sie4H\nDpXgAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Box plot of SSP\n",
    "box <- cbind(CellChat_SSPA,CellChat_SSPB,CellChat_SSPC,CellChat_SSPD,\n",
    "iTalk_SSPA,iTalk_SSPB,iTalk_SSPC,iTalk_SSPD,\n",
    "SingleCellSignalR_SSPA,SingleCellSignalR_SSPB,SingleCellSignalR_SSPC,SingleCellSignalR_SSPD,\n",
    "CCInx_SSPA,CCInx_SSPB,CCInx_SSPC,CCInx_SSPD,\n",
    "scMLnet_SSPA,scMLnet_SSPB,scMLnet_SSPC,scMLnet_SSPD,\n",
    "CellPhoneDB_SSPA,CellPhoneDB_SSPB,CellPhoneDB_SSPC,CellPhoneDB_SSPD,\n",
    "NATMI_SSPA,NATMI_SSPB,NATMI_SSPC,NATMI_SSPD)\n",
    "\n",
    "tool <- unlist(strsplit(colnames(box),split = \"_\"))[seq(1,ncol(box)*2,2)]\n",
    "\n",
    "acc <- unlist(box[1,])\n",
    "accbox <- data.frame(Tool=tool,Accuracy=acc)\n",
    "plot1 <- boxplot(Accuracy~Tool, accbox)\n",
    "\n",
    "sen <- unlist(box[2,])\n",
    "senbox <- data.frame(Tool=tool,Sensitivity=sen)\n",
    "plot2 <- boxplot(Sensitivity~Tool, senbox)\n",
    "\n",
    "spe <- unlist(box[3,])\n",
    "spebox <- data.frame(Tool=tool,Specificity=spe)\n",
    "plot3 <- boxplot(Specificity~Tool, spebox)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "7ad2b8f5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "5b95f0b6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0cb13f9e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
